<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Shixiang Wang" />
<meta name="author" content="Huimin Li" />
<meta name="author" content="Xuan Wang" />
<meta name="author" content="Minfang Song" />
<meta name="author" content="Xue-Song Liu (Corresponding author)" />

<meta name="date" content="2020-02-10" />

<title>Prostate Cancer Variation Signature Analysis Report</title>

<script src="index_files/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.6/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.6/js/bootstrap.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<script src="index_files/navigation-1.1/codefolding.js"></script>
<link href="index_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="index_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="index_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
<script src="index_files/readthedown-0.1/readthedown.js"></script>
<script src="index_files/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="index_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="index_files/datatables-binding-0.12/datatables.js"></script>
<link href="index_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="index_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="index_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="index_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="index_files/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


</head>

<body>


<div id="content" data-toggle="wy-nav-shift">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("text-table-of-contents");
});
</script>

<!-- code folding -->

<nav id="nav-top" role="navigation" aria-label="top navigation">
    <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
</nav>


<div id="header">
<h1 class="title">Prostate Cancer Variation Signature Analysis Report</h1>
</div>


<div id="table-of-contents">
    <h2><a href="#content">Prostate Cancer Variation Signature Analysis Report</a></h2>
    <div id="text-table-of-contents">
      <ul>
      <li><a href="#data-preprocessing">Data Preprocessing</a><ul>
      <li><a href="#data-downloading">Data downloading</a></li>
      <li><a href="#data-preprocessing-1">Data preprocessing</a></li>
      <li><a href="#data-cleaning">Data cleaning</a></li>
      </ul></li>
      <li><a href="#signature-identification">Signature Identification</a><ul>
      <li><a href="#tally-variation-records">Tally variation records</a></li>
      <li><a href="#estimate-signature-number">Estimate signature number</a></li>
      <li><a href="#extract-signatures">Extract signatures</a></li>
      <li><a href="#show-signature-profile-and-aetiology">Show signature profile and aetiology</a></li>
      </ul></li>
      <li><a href="#association-analysis">Association Analysis</a><ul>
      <li><a href="#data-integration">Data integration</a></li>
      <li><a href="#association-analysis-1">Association analysis</a></li>
      </ul></li>
      <li><a href="#group-analysis">Group Analysis</a><ul>
      <li><a href="#group-comparison">Group comparison</a></li>
      <li><a href="#group-mapping">Group mapping</a></li>
      <li><a href="#grouped-signature-exposure-profile">Grouped signature exposure profile</a></li>
      <li><a href="#copy-number-profile-of-different-groups">Copy number profile of different groups</a></li>
      </ul></li>
      <li><a href="#survival-analysis">Survival Analysis</a><ul>
      <li><a href="#prepare-1">Prepare</a></li>
      <li><a href="#os-analysis-fo-signatures">OS analysis fo signatures</a></li>
      <li><a href="#pfs-analysis-for-signatures">PFS analysis for signatures</a></li>
      <li><a href="#os-analysis-for-features">OS analysis for features</a></li>
      <li><a href="#pfs-analysis-for-features">PFS analysis for features</a></li>
      </ul></li>
      <li><a href="#variation-summary">Variation Summary</a><ul>
      <li><a href="#oncoplot">Oncoplot</a></li>
      <li><a href="#summary-plots">Summary plots</a></li>
      </ul></li>
      <li><a href="#supplementary-analysis-and-visualization">Supplementary Analysis and Visualization</a><ul>
      <li><a href="#facets-results">FACETS results</a></li>
      <li><a href="#method-comparison-between-macintyre-et-al-and-our-study">Method comparison between Macintyre et al and our study</a></li>
      </ul></li>
      <li><a href="#reference">Reference</a></li>
      </ul>
    </div>
</div>

<div id="main">
<p>Contact Prof. Xue-Song Liu: <a href="mailto:liuxs@shanghaitech.edu.cn" class="email">liuxs@shanghaitech.edu.cn</a></p>
<p>This report is written to help readers understand what and how we did in this project. Please read the formal manuscript <strong>TBD</strong> for more details.</p>
<blockquote>
<p>This document is compiled from an Rmarkdown file which contains all code or description necessary to (auto-)reproduce the analysis for the accompanying project. Each section below describes a different component of the analysis and all numbers and figures are generated directly from the underlying data on compilation.</p>
</blockquote>
<div id="data-preprocessing" class="section level1">
<h1>Data Preprocessing</h1>
<p>This part focuses on data preprocessing, it contains the following sections:</p>
<ul>
<li><a href="#data-downloading">Data downloading</a></li>
<li><a href="#data-preprocessing">Data preprocessing</a>
<ul>
<li>Pipeline</li>
<li>Copy number calling</li>
<li>Mutation calling</li>
</ul></li>
<li><a href="#data-cleaning">Data cleaning</a></li>
</ul>
<div id="data-downloading" class="section level2">
<h2>Data downloading</h2>
<p>This section describes where and how we downloaded the data.</p>
<div id="wes-data" class="section level3">
<h3>WES data</h3>
<p>Whole exome sequencing raw data were downloaded from the following 6 dbGap studies:</p>
<ul>
<li><code>phs000178</code>: this is TCGA prostate cancer cohort, the raw data is <code>bam</code> format stored in <a href="https://portal.gdc.cancer.gov/">GDC portal</a>, we access them by dbGap permission</li>
<li><code>phs000447</code>: <code>sra</code> data</li>
<li><code>phs000554</code>: <code>sra</code> data</li>
<li><code>phs000909</code>: <code>sra</code> data</li>
<li><code>phs000915</code>: <code>sra</code> data</li>
<li><code>phs001141</code>: <code>sra</code> data</li>
</ul>
<p>A recent study <span class="citation">(Armenia et al. <a href="#ref-armenia2018long">2018</a>)</span> used 5 dbGap studies:</p>
<ul>
<li><code>phs000178</code></li>
<li><code>phs000447</code></li>
<li><code>phs000909</code></li>
<li><code>phs000915</code></li>
<li><code>phs000945</code></li>
</ul>
<p>Our study contains two new datasets <code>phs000554</code> and <code>phs001141</code> than <span class="citation">Armenia et al. (<a href="#ref-armenia2018long">2018</a>)</span>, however, <code>phs000945</code> was excluded from our study due to its unavailable status in dbGap database.</p>
<p>Therefore, to our knowledge, we included all available raw WES data in our study.</p>
<p>After selecting the raw WES data, we downloaded them by either <code>sratools</code> for dbGap data or <code>gdc-client</code> for TCGA data.</p>
</div>
<div id="phenotype-data" class="section level3">
<h3>Phenotype data</h3>
<p>Phenotype data of <code>phs000447</code>, <code>phs000554</code>, <code>phs000909</code>, <code>phs000915</code> and <code>phs001141</code> were downloaded along with the raw WES data. Phenotype data of TCGA prostate cancer cohort were downloaded from UCSC Xena by UCSCXenaTools <span class="citation">(Wang and Liu <a href="#ref-wang2019ucscxenatools">2019</a>)</span>. The cleaning process is described in <a href="#data-cleaning">‘Data cleaning’</a> section.</p>
</div>
<div id="survival-data" class="section level3">
<h3>Survival data</h3>
<p>Survival data of <code>phs000554</code> were included in phenotype data described above. Survival data of TCGA prostate cancer cohort were downloaded from UCSC Xena by UCSCXenaTools <span class="citation">(Wang and Liu <a href="#ref-wang2019ucscxenatools">2019</a>)</span>. The cleaning process is described in <a href="#data-cleaning">‘Data cleaning’</a> section.</p>
</div>
</div>
<div id="data-preprocessing-1" class="section level2">
<h2>Data preprocessing</h2>
<p>This section describes how we preprocessed the data.</p>
<div id="pipeline" class="section level3">
<h3>Pipeline</h3>
<p>The following diagram shows our upstream workflow. Steps before signature identification are belong to preprocessing. The preprocessing pipeline generates the variation records of samples.</p>
<p><img src="fig/PRAD_Analysis_Pipeline.png" width="600px" style="display: block; margin: auto;" /></p>
<center>
<strong>Genomic variation signature identification pipeline of prostate cancer</strong>
</center>
</div>
<div id="sequence-alignment" class="section level3">
<h3>Sequence alignment</h3>
<p>For cohorts including <code>phs000447</code>, <code>phs000554</code>, <code>phs000909</code>, <code>phs000915</code> and <code>phs001141</code>, the raw data are in <code>sra</code> format. We did sequence alignment to them. To keep in line with TCGA bam data, we used the same reference genome (downloaded from <a href="https://portal.gdc.cancer.gov/">GDC portal</a>) and operations (see <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/">GDC docs</a>).</p>
<blockquote>
<p>Of note, the preprocessing step was done on Linux server provided by High Performance Computing Service of ShanghaiTech University. So the work is presented in PBS scripts. If readers want to reproduce this work, please focus on the key code lines.</p>
</blockquote>
<p>The details are described as below.</p>
<div id="from-sra-to-fastq" class="section level4">
<h4>1. from <code>sra</code> to <code>fastq</code></h4>
<p>The raw WES data downloaded from dbGap database are in <code>sra</code> format, we need to convert them into <code>fastq</code> format firstly.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#PBS -N fastq_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#PBS -l walltime=20:00:00</span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#PBS -l mem=10gb</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="va">workdir=</span>/public/home/liuxs/ncbi/dbGaP-16533</a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="bu">cd</span> <span class="va">$workdir</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="ex">fastq-dump</span> -outdir fastq --split-3 --skip-technical --clip --gzip <span class="va">$workdir</span>/sra/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.sra</a></code></pre></div>
</div>
<div id="removed-the-adapters" class="section level4">
<h4>2. removed the adapters</h4>
<p>This step detected and removed the adapters with <code>trimgalore</code> <span class="citation">(Krueger <a href="#ref-krueger2015trim">2015</a>)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">#PBS -N &lt;sample&gt;_clean</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#PBS -l walltime=12:00:00</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="va">workdir=</span>/public/home/liuxs/ncbi/dbGaP-16533</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="ex">trim_galore</span> -q 20 --phred33 --stringency 3 --length 20 -e 0.1 --paired --gzip <span class="va">$workdir</span>/fastq/<span class="op">&lt;</span>sample<span class="op">&gt;</span>_1.fastq.gz \</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="va">$workdir</span>/fastq/<span class="op">&lt;</span>sample<span class="op">&gt;</span>_2.fastq.gz -o <span class="va">$workdir</span>/dnaseq/trimclean</a></code></pre></div>
</div>
<div id="bwa" class="section level4">
<h4>3. bwa</h4>
<p>This step aligned sequence reads to reference genome with recommended <code>BWA MEM</code> algorithm <span class="citation">(Li <a href="#ref-li2013aligning">2013</a>)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">#PBS -N mem_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#PBS -l walltime=35:00:00</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#PBS -l mem=20gb</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#PBS -q normal_3</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="ex">bwa</span> mem -M -R <span class="st">&quot;@RG\tID:&lt;sample&gt;\t\</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="st">LM:&lt;sample&gt;\t\</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="st">SM:&lt;sample&gt;\t\</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="st">PL:illumina\tPU:&lt;sample&gt;&quot;</span>\</a>
<a class="sourceLine" id="cb3-14" data-line-number="14"> /public/home/liuxs/biodata/reference/genome/gdc_hg38/GRCh38.d1.vd1.fa\</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/trimclean/<span class="op">&lt;</span>sample<span class="op">&gt;</span>_1_val_1.fq.gz\</a>
<a class="sourceLine" id="cb3-16" data-line-number="16"> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/trimclean/<span class="op">&lt;</span>sample<span class="op">&gt;</span>_2_val_2.fq.gz\</a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="op">&gt;&lt;</span>sample<span class="op">&gt;</span>.sam\</a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="kw">|</span><span class="fu">mv</span> <span class="op">&lt;</span>sample<span class="op">&gt;</span>.sam /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bwa</a></code></pre></div>
</div>
<div id="sam-to-bam" class="section level4">
<h4>4. <code>sam</code> to <code>bam</code></h4>
<p>This step converted <code>sam</code> files to <code>bam</code> files with <code>samtools</code> <span class="citation">(Li et al. <a href="#ref-li2009sequence">2009</a>)</span>, which are smaller.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co">#PBS -N bam_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#PBS -l walltime=15:00:00</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#PBS -l mem=20gb</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="ex">samtools</span> view -bS /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bwa/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.sam<span class="op">&gt;</span>\</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bam/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.bam</a></code></pre></div>
</div>
<div id="sort-bam" class="section level4">
<h4>5. sort <code>bam</code></h4>
<p>Next we sorted the <code>bam</code> files with <code>Picard</code> <span class="citation">(“Picard Toolkit” <a href="#ref-Picard2019toolkit">2019</a>)</span>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co">#PBS -N sort_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#PBS -l walltime=15:00:00</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#PBS -l mem=20gb</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="ex">java</span> -jar -Xmx12G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp \</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"> /public/home/liuxs/anaconda3/envs/wes/share/picard-2.20.6-0/picard.jar  SortSam\</a>
<a class="sourceLine" id="cb5-12" data-line-number="12"> I=/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bam/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.bam\</a>
<a class="sourceLine" id="cb5-13" data-line-number="13"> O=/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bamsort/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.sort.bam\</a>
<a class="sourceLine" id="cb5-14" data-line-number="14"> SORT_ORDER=coordinate</a></code></pre></div>
</div>
<div id="mark-duplications" class="section level4">
<h4>6. mark duplications</h4>
<p>This step marked duplications with <code>Picard</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co">#PBS -N rmdump_&lt;sample&gt; </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">#PBS -l walltime=10:00:00</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co">#PBS -l mem=10gb</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="ex">java</span> -jar /public/home/liuxs/anaconda3/envs/wes/share/picard-2.20.6-0/picard.jar MarkDuplicates\</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"> I=/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/bamsort/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.sort.bam\</a>
<a class="sourceLine" id="cb6-12" data-line-number="12"> O=/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/markqump/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.rdup.bam\</a>
<a class="sourceLine" id="cb6-13" data-line-number="13"> VALIDATION_STRINGENCY=LENIENT \</a>
<a class="sourceLine" id="cb6-14" data-line-number="14"> MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=1000 \</a>
<a class="sourceLine" id="cb6-15" data-line-number="15"> M=/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/markqump/matric/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.sort.addhead.rmdup.metric</a></code></pre></div>
<p>We then built index with <code>samtools</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co">#PBS -N index_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#PBS -l walltime=10:00:00</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb7-9" data-line-number="9"></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="ex">samtools</span> index /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/markqump/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.rdup.bam\</a>
<a class="sourceLine" id="cb7-11" data-line-number="11"> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/markqump/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.rdup.bam.bai</a></code></pre></div>
</div>
<div id="bqsr" class="section level4">
<h4>7. BQSR</h4>
<p>A base quality score recalibration (BQSR) step was then performed.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co">#PBS -N bqsr_&lt;sample1&gt;</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#PBS -l walltime=30:00:00</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="bu">cd</span> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/markqump</a>
<a class="sourceLine" id="cb8-10" data-line-number="10"></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="va">sample=</span><span class="op">&lt;</span><span class="ex">sample1</span><span class="op">&gt;</span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="va">dbsnp=</span>/public/home/liuxs/biodata/reference/genome/hg38/ftp.broadinstitute.org/bundle/hg38/dbsnp_146.hg38.vcf.gz</a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="va">dbsnp1000G=</span>/public/home/liuxs/biodata/reference/genome/hg38/ftp.broadinstitute.org/bundle/hg38/1000G_phase1.snps.high_confidence.hg38.vcf.gz </a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="va">dbindel100G=</span>/public/home/liuxs/biodata/reference/genome/hg38/ftp.broadinstitute.org/bundle/hg38/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz</a>
<a class="sourceLine" id="cb8-15" data-line-number="15"></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="kw">if</span><span class="bu"> [</span> <span class="ot">-e</span> <span class="va">$sample</span>.rdup.bam<span class="bu"> ]</span>; <span class="kw">then</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17">   <span class="ex">java</span> -jar -Xmx12G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp  /public/home/liuxs/anaconda3/envs/wes/share/gatk4-4.1.3.0-0/gatk-package-4.1.3.0-local.jar BaseRecalibrator \</a>
<a class="sourceLine" id="cb8-18" data-line-number="18">    -R /public/home/liuxs/biodata/reference/genome/gdc_hg38/GRCh38.d1.vd1.fa \</a>
<a class="sourceLine" id="cb8-19" data-line-number="19">    -I <span class="va">${sample}</span>.rdup.bam \</a>
<a class="sourceLine" id="cb8-20" data-line-number="20">    -O /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/table/<span class="va">${sample}</span>.recal_data.table \</a>
<a class="sourceLine" id="cb8-21" data-line-number="21">    --known-sites <span class="va">$dbsnp</span> --known-sites <span class="va">$dbsnp1000G</span> --known-sites <span class="va">$dbindel100G</span> <span class="kw">&amp;&amp;</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22">   <span class="ex">java</span> -jar -Xmx12G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp   /public/home/liuxs/anaconda3/envs/wes/share/gatk4-4.1.3.0-0/gatk-package-4.1.3.0-local.jar  ApplyBQSR  \</a>
<a class="sourceLine" id="cb8-23" data-line-number="23">    -R /public/home/liuxs/biodata/reference/genome/gdc_hg38/GRCh38.d1.vd1.fa \</a>
<a class="sourceLine" id="cb8-24" data-line-number="24">    -I <span class="va">${sample}</span>.rdup.bam \</a>
<a class="sourceLine" id="cb8-25" data-line-number="25">    --bqsr-recal-file /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/table/<span class="va">${sample}</span>.recal_data.table \</a>
<a class="sourceLine" id="cb8-26" data-line-number="26">    -O /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam/<span class="va">${sample}</span>.sorted.marked.BQSR.bam </a>
<a class="sourceLine" id="cb8-27" data-line-number="27">    <span class="bu">echo</span> gatk-BQSR <span class="kw">`</span><span class="fu">date</span><span class="kw">`</span></a>
<a class="sourceLine" id="cb8-28" data-line-number="28"><span class="kw">fi</span></a></code></pre></div>
<p>The result <code>bam</code> files were then used for copy number calling and mutation calling.</p>
</div>
</div>
<div id="copy-number-calling" class="section level3">
<h3>Copy number calling</h3>
<p>The absolute copy number profile for each sample was detected by two well known softwares <a href="https://github.com/mskcc/facets">FACETS</a> <span class="citation">(Shen and Seshan <a href="#ref-shen_facets:_2016">2016</a>)</span> and <a href="https://sequenzatools.bitbucket.io/#/home">Sequenza</a> <span class="citation">(Favero et al. <a href="#ref-favero2014sequenza">2014</a>)</span>.</p>
<p>The code to generate tumor-normal pair for FACETS is <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/00-generate-samplefile-for-facets-cnv-calling.R">here</a>, the code to generate tumor-normal pair for Sequenza is <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/00-generate-samplefile-for-seqz-cnv-calling.R">here</a>.</p>
<div id="facets" class="section level4">
<h4>FACETS</h4>
<p>We followed the standard pipeline of FACETS described in <a href="https://github.com/mskcc/facets/blob/master/vignettes/FACETS.pdf">vignette</a> to call absolute copy number.</p>
<div id="snp-pileup" class="section level5">
<h5>1. SNP pileup</h5>
<p>This step followed the <a href="https://github.com/mskcc/facets/blob/master/inst/extcode/README.txt" class="uri">https://github.com/mskcc/facets/blob/master/inst/extcode/README.txt</a>.</p>
<p>For dbGap studies:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">#PBS -N &lt;sample&gt;_snp</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="co">#PBS -l walltime=10:00:00</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="bu">cd</span> /public/home/liuxs/anaconda3/envs/R/lib/R/library/facets/extcode</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="bu">source</span> activate R</a>
<a class="sourceLine" id="cb9-10" data-line-number="10"></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co"># &lt;sample1&gt;=tumorbam</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co"># &lt;sample2&gt;=normalbam</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="va">path1=</span><span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/out&quot;</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="va">path2=</span><span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam&quot;</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="va">path3=</span><span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/bqsrbam&quot;</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="va">name=</span><span class="op">&lt;</span><span class="ex">sample1</span><span class="op">&gt;</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="kw">if</span><span class="bu"> [</span> <span class="ot">-e</span> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam/<span class="va">$name</span>.sorted.marked.BQSR.bam<span class="bu"> ]</span>; </a>
<a class="sourceLine" id="cb9-21" data-line-number="21"> <span class="kw">then</span> </a>
<a class="sourceLine" id="cb9-22" data-line-number="22">  <span class="ex">./snp-pileup</span> -g -q15 -Q20 -P100 -r25,0 /public/home/liuxs/biodata/reference/genome/hg38/snpvcf/common_all_20180418.vcf.gz\</a>
<a class="sourceLine" id="cb9-23" data-line-number="23">  <span class="va">${path1}</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.out.gz \</a>
<a class="sourceLine" id="cb9-24" data-line-number="24">  <span class="va">${path2}</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.sorted.marked.BQSR.bam \</a>
<a class="sourceLine" id="cb9-25" data-line-number="25">  <span class="va">${path2}</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>.sorted.marked.BQSR.bam</a>
<a class="sourceLine" id="cb9-26" data-line-number="26"> <span class="kw">else</span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27">  <span class="ex">./snp-pileup</span> -g -q15 -Q20 -P100 -r25,0 /public/home/liuxs/biodata/reference/genome/hg38/snpvcf/common_all_20180418.vcf.gz\</a>
<a class="sourceLine" id="cb9-28" data-line-number="28"> <span class="va">${path1}</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.out.gz \</a>
<a class="sourceLine" id="cb9-29" data-line-number="29"> <span class="va">${path3}</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.sorted.marked.BQSR.bam  \</a>
<a class="sourceLine" id="cb9-30" data-line-number="30"> <span class="va">${path3}</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>.sorted.marked.BQSR.bam </a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="kw">fi</span></a></code></pre></div>
<p>For TCGA:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">#PBS -N &lt;sample&gt;_snp</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">#PBS -l walltime=10:00:00</span></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb10-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7"></a>
<a class="sourceLine" id="cb10-8" data-line-number="8"><span class="bu">cd</span> /public/home/liuxs/anaconda3/envs/R/lib/R/library/facets/extcode</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"><span class="bu">source</span> activate R</a>
<a class="sourceLine" id="cb10-10" data-line-number="10"></a>
<a class="sourceLine" id="cb10-11" data-line-number="11"><span class="co"># &lt;sample1&gt;=tumorbam</span></a>
<a class="sourceLine" id="cb10-12" data-line-number="12"><span class="co"># &lt;sample2&gt;=normalbam</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13"></a>
<a class="sourceLine" id="cb10-14" data-line-number="14"><span class="va">path1=</span><span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/csv&quot;</span></a>
<a class="sourceLine" id="cb10-15" data-line-number="15"><span class="va">path2=</span><span class="st">&quot;/public/home/liuxs/biodata/gdc/links/TCGA_PRAD&quot;</span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16"></a>
<a class="sourceLine" id="cb10-17" data-line-number="17"></a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="ex">./snp-pileup</span> -g -q15 -Q20 -P100 -r25,0 /public/home/liuxs/biodata/reference/genome/hg38/snpvcf/common_all_20180418.vcf.gz <span class="va">${path1}</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.out.gz <span class="va">${path2}</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.bam <span class="va">${path2}</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>.bam</a></code></pre></div>
</div>
<div id="run-facets-package" class="section level5">
<h5>2. run <code>facets</code> package</h5>
<p>This is a template R script used for copy number calling.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">#! /usr/bin/env Rscript</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw">library</span>(<span class="st">&quot;pctGCdata&quot;</span>)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="kw">library</span>(<span class="st">&quot;facets&quot;</span>)</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="kw">set.seed</span>(<span class="dv">1234</span>)</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">rcmat =<span class="st"> </span><span class="kw">readSnpMatrix</span>(<span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/out/&lt;sample&gt;.out.gz&quot;</span>)</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">xx =<span class="st"> </span><span class="kw">preProcSample</span>(rcmat,<span class="dt">gbuild =</span> <span class="st">&quot;hg38&quot;</span>)</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">oo=<span class="kw">procSample</span>(xx,<span class="dt">cval=</span><span class="dv">150</span>)</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">fit=<span class="kw">emcncf</span>(oo)</a>
<a class="sourceLine" id="cb11-10" data-line-number="10"></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co">#plot </span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="kw">pdf</span>(<span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/facetdata_150/&lt;sample&gt;.pdf&quot;</span>)</a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="kw">plotSample</span>(<span class="dt">x=</span>oo,<span class="dt">emfit=</span>fit)</a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="kw">logRlogORspider</span>(oo<span class="op">$</span>out, oo<span class="op">$</span>dipLogR) </a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="cf">while</span> (<span class="op">!</span><span class="kw">is.null</span>(<span class="kw">dev.list</span>()))  <span class="kw">dev.off</span>()</a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co"># </span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="kw">save</span>(fit,<span class="dt">file =</span> <span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/facetdata_150/&lt;sample&gt;.Rdata&quot;</span>)</a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="co"># output purity and ploidy -----</span></a>
<a class="sourceLine" id="cb11-19" data-line-number="19">purity=fit<span class="op">$</span>purity</a>
<a class="sourceLine" id="cb11-20" data-line-number="20">purity=<span class="kw">round</span>(purity,<span class="dv">2</span>)</a>
<a class="sourceLine" id="cb11-21" data-line-number="21">ploidy=fit<span class="op">$</span>ploidy</a>
<a class="sourceLine" id="cb11-22" data-line-number="22">ploidy=<span class="kw">round</span>(ploidy,<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb11-23" data-line-number="23">output &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;&lt;sample&gt;&quot;</span>, purity, ploidy, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="kw">write</span>(output, <span class="st">&quot;/public/home/liuxs/ncbi/dbGaP-16533/copy/facetdata_150/&lt;sample&gt;.txt&quot;</span>, <span class="dt">append =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>This is the PBS script used for calling the R script above.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co">##### submit R script</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">#PBS -N R</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">#PBS -l walltime=50:00:00</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb12-8" data-line-number="8"></a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="bu">source</span> activate R</a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="kw">for</span> <span class="ex">i</span> in /public/home/liuxs/ncbi/dbGaP-16533/copy/Rdata/re150/pbs_150/*pbs<span class="kw">;do</span> <span class="ex">Rscript</span> <span class="va">$i</span><span class="kw">;done</span></a></code></pre></div>
</div>
</div>
<div id="sequenza" class="section level4">
<h4>Sequenza</h4>
<p>We followed the standard pipeline of Sequenza described in <a href="https://cran.r-project.org/web/packages/sequenza/vignettes/sequenza.html">vignette</a> to call absolute copy number. A modified <a href="https://github.com/ShixiangWang/copynumber">copynumber</a> package was used to work with hg38 genome build.</p>
<div id="process-a-fasta-file-to-produce-a-gc-wiggle-track-file" class="section level5">
<h5>1. process a <code>fasta</code> file to produce a GC <code>wiggle</code> track file</h5>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="bu">source</span> activate python3</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"></a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="va">ref_file=</span>/public/home/liuxs/biodata/reference/genome/gdc_hg38/GRCh38.d1.vd1.fa</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="va">gc_file=</span>/public/home/wangshx/wangshx/PRAD_Sig/hg38.gc50Base.wig.gz</a>
<a class="sourceLine" id="cb13-5" data-line-number="5"></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="fu">mkdir</span> -p <span class="va">$(</span><span class="fu">dirname</span> <span class="va">$gc_file)</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co"># sequenza-utils version: 3.0.0</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="ex">sequenza-utils</span> gc_wiggle -w 50 --fasta <span class="va">$ref_file</span> -o <span class="va">$gc_file</span></a></code></pre></div>
</div>
<div id="process-bam-and-wiggle-files-to-produce-a-small-seqz-file" class="section level5">
<h5>2. process <code>bam</code> and <code>wiggle</code> files to produce a small <code>seqz</code> file:</h5>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co">#PBS -N PBS_&lt;sample&gt;_seqz</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#PBS -l walltime=70:00:00</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="bu">source</span> activate python3</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="va">ref_file=</span>/public/home/liuxs/biodata/reference/genome/gdc_hg38/GRCh38.d1.vd1.fa</a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="va">gc_file=</span>/public/home/wangshx/wangshx/PRAD_Sig/hg38.gc50Base.wig.gz</a>
<a class="sourceLine" id="cb14-12" data-line-number="12"></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="va">out_dir=</span>/public/home/wangshx/wangshx/PRAD_Sig</a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="va">seqz_dir=$out_dir</span><span class="st">&quot;/seqz&quot;</span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="va">sseqz_dir=$out_dir</span><span class="st">&quot;/small-seqz&quot;</span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="fu">mkdir</span> -p <span class="va">$seqz_dir</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="fu">mkdir</span> -p <span class="va">$sseqz_dir</span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#tumor=&lt;tumor&gt;</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#normal=&lt;normal&gt;</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="ex">sequenza-utils</span> bam2seqz -n <span class="op">&lt;</span>normal<span class="op">&gt;</span> -t <span class="op">&lt;</span>tumor<span class="op">&gt;</span> \</a>
<a class="sourceLine" id="cb14-24" data-line-number="24">    --fasta <span class="va">$ref_file</span> -gc <span class="va">$gc_file</span> \</a>
<a class="sourceLine" id="cb14-25" data-line-number="25">    -o <span class="va">$seqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.seqz.gz&quot;</span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"></a>
<a class="sourceLine" id="cb14-27" data-line-number="27"><span class="ex">sequenza-utils</span> seqz_binning --seqz <span class="va">$seqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.seqz.gz&quot;</span> \</a>
<a class="sourceLine" id="cb14-28" data-line-number="28">    -w 50 -o <span class="va">$sseqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.small.seqz.gz&quot;</span></a>
<a class="sourceLine" id="cb14-29" data-line-number="29"></a>
<a class="sourceLine" id="cb14-30" data-line-number="30"><span class="co"># Only keep chr1-22,X,Y</span></a>
<a class="sourceLine" id="cb14-31" data-line-number="31"><span class="fu">zcat</span> <span class="va">$sseqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.small.seqz.gz&quot;</span> <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb14-32" data-line-number="32">    <span class="fu">awk</span> <span class="st">&#39;/^chromosome|chr[12]?[0-9XY]\t/{print}&#39;</span> <span class="kw">|</span> <span class="kw">\</span></a>
<a class="sourceLine" id="cb14-33" data-line-number="33">    <span class="fu">gzip</span> <span class="op">&gt;</span> <span class="va">$sseqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.small_filter.seqz.gz&quot;</span></a></code></pre></div>
</div>
<div id="call-copy-number-in-r" class="section level5">
<h5>3. call copy number in R</h5>
<p>This is a template R script used for copy number calling.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co">#! /usr/bin/env Rscript</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"></a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw">library</span>(<span class="st">&quot;sequenza&quot;</span>)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"></a>
<a class="sourceLine" id="cb15-5" data-line-number="5">args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">sample_id &lt;-<span class="st"> </span>args[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">input_file &lt;-<span class="st"> </span>args[<span class="dv">2</span>]</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">out_dir &lt;-<span class="st"> </span>args[<span class="dv">3</span>]</a>
<a class="sourceLine" id="cb15-9" data-line-number="9"></a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co"># https://github.com/ShixiangWang/copynumber</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11">test &lt;-<span class="st"> </span><span class="kw">sequenza.extract</span>(input_file, <span class="dt">assembly =</span> <span class="st">&quot;hg38&quot;</span>)</a>
<a class="sourceLine" id="cb15-12" data-line-number="12"></a>
<a class="sourceLine" id="cb15-13" data-line-number="13">CP &lt;-<span class="st"> </span><span class="kw">sequenza.fit</span>(test, <span class="dt">female =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb15-14" data-line-number="14"></a>
<a class="sourceLine" id="cb15-15" data-line-number="15"><span class="kw">sequenza.results</span>(</a>
<a class="sourceLine" id="cb15-16" data-line-number="16">  <span class="dt">sequenza.extract =</span> test,</a>
<a class="sourceLine" id="cb15-17" data-line-number="17">  <span class="dt">cp.table =</span> CP,</a>
<a class="sourceLine" id="cb15-18" data-line-number="18">  <span class="dt">sample.id =</span> sample_id,</a>
<a class="sourceLine" id="cb15-19" data-line-number="19">  <span class="dt">out.dir =</span> out_dir,</a>
<a class="sourceLine" id="cb15-20" data-line-number="20">  <span class="dt">female =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb15-21" data-line-number="21">)</a></code></pre></div>
<p>This is a template PBS script used for calling the R script above.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co">#PBS -N PBS_&lt;sample&gt;_seqz</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=1</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">#PBS -l walltime=70:00:00</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"></a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="bu">source</span> activate R_36</a>
<a class="sourceLine" id="cb16-9" data-line-number="9"></a>
<a class="sourceLine" id="cb16-10" data-line-number="10"><span class="va">out_dir=</span>/public/home/wangshx/wangshx/PRAD_Sig</a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="va">sseqz_dir=$out_dir</span><span class="st">&quot;/small-seqz&quot;</span></a>
<a class="sourceLine" id="cb16-12" data-line-number="12"><span class="va">res_dir=$out_dir</span><span class="st">&quot;/seqz_wes_result&quot;</span></a>
<a class="sourceLine" id="cb16-13" data-line-number="13"></a>
<a class="sourceLine" id="cb16-14" data-line-number="14"><span class="ex">Rscript</span> <span class="va">$out_dir</span><span class="st">&quot;/3-sequenza.R&quot;</span> <span class="op">&lt;</span>sample<span class="op">&gt;</span> <span class="va">$sseqz_dir</span>/<span class="st">&quot;&lt;sample&gt;.small_filter.seqz.gz&quot;</span> <span class="va">$res_dir</span></a></code></pre></div>
</div>
</div>
</div>
<div id="mutation-calling" class="section level3">
<h3>Mutation calling</h3>
<p>Somatic mutations were detected by following GATK best practice with Mutect2 <span class="citation">(Van der Auwera et al. <a href="#ref-van2013fastq">2013</a>)</span>.</p>
<div id="section" class="section level4">
<h4>1.</h4>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="co">#PBS -N call-&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=7</span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="co">#PBS -l walltime=30:00:00</span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"></a>
<a class="sourceLine" id="cb17-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb17-9" data-line-number="9"><span class="va">gene=</span>/public/home/liuxs/biodata/reference/genome/gdc_hg38</a>
<a class="sourceLine" id="cb17-10" data-line-number="10"><span class="va">normal=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam</a>
<a class="sourceLine" id="cb17-11" data-line-number="11"><span class="va">tumor=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam</a>
<a class="sourceLine" id="cb17-12" data-line-number="12"><span class="va">path=</span>/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/bqsrbam</a>
<a class="sourceLine" id="cb17-13" data-line-number="13"><span class="va">ref=</span>/public/home/liuxs/biodata/reference/genome/hg38/test/ftp.broadinstitute.org/bundle/Mutect2</a>
<a class="sourceLine" id="cb17-14" data-line-number="14"><span class="va">out=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf</a>
<a class="sourceLine" id="cb17-15" data-line-number="15"><span class="va">inter=</span>/public/home/liuxs/ncbi/dbGaP-16533/run/gatk/mutect</a>
<a class="sourceLine" id="cb17-16" data-line-number="16"><span class="va">out2=</span>/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/vcf</a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="va">name=</span><span class="op">&lt;</span><span class="ex">sample1</span><span class="op">&gt;</span></a>
<a class="sourceLine" id="cb17-18" data-line-number="18"><span class="kw">if</span><span class="bu"> [</span> <span class="ot">-e</span> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam/<span class="va">$name</span>.sorted.marked.BQSR.bam<span class="bu"> ]</span>; </a>
<a class="sourceLine" id="cb17-19" data-line-number="19"><span class="kw">then</span> </a>
<a class="sourceLine" id="cb17-20" data-line-number="20"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp&quot;</span>  Mutect2 \</a>
<a class="sourceLine" id="cb17-21" data-line-number="21"> -R <span class="va">$gene</span>/GRCh38.d1.vd1.fa \</a>
<a class="sourceLine" id="cb17-22" data-line-number="22"> -L <span class="va">$inter</span>/Homo_sapiens_assembly38.targets.interval_list \</a>
<a class="sourceLine" id="cb17-23" data-line-number="23"> -I <span class="va">$tumor</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>.sorted.marked.BQSR.bam \</a>
<a class="sourceLine" id="cb17-24" data-line-number="24"> -tumor <span class="op">&lt;</span>sample1<span class="op">&gt;</span> \</a>
<a class="sourceLine" id="cb17-25" data-line-number="25"> -I <span class="va">$normal</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.sorted.marked.BQSR.bam  \</a>
<a class="sourceLine" id="cb17-26" data-line-number="26"> -normal <span class="op">&lt;</span>sample2<span class="op">&gt;</span>\</a>
<a class="sourceLine" id="cb17-27" data-line-number="27"> --germline-resource <span class="va">$ref</span>/af-only-gnomad.hg38.vcf.gz \</a>
<a class="sourceLine" id="cb17-28" data-line-number="28"> --panel-of-normals /public/home/liuxs/biodata/reference/genome/hg38/test/mu_call_ref/ref/ref/test/1000g_pon.hg38.vcf.gz \</a>
<a class="sourceLine" id="cb17-29" data-line-number="29"> --af-of-alleles-not-in-resource 0.0000025 \</a>
<a class="sourceLine" id="cb17-30" data-line-number="30"> --disable-read-filter  MateOnSameContigOrNoMappedMateReadFilter \</a>
<a class="sourceLine" id="cb17-31" data-line-number="31"> --bam-output <span class="va">$out</span>/bam/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.bam \</a>
<a class="sourceLine" id="cb17-32" data-line-number="32"> --f1r2-tar-gz <span class="va">$out</span>/fir2/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.tar.gz \</a>
<a class="sourceLine" id="cb17-33" data-line-number="33"> -O <span class="va">$out</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.vcf</a>
<a class="sourceLine" id="cb17-34" data-line-number="34"><span class="kw">else</span></a>
<a class="sourceLine" id="cb17-35" data-line-number="35"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp&quot;</span>  Mutect2 \</a>
<a class="sourceLine" id="cb17-36" data-line-number="36">   -R <span class="va">$gene</span>/GRCh38.d1.vd1.fa \</a>
<a class="sourceLine" id="cb17-37" data-line-number="37">   -L <span class="va">$inter</span>/Homo_sapiens_assembly38.targets.interval_list \</a>
<a class="sourceLine" id="cb17-38" data-line-number="38">   -I <span class="va">$path</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>.sorted.marked.BQSR.bam \</a>
<a class="sourceLine" id="cb17-39" data-line-number="39">   -tumor <span class="op">&lt;</span>sample1<span class="op">&gt;</span> \</a>
<a class="sourceLine" id="cb17-40" data-line-number="40">   -I <span class="va">$path</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.sorted.marked.BQSR.bam  \</a>
<a class="sourceLine" id="cb17-41" data-line-number="41">   -normal <span class="op">&lt;</span>sample2<span class="op">&gt;</span>\</a>
<a class="sourceLine" id="cb17-42" data-line-number="42">   --germline-resource <span class="va">$ref</span>/af-only-gnomad.hg38.vcf.gz \</a>
<a class="sourceLine" id="cb17-43" data-line-number="43">   --panel-of-normals /public/home/liuxs/biodata/reference/genome/hg38/test/mu_call_ref/ref/ref/test/1000g_pon.hg38.vcf.gz \</a>
<a class="sourceLine" id="cb17-44" data-line-number="44">   --af-of-alleles-not-in-resource 0.0000025 \</a>
<a class="sourceLine" id="cb17-45" data-line-number="45">   --disable-read-filter  MateOnSameContigOrNoMappedMateReadFilter \</a>
<a class="sourceLine" id="cb17-46" data-line-number="46">   --bam-output <span class="va">$out2</span>/bam/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.bam \</a>
<a class="sourceLine" id="cb17-47" data-line-number="47">   --f1r2-tar-gz <span class="va">$out2</span>/fir2/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.tar.gz \</a>
<a class="sourceLine" id="cb17-48" data-line-number="48">   -O <span class="va">$out2</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.vcf</a>
<a class="sourceLine" id="cb17-49" data-line-number="49"><span class="kw">fi</span></a></code></pre></div>
</div>
<div id="section-1" class="section level4">
<h4>2.</h4>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co">#PBS -N getpile_normal_&lt;sample2&gt;</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=7</span></a>
<a class="sourceLine" id="cb18-3" data-line-number="3"><span class="co">#PBS -l walltime=30:00:00</span></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb18-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb18-7" data-line-number="7"></a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="va">bam=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/bqsrbam</a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="va">SNP=</span>/public/home/liuxs/biodata/reference/genome/hg38/test/ftp.broadinstitute.org/bundle/Mutect2</a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="va">out=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/piptable</a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="va">path=</span>/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/bqsrbam</a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="va">out2=</span>/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/vcf/piptable</a>
<a class="sourceLine" id="cb18-14" data-line-number="14"></a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/wangshx/wx/tmp&quot;</span>  GetPileupSummaries \</a>
<a class="sourceLine" id="cb18-16" data-line-number="16">-I <span class="va">$bam</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>.sorted.marked.BQSR.bam \</a>
<a class="sourceLine" id="cb18-17" data-line-number="17">-L /public/home/liuxs/ncbi/dbGaP-16533/run/gatk/mutect/Homo_sapiens_assembly38.targets.interval_list \</a>
<a class="sourceLine" id="cb18-18" data-line-number="18">-V <span class="va">$SNP</span>/af-only-gnomad.hg38.vcf.gz \</a>
<a class="sourceLine" id="cb18-19" data-line-number="19">-O <span class="va">$out</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>-normal.pileups.table</a></code></pre></div>
</div>
<div id="section-2" class="section level4">
<h4>3.</h4>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co">#PBS -N seg_and_cont</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=7</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="co">#PBS -l walltime=30:00:00</span></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb19-9" data-line-number="9"></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="va">input=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/piptable</a>
<a class="sourceLine" id="cb19-11" data-line-number="11"><span class="va">input2=</span>/public/home/liuxs/ncbi/dbGaP-21926/dnaseq/BQSR/vcf/piptable</a>
<a class="sourceLine" id="cb19-12" data-line-number="12"></a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="va">name=</span><span class="op">&lt;</span><span class="ex">sample1</span><span class="op">&gt;</span></a>
<a class="sourceLine" id="cb19-14" data-line-number="14"><span class="kw">if</span><span class="bu"> [</span> <span class="ot">-e</span> /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/piptable/<span class="va">$name</span>-tumor.pileups.table<span class="bu"> ]</span>; </a>
<a class="sourceLine" id="cb19-15" data-line-number="15"><span class="kw">then</span></a>
<a class="sourceLine" id="cb19-16" data-line-number="16"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/wangshx/wx/tmp&quot;</span>  CalculateContamination \</a>
<a class="sourceLine" id="cb19-17" data-line-number="17">-I <span class="va">$input</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>-tumor.pileups.table \</a>
<a class="sourceLine" id="cb19-18" data-line-number="18">-matched <span class="va">$input</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>-normal.pileups.table \</a>
<a class="sourceLine" id="cb19-19" data-line-number="19">-O <span class="va">$input</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-contamination.table \</a>
<a class="sourceLine" id="cb19-20" data-line-number="20">-tumor-segmentation <span class="va">$input</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-segments.table</a>
<a class="sourceLine" id="cb19-21" data-line-number="21"><span class="kw">else</span></a>
<a class="sourceLine" id="cb19-22" data-line-number="22"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/wangshx/wx/tmp&quot;</span>  CalculateContamination \</a>
<a class="sourceLine" id="cb19-23" data-line-number="23"> -I <span class="va">$input2</span>/<span class="op">&lt;</span>sample1<span class="op">&gt;</span>-tumor.pileups.table \</a>
<a class="sourceLine" id="cb19-24" data-line-number="24"> -matched <span class="va">$input2</span>/<span class="op">&lt;</span>sample2<span class="op">&gt;</span>-normal.pileups.table \</a>
<a class="sourceLine" id="cb19-25" data-line-number="25"> -O <span class="va">$input2</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-contamination.table \</a>
<a class="sourceLine" id="cb19-26" data-line-number="26"> -tumor-segmentation <span class="va">$input2</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-segments.table</a>
<a class="sourceLine" id="cb19-27" data-line-number="27"><span class="kw">fi</span></a></code></pre></div>
</div>
<div id="section-3" class="section level4">
<h4>4.</h4>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="co">#PBS -N flr_&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=7</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="co">#PBS -l walltime=20:00:00</span></a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb20-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb20-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7"></a>
<a class="sourceLine" id="cb20-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb20-9" data-line-number="9"><span class="va">out=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/fir2</a>
<a class="sourceLine" id="cb20-10" data-line-number="10"></a>
<a class="sourceLine" id="cb20-11" data-line-number="11"><span class="ex">java</span> -jar /public/home/liuxs/anaconda3/envs/wes/share/gatk4-4.1.3.0-0/gatk-package-4.1.3.0-local.jar  LearnReadOrientationModel \</a>
<a class="sourceLine" id="cb20-12" data-line-number="12">-I /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/fir2/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.tar.gz  \</a>
<a class="sourceLine" id="cb20-13" data-line-number="13">-O <span class="va">$out</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-tumor-artifact-prior.tar.gz</a></code></pre></div>
</div>
<div id="section-4" class="section level4">
<h4>5.</h4>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="co">#PBS -N filter-&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="co">#PBS -l walltime=20:00:00</span></a>
<a class="sourceLine" id="cb21-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb21-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb21-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb21-9" data-line-number="9"><span class="va">gene=</span>/public/home/liuxs/biodata/reference/genome/gdc_hg38</a>
<a class="sourceLine" id="cb21-10" data-line-number="10"><span class="va">out=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf</a>
<a class="sourceLine" id="cb21-11" data-line-number="11"><span class="va">input=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/piptable</a>
<a class="sourceLine" id="cb21-12" data-line-number="12"><span class="va">fir2=</span>/public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/vcf/fir2</a>
<a class="sourceLine" id="cb21-13" data-line-number="13"></a>
<a class="sourceLine" id="cb21-14" data-line-number="14"></a>
<a class="sourceLine" id="cb21-15" data-line-number="15"><span class="ex">gatk</span> --java-options <span class="st">&quot;-Xmx20G -Djava.io.tmpdir=/public/home/liuxs/lhm/tmp&quot;</span> FilterMutectCalls \</a>
<a class="sourceLine" id="cb21-16" data-line-number="16">-R <span class="va">$gene</span>/GRCh38.d1.vd1.fa \</a>
<a class="sourceLine" id="cb21-17" data-line-number="17">-V <span class="va">$out</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.vcf \</a>
<a class="sourceLine" id="cb21-18" data-line-number="18">-O /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/filter/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.filter.vcf \</a>
<a class="sourceLine" id="cb21-19" data-line-number="19">--tumor-segmentation <span class="va">$input</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-segments.table \</a>
<a class="sourceLine" id="cb21-20" data-line-number="20">--contamination-table <span class="va">$input</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-contamination.table \</a>
<a class="sourceLine" id="cb21-21" data-line-number="21">--ob-priors <span class="va">$fir2</span>/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-tumor-artifact-prior.tar.gz \</a>
<a class="sourceLine" id="cb21-22" data-line-number="22">--min-allele-fraction 0.05</a></code></pre></div>
</div>
<div id="section-5" class="section level4">
<h4>6.</h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co">#PBS -N call-&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=7</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">#PBS -l walltime=30:00:00</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="bu">source</span> activate wes</a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="ex">vcftools</span> --vcf /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/filter/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.filter.vcf\</a>
<a class="sourceLine" id="cb22-10" data-line-number="10">  --remove-filtered-all --recode --recode-INFO-all --out /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/PASS/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-SNvs_only</a></code></pre></div>
</div>
<div id="mutation-annotation" class="section level4">
<h4>7. mutation annotation</h4>
<p>This step annotated somatic mutations with VEP <span class="citation">(McLaren et al. <a href="#ref-mclaren2016ensembl">2016</a>)</span>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="co">#PBS -N annotate-&lt;sample&gt;</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">#PBS -l nodes=1:ppn=4</span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="co">#PBS -l walltime=20:00:00</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="co">#PBS -S /bin/bash</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">#PBS -q normal_8</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="co">#PBS -j oe</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="bu">source</span> activate neo</a>
<a class="sourceLine" id="cb23-9" data-line-number="9"></a>
<a class="sourceLine" id="cb23-10" data-line-number="10"><span class="ex">vep</span>  --input_file /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/PASS/<span class="op">&lt;</span>sample<span class="op">&gt;</span>-SNvs_only.recode.vcf \</a>
<a class="sourceLine" id="cb23-11" data-line-number="11">--output_file /public/home/liuxs/ncbi/dbGaP-16533/dnaseq/BQSR/annonate/<span class="op">&lt;</span>sample<span class="op">&gt;</span>.anno.vcf \</a>
<a class="sourceLine" id="cb23-12" data-line-number="12">--symbol --term SO  --format vcf --vcf --cache --assembly GRCh38 --tsl \</a>
<a class="sourceLine" id="cb23-13" data-line-number="13">--hgvs --fasta ~/.vep/homo_sapiens/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz \</a>
<a class="sourceLine" id="cb23-14" data-line-number="14">--offline --cache --dir_cache ~/.vep/ \</a>
<a class="sourceLine" id="cb23-15" data-line-number="15">--transcript_version --cache_version 98 \</a>
<a class="sourceLine" id="cb23-16" data-line-number="16">--plugin Downstream  --plugin Wildtype \</a>
<a class="sourceLine" id="cb23-17" data-line-number="17">--dir_plugins ~/.vep/Plugins </a></code></pre></div>
</div>
</div>
</div>
<div id="data-cleaning" class="section level2">
<h2>Data cleaning</h2>
<p>This section describes how we cleaned the data for downstream analysis.</p>
<div id="phenotype-data-1" class="section level3">
<h3>Phenotype data</h3>
<p>2 info tables of SRA runs downloaded from dbGap are stored in <a href="https://github.com/ShixiangWang/prad_signature/tree/master/dbGap">here</a>. Phenotype data of dbGap studies are stored in <a href="https://github.com/ShixiangWang/prad_signature/tree/master/dbGap/phenotype">here</a>. Phenotype data of some cohorts were also downloaded from original articles and cleaned by hand. We then carefully checked and compared all available data, extracted key info and generated tidy datasets including sample-pair data used for variation calling and phenotype data used for downstream analysis.</p>
<p>The whole procedure is recorded in <a href="https://github.com/ShixiangWang/prad_signature/blob/master/dbGap/01-tidy_mapping.R">01-tidy_mapping.R</a>, <a href="https://github.com/ShixiangWang/prad_signature/blob/master/dbGap/02-tidy_clinical_data.R">02-tidy_clinical_data.R</a> and <a href="https://github.com/ShixiangWang/prad_signature/blob/master/dbGap/03-combine_and_clean_IDs.R">03-combine_and_clean_IDs.R</a>.</p>
</div>
<div id="survival-data-1" class="section level3">
<h3>Survival data</h3>
<p>We merged two available clinical datasets from TCGA and <code>phs000554</code> with <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/01-prepare-survival-data.R">R script</a>.</p>
</div>
<div id="variation-profile" class="section level3">
<h3>Variation profile</h3>
<p><a href="#copy-number-calling">Copy number calling</a> and <a href="#mutation-calling">mutation calling</a> all generated per-sample results. We go further merged them into one file, respectively.</p>
<p>For results from Sequenza, we merged them with R script <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/00-generate-samplefile-for-seqz-cnv-calling.R">00-generate-samplefile-for-seqz-cnv-calling.R</a>.</p>
<p>For results from FACETS, we merged them with R script <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/00-generate-samplefile-for-facets-cnv-calling.R">00-generate-samplefile-for-facets-cnv-calling.R</a>.</p>
<p>For results from Mutect2, we merged VCFs into MAF with tool <a href="https://github.com/mskcc/vcf2maf">vcf2maf</a>. TODO: more detail code.</p>
</div>
</div>
</div>
<div id="signature-identification" class="section level1">
<h1>Signature Identification</h1>
<p>This part has been divided into 4 sections as the following, it describes how to get the genomic variation (or instability) signatures from mutation profile (including copy number profile and single base substitution (SBS) profile).</p>
<ul>
<li><a href="#tally-variation-records">Tally variation records</a></li>
<li><a href="#estimate-signature-number">Estimate signature number</a></li>
<li><a href="#extract-signatures">Extract signatures</a></li>
<li><a href="#show-signature-profile-and-aetiology">Show signature profile and aetiology</a></li>
</ul>
<p>All the processes have been implemented in R package <a href="https://github.com/ShixiangWang/sigminer"><strong>sigminer</strong></a>.</p>
<div id="tally-variation-records" class="section level2">
<h2>Tally variation records</h2>
<div id="methods" class="section level3">
<h3>Methods</h3>
<p>For SBS profile, same as previously reported <span class="citation">(Alexandrov et al. <a href="#ref-alexandrov2013signatures">2013</a>)</span>, for each sample, we firstly classified mutation records into six substitution subtypes: C&gt;A, C&gt;G, C&gt;T, T&gt;A, T&gt;C, and T&gt;G (all substitutions are referred to by the pyrimidine of the mutated Watson–Crick base pair). Further, each of the substitutions was examined by incorporating information on the bases immediately 5’ and 3’ to each mutated base generating 96 possible mutation types (6 types of substitution ∗ 4 types of 5’ base ∗ 4 types of 3’ base). Each of 96 mutation types is called <strong>component</strong> here.</p>
<p>For copy number profile, we firstly computed the genome-wide distributions of 8 fundamental copy number features for each sample:</p>
<ul>
<li>the breakpoint count per 10 Mb (named <strong>BP10MB</strong>)</li>
<li>the breakpoint count per chromosome arm (named <strong>BPArm</strong>)</li>
<li>the copy number of the segments (named <strong>CN</strong>)</li>
<li>the difference in copy number between adjacent segments (named <strong>CNCP</strong>)</li>
<li>the lengths of oscillating copy number segment chains (named <strong>OsCN</strong>)</li>
<li>the log10 based size of segments (named <strong>SS</strong>)</li>
<li>the minimal number of chromosome with 50% copy number variation (named <strong>NC50</strong>)</li>
<li>the burden of chromosome (named <strong>BoChr</strong>)</li>
</ul>
<p>These features were selected as hallmarks of previously reported genomic aberations like chromothripsis or to denote the distribution pattern of copy number events. The former 6 features have been used in <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy">2018</a>)</span> to uncover the mutational processes in ovarian carcinoma.</p>
<p>Next, unlike <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy">2018</a>)</span> applied mixture modeling to separate the first 6 copy number features distributions into mixtures of Poisson or Gaussian distributions, we directly classified 8 copy number features distributions into different components according to the comprehensive consideration of value range, abundance and biological significance. Most of the result are discrete values, and the others are range values.</p>
<p>The setting of 8 features with 80 components are shown as below.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(sigminer<span class="op">::</span>CN.features)</a></code></pre></div>
<div id="htmlwidget-621273037699d23904ce" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-621273037699d23904ce">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","CN","CN","CN","CN","CN","CN","CN","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","SS","SS","SS","SS","SS","SS","SS","SS","NC50","NC50","NC50","NC50","NC50","NC50","NC50","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr"],["BP10MB[0]","BP10MB[1]","BP10MB[2]","BP10MB[3]","BP10MB[4]","BP10MB[5]","BP10MB[&gt;5]","BPArm[0]","BPArm[1]","BPArm[2]","BPArm[3]","BPArm[4]","BPArm[5]","BPArm[6]","BPArm[7]","BPArm[8]","BPArm[9]","BPArm[10]","BPArm[&gt;10 &amp; &lt;=20]","BPArm[&gt;20 &amp; &lt;=30]","BPArm[&gt;30]","CN[0]","CN[1]","CN[2]","CN[3]","CN[4]","CN[&gt;4 &amp; &lt;=8]","CN[&gt;8]","CNCP[0]","CNCP[1]","CNCP[2]","CNCP[3]","CNCP[4]","CNCP[&gt;4 &amp; &lt;=8]","CNCP[&gt;8]","OsCN[0]","OsCN[1]","OsCN[2]","OsCN[3]","OsCN[4]","OsCN[&gt;4 &amp; &lt;=10]","OsCN[&gt;10]","SS[&lt;=2]","SS[&gt;2 &amp; &lt;=3]","SS[&gt;3 &amp; &lt;=4]","SS[&gt;4 &amp; &lt;=5]","SS[&gt;5 &amp; &lt;=6]","SS[&gt;6 &amp; &lt;=7]","SS[&gt;7 &amp; &lt;=8]","SS[&gt;8]","NC50[&lt;=2]","NC50[3]","NC50[4]","NC50[5]","NC50[6]","NC50[7]","NC50[&gt;7]","BoChr[1]","BoChr[2]","BoChr[3]","BoChr[4]","BoChr[5]","BoChr[6]","BoChr[7]","BoChr[8]","BoChr[9]","BoChr[10]","BoChr[11]","BoChr[12]","BoChr[13]","BoChr[14]","BoChr[15]","BoChr[16]","BoChr[17]","BoChr[18]","BoChr[19]","BoChr[20]","BoChr[21]","BoChr[22]","BoChr[23]"],["point","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","range","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","range","range","range","range","range","range","range","range","range","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point"],[0,1,2,3,4,5,5,0,1,2,3,4,5,6,7,8,9,10,10,20,30,0,1,2,3,4,4,8,0,1,2,3,4,4,8,0,1,2,3,4,4,10,null,2,3,4,5,6,7,8,null,3,4,5,6,7,7,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23],[0,1,2,3,4,5,null,0,1,2,3,4,5,6,7,8,9,10,20,30,null,0,1,2,3,4,8,null,0,1,2,3,4,8,null,0,1,2,3,4,10,null,2,3,4,5,6,7,8,null,2,3,4,5,6,7,null,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>feature<\/th>\n      <th>component<\/th>\n      <th>label<\/th>\n      <th>min<\/th>\n      <th>max<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<blockquote>
<p>When component is a discrete value (label is ‘point’), min=max.</p>
<p>When component is a range value (label is ‘range’), the range is left open and right closed.</p>
<p>Of note, the blank in <code>min</code> column is <code>-Inf</code> and the blank in <code>max</code> column is <code>Inf</code>. <code>[23]</code> in <code>BoChr</code> represents chromosome X (chromosome Y is excluded).</p>
</blockquote>
<p>Compare to the method from <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy">2018</a>)</span>, our method has several advantages:</p>
<ol style="list-style-type: decimal">
<li>the computation is more efficient.</li>
<li>the meaning of component is easier to read and understand.</li>
<li>the new features <strong>NC50</strong> and <strong>BoChr</strong> can be used to determine the distribution pattern of copy number events (global or local). and the contribution of each chromosome.</li>
<li>most importantly, the classification is fixed, so it is much easier to compare the signatures within/across tumor types, the results across different studies and the result signatures to reference signatures. In a word, our method constructs the standard to extract, study and compare copy number signatures.</li>
<li>our method is more extensible.</li>
</ol>
<p>I talk more about No.1-4 in an individual section (<a href="#method-comparison-between-macintyre-et-al-and-our-study">“Method comparison between Macintyre et al and our study”</a>), the No.5 will be described in <a href="">sigminer vignette</a>(<strong>TBD</strong>).</p>
<p>We will generate sample-by-component matrix using the methods above and treate the result as the input of nonnegative matrix decomposition (NMF) algorithm for extracting signatures.</p>
</div>
<div id="tally-records" class="section level3">
<h3>Tally records</h3>
<div id="copy-number" class="section level4">
<h4>Copy number</h4>
<p>Read absolute copy number profile as a <code>CopyNumber</code> object.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="co"># Load packages -----------------------------------------------------------</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"></a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(sigminer))</a>
<a class="sourceLine" id="cb25-4" data-line-number="4"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</a>
<a class="sourceLine" id="cb25-5" data-line-number="5"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(NMF))</a></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># Set this per R session</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">options</span>(<span class="dt">sigminer.sex =</span> <span class="st">&quot;male&quot;</span>, <span class="dt">sigminer.copynumber.max =</span> 20L)</a>
<a class="sourceLine" id="cb26-3" data-line-number="3"></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="co"># Generate CopyNumber object ----------------------------------------------</span></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"></a>
<a class="sourceLine" id="cb26-6" data-line-number="6">CNV.seqz &lt;-<span class="st"> </span><span class="kw">read_copynumber</span>(<span class="st">&quot;../data/CNV_from_sequenza.tsv&quot;</span>,</a>
<a class="sourceLine" id="cb26-7" data-line-number="7">  <span class="dt">genome_build =</span> <span class="st">&quot;hg38&quot;</span>,</a>
<a class="sourceLine" id="cb26-8" data-line-number="8">  <span class="dt">complement =</span> <span class="ot">FALSE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb26-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb26-10" data-line-number="10"></a>
<a class="sourceLine" id="cb26-11" data-line-number="11"><span class="co"># remove WCMC160-SRR3146971 with only one CNV</span></a>
<a class="sourceLine" id="cb26-12" data-line-number="12">CNV.seqz &lt;-<span class="st"> </span><span class="kw">subset</span>(CNV.seqz, <span class="dt">subset =</span> <span class="op">!</span>sample <span class="op">%in%</span><span class="st"> &quot;WCMC160-SRR3146971&quot;</span>)</a>
<a class="sourceLine" id="cb26-13" data-line-number="13"><span class="kw">save</span>(CNV.seqz, <span class="dt">file =</span> <span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</a></code></pre></div>
<p>Tally the alteration records.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">ncores &lt;-<span class="st"> </span><span class="dv">20</span> <span class="co"># Set a proper number of thread</span></a>
<a class="sourceLine" id="cb27-2" data-line-number="2"></a>
<a class="sourceLine" id="cb27-3" data-line-number="3"><span class="co"># Use classfication method devised by me (&quot;W&quot;)</span></a>
<a class="sourceLine" id="cb27-4" data-line-number="4">CNV.seqz.tally.W &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(CNV.seqz, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">cores =</span> ncores, <span class="dt">feature_setting =</span> CN.features)</a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="kw">save</span>(CNV.seqz.tally.W, <span class="dt">file =</span> <span class="st">&quot;../output/CNV.seqz.tally.W.RData&quot;</span>)</a></code></pre></div>
</div>
<div id="sbs" class="section level4">
<h4>SBS</h4>
<p>Read SBS profile as a <code>MAF</code> object.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="co"># Reading data ------------------------------------------------------------</span></a>
<a class="sourceLine" id="cb28-2" data-line-number="2">Maf &lt;-<span class="st"> </span>data.table<span class="op">::</span><span class="kw">fread</span>(<span class="st">&quot;/public/data/maf/all.maf&quot;</span>)</a>
<a class="sourceLine" id="cb28-3" data-line-number="3"><span class="co"># Remove all NA columns</span></a>
<a class="sourceLine" id="cb28-4" data-line-number="4">Maf &lt;-<span class="st"> </span>Maf[, <span class="kw">which</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(Maf, <span class="cf">function</span>(x) <span class="op">!</span><span class="kw">all</span>(<span class="kw">is.na</span>(x))))), with =<span class="st"> </span>F]</a>
<a class="sourceLine" id="cb28-5" data-line-number="5">Maf &lt;-<span class="st"> </span><span class="kw">read_maf</span>(Maf)</a>
<a class="sourceLine" id="cb28-6" data-line-number="6"><span class="kw">save</span>(Maf, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a></code></pre></div>
<p>The result <code>MAF</code> object can be used in any analysis provided by R package <strong>maftools</strong> <span class="citation">(Mayakonda and Koeffler <a href="#ref-mayakonda2016maftools">2016</a>)</span>.</p>
<p>Tally the SBS records.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">Maf.tally &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(Maf,</a>
<a class="sourceLine" id="cb29-2" data-line-number="2">  <span class="dt">cores =</span> ncores, <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg38&quot;</span>,</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">  <span class="dt">useSyn =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4">)</a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="kw">save</span>(Maf.tally, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.tally.RData&quot;</span>)</a></code></pre></div>
</div>
</div>
</div>
<div id="estimate-signature-number" class="section level2">
<h2>Estimate signature number</h2>
<p>Here, r package <strong>NMF</strong> <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible">2010</a>)</span> is used for running NMF algorithm Factorization rank <code>r</code> in NMF defines the number of signatures used to approximate the target sample-by-component matrix. A common way for deciding on <code>r</code> is to try different values, compute some quality measure of the results, and choose the best value according to this quality criteria.</p>
<p>As suggested, performing 30-50 runs is considered sufficient to get a robust estimation of the <code>r</code> value. Here we perform 50 runs.</p>
<div id="copy-number-1" class="section level3">
<h3>Copy number</h3>
<p>Estimate the number of copy number signature from 2 to 12.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1">EST.seqz.W.all &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(CNV.seqz.tally.W<span class="op">$</span>nmf_matrix,</a>
<a class="sourceLine" id="cb30-2" data-line-number="2">  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">12</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores, <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb30-3" data-line-number="3">  <span class="dt">save_plots =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb30-4" data-line-number="4">  <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb30-5" data-line-number="5">)</a>
<a class="sourceLine" id="cb30-6" data-line-number="6"><span class="kw">save</span>(EST.seqz.W.all, <span class="dt">file =</span> <span class="st">&quot;../output/EST.seqz.W.all.RData&quot;</span>)</a></code></pre></div>
</div>
<div id="sbs-1" class="section level3">
<h3>SBS</h3>
<p>Estimate the number of SBS signature from 2 to 10.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="co"># Remove the effect of hyper mutated samples (not removing hyper-mutated samples)</span></a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="co"># The idead is adopted from SignatureAnalyzer package</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3">nmf_matrix &lt;-<span class="st"> </span><span class="kw">handle_hyper_mutation</span>(Maf.tally<span class="op">$</span>nmf_matrix)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4"></a>
<a class="sourceLine" id="cb31-5" data-line-number="5">EST.Maf &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(nmf_matrix,</a>
<a class="sourceLine" id="cb31-6" data-line-number="6">  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores, <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb31-7" data-line-number="7">  <span class="dt">save_plots =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb31-8" data-line-number="8">  <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb31-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb31-10" data-line-number="10"></a>
<a class="sourceLine" id="cb31-11" data-line-number="11"><span class="kw">save</span>(EST.Maf, <span class="dt">file =</span> <span class="st">&quot;../output/EST.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb31-12" data-line-number="12"><span class="kw">save</span>(nmf_matrix, <span class="dt">file =</span> <span class="st">&quot;../output/Maf_matrix.RData&quot;</span>)</a></code></pre></div>
</div>
<div id="result" class="section level3">
<h3>Result</h3>
<p>Let’s load the estimated results.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/EST.seqz.W.all.RData&quot;</span>)</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/EST.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a></code></pre></div>
<p>The most common approach is to use the cophenetic correlation coefficient. I show the measure vs. signature number as below.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">show_sig_number_survey</span>(EST.seqz.W.all, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" width="2400" /></p>
<p>The best possible signature number value is the one at which the cophenetic correlation value on the y-axis drops significantly <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible">2010</a>)</span>. Here we can take that the value <code>5</code> is the optimal signature number. For <code>3</code>, it is small to produce the result with biological significance due to the hugo contribution of CNV in prostate cancer; for <code>8</code>, it is a little bigger, so it may be not easy to understand the result considering currently we know few about the mechanism and consequence of copy number events.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">show_sig_number_survey</span>(EST.seqz.W.all, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw">add_h_arrow</span>(p, <span class="dt">x =</span> <span class="fl">5.1</span>, <span class="dt">y =</span> <span class="fl">0.99</span>, <span class="dt">seg_len =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-19-1.png" width="2400" /></p>
<p>Next we take a look at SBS.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">show_sig_number_survey</span>(EST.Maf, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb35-2" data-line-number="2"><span class="kw">add_h_arrow</span>(p, <span class="dt">x =</span> <span class="fl">3.1</span>, <span class="dt">y =</span> <span class="fl">0.982</span>, <span class="dt">seg_len =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="2400" /></p>
<p>For SBS, we select <code>3</code>. We have the knowledge that <code>3</code> mutational signatures from TCGA prostate cancers are recorded in COSMIC database (COSMIC 1, 5, 6), and the cophenetic has a sharp decrease after <code>3</code> indicating that substantially less stability is achieved using more than <code>3</code> clusters.</p>
</div>
</div>
<div id="extract-signatures" class="section level2">
<h2>Extract signatures</h2>
<p>Now that we have determined the signature number, we then extract the signatures with 50 runs.</p>
<p>Copy number:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="co"># Extract signatures ------------------------------------------------------</span></a>
<a class="sourceLine" id="cb36-2" data-line-number="2">Sig.CNV.seqz.W &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(CNV.seqz.tally.W<span class="op">$</span>nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">5</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3"><span class="kw">save</span>(Sig.CNV.seqz.W, <span class="dt">file =</span> <span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a></code></pre></div>
<p>SBS:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co"># Extract signatures ------------------------------------------------------</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2">Sig.SNV &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">3</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores)</a>
<a class="sourceLine" id="cb37-3" data-line-number="3"><span class="kw">save</span>(Sig.SNV, <span class="dt">file =</span> <span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a></code></pre></div>
</div>
<div id="show-signature-profile-and-aetiology" class="section level2">
<h2>Show signature profile and aetiology</h2>
<p>Now that we have the extracted signatures, we can show the signature profile.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb38-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a></code></pre></div>
<div id="sbs-signature-profile" class="section level3">
<h3>SBS signature profile</h3>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># I provide two style &#39;default&#39; and &#39;cosmic&#39; in sigminer</span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="kw">show_sig_profile</span>(Sig.SNV, <span class="dt">mode =</span> <span class="st">&quot;SBS&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">x_label_angle =</span> <span class="dv">90</span>, <span class="dt">x_label_vjust =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="3000" /></p>
<blockquote>
<p>SBS signatures are displayed based on the observed component frequency of the human genome, i.e., representing the relative proportions of mutations generated by each signature based on the actual trinucleotide frequencies of the reference human genome.</p>
</blockquote>
<p>The aetiology of SBS signatures is more clear than the aetiology of copy number signatures. We can find aetiologies of the 3 signatures above by computing their cosine similarity to COSMIC reference signatures.</p>
<p>We compare the 3 signature to COSMIC signature database v2 and v3.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">sim_v2 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(Sig.SNV, <span class="dt">sig_db =</span> <span class="st">&quot;legacy&quot;</span>)</a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="co">#&gt; -Comparing against COSMIC signatures</span></a>
<a class="sourceLine" id="cb40-3" data-line-number="3"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb40-4" data-line-number="4"><span class="co">#&gt; --Found Sig1 most similar to COSMIC_3</span></a>
<a class="sourceLine" id="cb40-5" data-line-number="5"><span class="co">#&gt;    Aetiology: defects in DNA-DSB repair by HR [similarity: 0.844]</span></a>
<a class="sourceLine" id="cb40-6" data-line-number="6"><span class="co">#&gt; --Found Sig2 most similar to COSMIC_15</span></a>
<a class="sourceLine" id="cb40-7" data-line-number="7"><span class="co">#&gt;    Aetiology: defective DNA mismatch repair [similarity: 0.943]</span></a>
<a class="sourceLine" id="cb40-8" data-line-number="8"><span class="co">#&gt; --Found Sig3 most similar to COSMIC_1</span></a>
<a class="sourceLine" id="cb40-9" data-line-number="9"><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.935]</span></a>
<a class="sourceLine" id="cb40-10" data-line-number="10"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb40-11" data-line-number="11"><span class="co">#&gt; Return result invisiblely.</span></a></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">sim_v3 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(Sig.SNV, <span class="dt">sig_db =</span> <span class="st">&quot;SBS&quot;</span>)</a>
<a class="sourceLine" id="cb41-2" data-line-number="2"><span class="co">#&gt; -Comparing against COSMIC signatures</span></a>
<a class="sourceLine" id="cb41-3" data-line-number="3"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb41-4" data-line-number="4"><span class="co">#&gt; --Found Sig1 most similar to SBS40</span></a>
<a class="sourceLine" id="cb41-5" data-line-number="5"><span class="co">#&gt;    Aetiology: Unknown [similarity: 0.883]</span></a>
<a class="sourceLine" id="cb41-6" data-line-number="6"><span class="co">#&gt; --Found Sig2 most similar to SBS15</span></a>
<a class="sourceLine" id="cb41-7" data-line-number="7"><span class="co">#&gt;    Aetiology: Defective DNA mismatch repair [similarity: 0.967]</span></a>
<a class="sourceLine" id="cb41-8" data-line-number="8"><span class="co">#&gt; --Found Sig3 most similar to SBS1</span></a>
<a class="sourceLine" id="cb41-9" data-line-number="9"><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.966]</span></a>
<a class="sourceLine" id="cb41-10" data-line-number="10"><span class="co">#&gt; ------------------------------------</span></a>
<a class="sourceLine" id="cb41-11" data-line-number="11"><span class="co">#&gt; Return result invisiblely.</span></a></code></pre></div>
<p>From the results above, basically we can confirm that the aetiology of <code>Sig2</code> is <strong>dMMR</strong> (defects in mismatch repair) and the aetiology of <code>Sig3</code> is <strong>aging</strong>. The aetiology of <code>Sig1</code> may be <strong>HRD</strong> (Defective homologous recombination-based DNA damage repair) or an unknown reason.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">show_sig_profile</span>(Sig.SNV, <span class="dt">mode =</span> <span class="st">&quot;SBS&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">x_label_angle =</span> <span class="dv">90</span>, <span class="dt">x_label_vjust =</span> <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="kw">add_labels</span>(p,</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">  <span class="dt">x =</span> <span class="fl">0.92</span>, <span class="dt">y =</span> <span class="fl">0.3</span>, <span class="dt">y_end =</span> <span class="fl">0.85</span>, <span class="dt">n_label =</span> <span class="dv">3</span>,</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">  <span class="dt">labels =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;HRD or unknown&quot;</span>, <span class="st">&quot;dMMR&quot;</span>, <span class="st">&quot;Aging&quot;</span>)), <span class="dt">hjust =</span> <span class="dv">1</span></a>
<a class="sourceLine" id="cb42-5" data-line-number="5">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-27-1.png" width="3000" /></p>
<p>The exposure of signatures in each sample is plotted as the below.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">show_sig_exposure</span>(Sig.SNV, <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-28-1.png" width="2400" /></p>
<p>Set a cutoff 2000 to remove some outliers.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"><span class="kw">show_sig_exposure</span>(Sig.SNV,</a>
<a class="sourceLine" id="cb44-2" data-line-number="2">  <span class="dt">cutoff =</span> <span class="dv">2000</span>,</a>
<a class="sourceLine" id="cb44-3" data-line-number="3">  <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span></a>
<a class="sourceLine" id="cb44-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="2400" /></p>
<blockquote>
<p>The <code>Est_Counts</code> panel shows the estimated SBS count.</p>
</blockquote>
<p>We can clearly see that</p>
<ul>
<li>most of samples have very few SBS signature exposures.</li>
<li><code>Sig1</code> and <code>Sig3</code> are the major resources.</li>
</ul>
</div>
<div id="copy-number-signature-profile" class="section level3">
<h3>Copy number signature profile</h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.W, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">normalize =</span> <span class="st">&quot;feature&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-30-1.png" width="3600" /></p>
<blockquote>
<p>Similar to SBS signatures, copy number signatures are displayed based on the observed component frequency of the human genome. Of note, considering the count process of each feature is relatively independent, the profile is row normalized by each feature, unlike <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy">2018</a>)</span> did column normalization (this method is easy to mislead readers), so the bar height can be compared within/between features.</p>
</blockquote>
<p>There is no reference database for copy number signatures, we cannot get aetiologies of the 5 signatures by similarity computation. By reading the description of previous reports <span class="citation">(Macintyre et al. <a href="#ref-macintyre2018copy">2018</a>; Yi and Ju <a href="#ref-yi2018patterns">2018</a>)</span> and studying the signature profile carefully, we determine or limite aetiologies of the 5 signatures with also the downstream analyses supported.</p>
<ul>
<li>Sig1 - Focal amplification due to failure of cell cyle control
<ul>
<li>large copy number changes resulting high copy number states</li>
</ul></li>
<li>Sig2 - HRD;MMBIR;Tandem duplication through CDK12 inactivation
<ul>
<li>match the pattern of MMBIR (microhomology mediated break-induced replication): many long <strong>OsCN</strong> with copy gains</li>
<li>related to CDK12 mutation</li>
<li>high number of breakpoints</li>
<li>small segment size</li>
<li>related to COSMIC 5 (v2) (from <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy">2018</a>)</span>)</li>
</ul></li>
<li>Sig3 - Whole-genome duplication (due to failure of cell cycle control)
<ul>
<li>high segment copy number</li>
<li>few OsCN</li>
<li>global CNV distribution</li>
</ul></li>
<li>Sig4 - Chromothripsis
<ul>
<li>1 copy number deletions resulting considerable small OsCN</li>
<li>large segment size</li>
</ul></li>
<li>Sig5 - BFB and copy number neutral like events including Chromoplexy, translocation, and etc. (these may be early driver events)
<ul>
<li>few CNV with focal distribution</li>
<li>positively associated with gene fusion</li>
</ul></li>
</ul>
<p>The exposure of signatures in each sample is plotted as the below.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1"><span class="kw">show_sig_exposure</span>(Sig.CNV.seqz.W, <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="2400" /></p>
<blockquote>
<p>The <code>Est_Counts</code> panel shows the estimated copy number segment count.</p>
</blockquote>
</div>
<div id="show-similarity-matrix" class="section level3">
<h3>Show similarity matrix</h3>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim_v2<span class="op">$</span>similarity,</a>
<a class="sourceLine" id="cb47-2" data-line-number="2">  <span class="dt">cluster_rows =</span> F</a>
<a class="sourceLine" id="cb47-3" data-line-number="3">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="3300" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim_v3<span class="op">$</span>similarity,</a>
<a class="sourceLine" id="cb48-2" data-line-number="2">  <span class="dt">cluster_rows =</span> F</a>
<a class="sourceLine" id="cb48-3" data-line-number="3">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-33-1.png" width="4500" /></p>
</div>
<div id="check-the-signature-exposure" class="section level3">
<h3>Check the signature exposure</h3>
<p>Signature exposure is the estimation of mutation calling records. Here we check this by correlation analysis.</p>
<div id="mutation" class="section level4">
<h4>Mutation</h4>
<p>Observe total SBS mutations from data.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb49-2" data-line-number="2">Maf_dt &lt;-<span class="st"> </span><span class="kw">rbind</span>(Maf<span class="op">@</span>data, Maf<span class="op">@</span>maf.silent)</a>
<a class="sourceLine" id="cb49-3" data-line-number="3">Mut_dt &lt;-<span class="st"> </span>Maf_dt[Variant_Type <span class="op">==</span><span class="st"> &quot;SNP&quot;</span>, .(<span class="dt">Total =</span> .N), by =<span class="st"> </span>Tumor_Sample_Barcode]</a></code></pre></div>
<p>Estimate SBS mutations from signature exposure.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">Mut_expo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.SNV)[, .(sample, <span class="dt">Est =</span> Sig1 <span class="op">+</span><span class="st"> </span>Sig2 <span class="op">+</span><span class="st"> </span>Sig3)]</a></code></pre></div>
<p>Merge the two <code>data.table</code> above.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">Mut_merged_dt &lt;-<span class="st"> </span><span class="kw">merge</span>(Mut_dt, Mut_expo, <span class="dt">by.x =</span> <span class="st">&quot;Tumor_Sample_Barcode&quot;</span>, <span class="dt">by.y =</span> <span class="st">&quot;sample&quot;</span>)</a>
<a class="sourceLine" id="cb51-2" data-line-number="2"><span class="co"># log10 the data</span></a>
<a class="sourceLine" id="cb51-3" data-line-number="3">Mut_merged_dt[, <span class="st">`</span><span class="dt">:=</span><span class="st">`</span>(<span class="dt">Total =</span> <span class="kw">log10</span>(Total), <span class="dt">Est =</span> <span class="kw">log10</span>(Est))]</a>
<a class="sourceLine" id="cb51-4" data-line-number="4"></a>
<a class="sourceLine" id="cb51-5" data-line-number="5"><span class="kw">head</span>(Mut_merged_dt)</a>
<a class="sourceLine" id="cb51-6" data-line-number="6"><span class="co">#&gt;    Tumor_Sample_Barcode     Total       Est</span></a>
<a class="sourceLine" id="cb51-7" data-line-number="7"><span class="co">#&gt; 1:           SRR2404024 1.6989700 1.6989700</span></a>
<a class="sourceLine" id="cb51-8" data-line-number="8"><span class="co">#&gt; 2:           SRR2404067 0.9542425 0.9542425</span></a>
<a class="sourceLine" id="cb51-9" data-line-number="9"><span class="co">#&gt; 3:           SRR2404095 1.9344985 1.9344984</span></a>
<a class="sourceLine" id="cb51-10" data-line-number="10"><span class="co">#&gt; 4:           SRR2404185 1.7708520 1.7708520</span></a>
<a class="sourceLine" id="cb51-11" data-line-number="11"><span class="co">#&gt; 5:           SRR2404230 1.8325089 1.8325089</span></a>
<a class="sourceLine" id="cb51-12" data-line-number="12"><span class="co">#&gt; 6:           SRR2404278 1.9777236 1.9777236</span></a></code></pre></div>
<p>Now we can see their correlation.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1">ggpubr<span class="op">::</span><span class="kw">ggscatter</span>(Mut_merged_dt,</a>
<a class="sourceLine" id="cb52-2" data-line-number="2">  <span class="dt">x =</span> <span class="st">&quot;Total&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Est&quot;</span>, <span class="dt">add =</span> <span class="st">&quot;reg.line&quot;</span>,</a>
<a class="sourceLine" id="cb52-3" data-line-number="3">  <span class="dt">xlab =</span> <span class="st">&quot;Log10 based total SBS counts (observed)&quot;</span>,</a>
<a class="sourceLine" id="cb52-4" data-line-number="4">  <span class="dt">ylab =</span> <span class="st">&quot;Log10 based total SBS counts (estimated)&quot;</span></a>
<a class="sourceLine" id="cb52-5" data-line-number="5">) <span class="op">+</span></a>
<a class="sourceLine" id="cb52-6" data-line-number="6"><span class="st">  </span>ggpubr<span class="op">::</span><span class="kw">stat_cor</span>(<span class="dt">label.x.npc =</span> <span class="fl">0.5</span>, <span class="dt">label.y.npc =</span> <span class="fl">0.8</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-37-1.png" width="2400" /></p>
<p>We can know that we did a perfect NMF for SBS data.</p>
</div>
<div id="copy-number-2" class="section level4">
<h4>Copy number</h4>
<p>Observe total segment counts from copy number data.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</a>
<a class="sourceLine" id="cb53-2" data-line-number="2">CN_dt &lt;-<span class="st"> </span>CNV.seqz<span class="op">@</span>data[, .(<span class="dt">Total =</span> .N), by =<span class="st"> </span>sample]</a></code></pre></div>
<p>Estimate segment counts from signature exposure.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1">CN_expo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.CNV.seqz.W)[, .(sample, <span class="dt">Est =</span> Sig1 <span class="op">+</span><span class="st"> </span>Sig2 <span class="op">+</span><span class="st"> </span>Sig3 <span class="op">+</span><span class="st"> </span>Sig4 <span class="op">+</span><span class="st"> </span>Sig5)]</a></code></pre></div>
<p>Merge the two <code>data.table</code> above.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1">CN_merged_dt &lt;-<span class="st"> </span><span class="kw">merge</span>(CN_dt, CN_expo, <span class="dt">by =</span> <span class="st">&quot;sample&quot;</span>)</a>
<a class="sourceLine" id="cb55-2" data-line-number="2"></a>
<a class="sourceLine" id="cb55-3" data-line-number="3"><span class="kw">head</span>(CN_merged_dt)</a>
<a class="sourceLine" id="cb55-4" data-line-number="4"><span class="co">#&gt;                 sample Total      Est</span></a>
<a class="sourceLine" id="cb55-5" data-line-number="5"><span class="co">#&gt; 1: 00-000450-SRR474658    58 61.71629</span></a>
<a class="sourceLine" id="cb55-6" data-line-number="6"><span class="co">#&gt; 2:   00-1165-SRR461844    51 51.71139</span></a>
<a class="sourceLine" id="cb55-7" data-line-number="7"><span class="co">#&gt; 3:    00-160-SRR471373    65 68.69707</span></a>
<a class="sourceLine" id="cb55-8" data-line-number="8"><span class="co">#&gt; 4:   00-1823-SRR471613    75 79.23348</span></a>
<a class="sourceLine" id="cb55-9" data-line-number="9"><span class="co">#&gt; 5:   01-1934-SRR471793    80 84.30801</span></a>
<a class="sourceLine" id="cb55-10" data-line-number="10"><span class="co">#&gt; 6:   01-2382-SRR471433    51 54.44634</span></a></code></pre></div>
<p>Now we can see their correlation.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">ggpubr<span class="op">::</span><span class="kw">ggscatter</span>(CN_merged_dt,</a>
<a class="sourceLine" id="cb56-2" data-line-number="2">  <span class="dt">x =</span> <span class="st">&quot;Total&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;Est&quot;</span>, <span class="dt">add =</span> <span class="st">&quot;reg.line&quot;</span>,</a>
<a class="sourceLine" id="cb56-3" data-line-number="3">  <span class="dt">xlab =</span> <span class="st">&quot;Log10 based total segment counts (observed)&quot;</span>,</a>
<a class="sourceLine" id="cb56-4" data-line-number="4">  <span class="dt">ylab =</span> <span class="st">&quot;Log10 based total segment counts (estimated)&quot;</span></a>
<a class="sourceLine" id="cb56-5" data-line-number="5">) <span class="op">+</span></a>
<a class="sourceLine" id="cb56-6" data-line-number="6"><span class="st">  </span>ggpubr<span class="op">::</span><span class="kw">stat_cor</span>(<span class="dt">label.x.npc =</span> <span class="fl">0.5</span>, <span class="dt">label.y.npc =</span> <span class="fl">0.8</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-41-1.png" width="2400" /></p>
<p>We can see that the estimation of copy number segment count fits observed data well. The residue can be further visualized by histgram.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1">CN_merged_dt[, Residue <span class="op">:</span><span class="er">=</span><span class="st"> </span>Est <span class="op">-</span><span class="st"> </span>Total]</a></code></pre></div>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb58-2" data-line-number="2"><span class="kw">ggplot</span>(CN_merged_dt) <span class="op">+</span></a>
<a class="sourceLine" id="cb58-3" data-line-number="3"><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Residue))</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-43-1.png" width="2400" /></p>
<hr />
</div>
</div>
</div>
</div>
<div id="association-analysis" class="section level1">
<h1>Association Analysis</h1>
<p>This part has been divided into 2 sections: data integration and association analysis.</p>
<div id="data-integration" class="section level2">
<h2>Data integration</h2>
<p>In this section, I integrate all genotype/phenotype data as a tidy data table used for downstream analyses.</p>
<p>Load packages and prepared data.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb59-2" data-line-number="2"><span class="kw">library</span>(sigminer)</a>
<a class="sourceLine" id="cb59-3" data-line-number="3"><span class="kw">library</span>(maftools)</a>
<a class="sourceLine" id="cb59-4" data-line-number="4"></a>
<a class="sourceLine" id="cb59-5" data-line-number="5"><span class="co"># Loading clinical related data -------------------------------------------</span></a>
<a class="sourceLine" id="cb59-6" data-line-number="6"></a>
<a class="sourceLine" id="cb59-7" data-line-number="7">Info &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../data/PRAD_CLINICAL.rds&quot;</span>)</a>
<a class="sourceLine" id="cb59-8" data-line-number="8"><span class="co"># Purity and ploidy info from facets</span></a>
<a class="sourceLine" id="cb59-9" data-line-number="9">PurityInfo &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/PRAD_Purity_and_Ploidy_Sequenza.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb59-10" data-line-number="10"></a>
<a class="sourceLine" id="cb59-11" data-line-number="11"><span class="co"># Processing CNV data -----------------------------------------------------</span></a>
<a class="sourceLine" id="cb59-12" data-line-number="12"></a>
<a class="sourceLine" id="cb59-13" data-line-number="13"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</a>
<a class="sourceLine" id="cb59-14" data-line-number="14"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a>
<a class="sourceLine" id="cb59-15" data-line-number="15">CNV &lt;-<span class="st"> </span>CNV.seqz</a>
<a class="sourceLine" id="cb59-16" data-line-number="16">Sig.CNV &lt;-<span class="st"> </span>Sig.CNV.seqz.W</a>
<a class="sourceLine" id="cb59-17" data-line-number="17"><span class="kw">rm</span>(Sig.CNV.seqz.W, CNV.seqz)</a></code></pre></div>
<p>After running NMF, we can get robust clusters from consensus matrix, more see <code>?NMF::predict</code>.</p>
<p>I also create a function <code>get_sig_exposure()</code> to get the absolute/relative exposure of signatures.</p>
<blockquote>
<p>Relative exposure is more useful for clustering and absolute exposure is more useful for association analysis.</p>
</blockquote>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1"><span class="co"># CNV</span></a>
<a class="sourceLine" id="cb60-2" data-line-number="2">CNVGroupInfo &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.CNV, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb60-3" data-line-number="3"><span class="co">#&gt; =&gt; Obtaining clusters from the hierarchical clustering of the consensus matrix...</span></a>
<a class="sourceLine" id="cb60-4" data-line-number="4"><span class="co">#&gt; =&gt; Finding the dominant signature of each group...</span></a>
<a class="sourceLine" id="cb60-5" data-line-number="5"><span class="co">#&gt; =&gt; Generating a table of group and dominant signature:</span></a>
<a class="sourceLine" id="cb60-6" data-line-number="6"><span class="co">#&gt;    </span></a>
<a class="sourceLine" id="cb60-7" data-line-number="7"><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5</span></a>
<a class="sourceLine" id="cb60-8" data-line-number="8"><span class="co">#&gt;   1    9    2   14   10    0</span></a>
<a class="sourceLine" id="cb60-9" data-line-number="9"><span class="co">#&gt;   2    0   60    0    3    0</span></a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="co">#&gt;   3    0    3  219    2    0</span></a>
<a class="sourceLine" id="cb60-11" data-line-number="11"><span class="co">#&gt;   4    0    0    0  170    0</span></a>
<a class="sourceLine" id="cb60-12" data-line-number="12"><span class="co">#&gt;   5    0    3   35  200  207</span></a>
<a class="sourceLine" id="cb60-13" data-line-number="13"><span class="co">#&gt; =&gt; Assigning a group to a signature with the maxium fraction (stored in &#39;map_table&#39; attr)...</span></a>
<a class="sourceLine" id="cb60-14" data-line-number="14"><span class="co">#&gt; =&gt; Summarizing...</span></a>
<a class="sourceLine" id="cb60-15" data-line-number="15"><span class="co">#&gt;  group #1: 35 samples with Sig3 enriched.</span></a>
<a class="sourceLine" id="cb60-16" data-line-number="16"><span class="co">#&gt;  group #2: 63 samples with Sig2 enriched.</span></a>
<a class="sourceLine" id="cb60-17" data-line-number="17"><span class="co">#&gt;  group #3: 224 samples with Sig3 enriched.</span></a>
<a class="sourceLine" id="cb60-18" data-line-number="18"><span class="co">#&gt;  group #4: 170 samples with Sig4 enriched.</span></a>
<a class="sourceLine" id="cb60-19" data-line-number="19"><span class="co">#&gt;  group #5: 445 samples with Sig5 enriched.</span></a>
<a class="sourceLine" id="cb60-20" data-line-number="20">CNVInfo &lt;-<span class="st"> </span>CNV<span class="op">@</span>summary.per.sample</a>
<a class="sourceLine" id="cb60-21" data-line-number="21"><span class="co"># Default is absolute exposure</span></a>
<a class="sourceLine" id="cb60-22" data-line-number="22">CNVExposureInfo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.CNV)</a></code></pre></div>
<p>We find that the group <code>1</code> is assigned to <code>Sig3</code> due to sample with <code>Sig3</code> dominant has the maximum fraction in group <code>1</code>. However, group <code>1</code> is the only group with <code>Sig1</code> enriched. So we modify this result.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1">CNVGroupInfo<span class="op">$</span>enrich_sig[CNVGroupInfo<span class="op">$</span>group <span class="op">==</span><span class="st"> &quot;1&quot;</span>] &lt;-<span class="st"> &quot;Sig1&quot;</span></a></code></pre></div>
<p>Next, we process mutation data to get TMB, driver info, clusters, and etc (the detail has been described in R script under <code>analysis/src/</code> of the repo).</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="co"># Processing mutation data ------------------------------------------------</span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2"></a>
<a class="sourceLine" id="cb62-3" data-line-number="3"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb62-4" data-line-number="4"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb62-5" data-line-number="5"></a>
<a class="sourceLine" id="cb62-6" data-line-number="6">TMBInfo &lt;-<span class="st"> </span>Maf<span class="op">@</span>variant.type.summary[, .(Tumor_Sample_Barcode, total, INS, DEL)]</a>
<a class="sourceLine" id="cb62-7" data-line-number="7">TMBInfo<span class="op">$</span>INDELs &lt;-<span class="st"> </span>TMBInfo<span class="op">$</span>INS <span class="op">+</span><span class="st"> </span>TMBInfo<span class="op">$</span>DEL</a>
<a class="sourceLine" id="cb62-8" data-line-number="8">TMBInfo<span class="op">$</span>INS &lt;-<span class="st"> </span>TMBInfo<span class="op">$</span>DEL &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb62-9" data-line-number="9"></a>
<a class="sourceLine" id="cb62-10" data-line-number="10"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_driver_info.RData&quot;</span>)</a>
<a class="sourceLine" id="cb62-11" data-line-number="11"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_heter_info.RData&quot;</span>)</a>
<a class="sourceLine" id="cb62-12" data-line-number="12"></a>
<a class="sourceLine" id="cb62-13" data-line-number="13">SNVGroupInfo &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.SNV, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb62-14" data-line-number="14"><span class="co">#&gt; =&gt; Obtaining clusters from the hierarchical clustering of the consensus matrix...</span></a>
<a class="sourceLine" id="cb62-15" data-line-number="15"><span class="co">#&gt; =&gt; Finding the dominant signature of each group...</span></a>
<a class="sourceLine" id="cb62-16" data-line-number="16"><span class="co">#&gt; =&gt; Generating a table of group and dominant signature:</span></a>
<a class="sourceLine" id="cb62-17" data-line-number="17"><span class="co">#&gt;    </span></a>
<a class="sourceLine" id="cb62-18" data-line-number="18"><span class="co">#&gt;     Sig1 Sig2 Sig3</span></a>
<a class="sourceLine" id="cb62-19" data-line-number="19"><span class="co">#&gt;   1   16   34   21</span></a>
<a class="sourceLine" id="cb62-20" data-line-number="20"><span class="co">#&gt;   2   30    0  193</span></a>
<a class="sourceLine" id="cb62-21" data-line-number="21"><span class="co">#&gt;   3  677    0    0</span></a>
<a class="sourceLine" id="cb62-22" data-line-number="22"><span class="co">#&gt; =&gt; Assigning a group to a signature with the maxium fraction (stored in &#39;map_table&#39; attr)...</span></a>
<a class="sourceLine" id="cb62-23" data-line-number="23"><span class="co">#&gt; =&gt; Summarizing...</span></a>
<a class="sourceLine" id="cb62-24" data-line-number="24"><span class="co">#&gt;  group #1: 71 samples with Sig2 enriched.</span></a>
<a class="sourceLine" id="cb62-25" data-line-number="25"><span class="co">#&gt;  group #2: 223 samples with Sig3 enriched.</span></a>
<a class="sourceLine" id="cb62-26" data-line-number="26"><span class="co">#&gt;  group #3: 677 samples with Sig1 enriched.</span></a>
<a class="sourceLine" id="cb62-27" data-line-number="27">SNVExposureInfo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.SNV)</a>
<a class="sourceLine" id="cb62-28" data-line-number="28"></a>
<a class="sourceLine" id="cb62-29" data-line-number="29"><span class="co"># Processing gene and pathway mutation ------------------------------------</span></a>
<a class="sourceLine" id="cb62-30" data-line-number="30"></a>
<a class="sourceLine" id="cb62-31" data-line-number="31"><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_gene_and_pathway_mutation.RData&quot;</span>)</a></code></pre></div>
<p>Next, we keep keys of all <code>data.frame</code>s are same and merge them into one.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="co"># Merge data --------------------------------------------------------------</span></a>
<a class="sourceLine" id="cb63-2" data-line-number="2">Info &lt;-<span class="st"> </span>Info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb63-4" data-line-number="4">    <span class="dt">CNV_ID =</span> dplyr<span class="op">::</span><span class="kw">case_when</span>(</a>
<a class="sourceLine" id="cb63-5" data-line-number="5">      <span class="op">!</span><span class="kw">startsWith</span>(tumor_Run, <span class="st">&quot;TCGA&quot;</span>) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(tumor_Run) <span class="op">~</span><span class="st"> </span><span class="kw">paste</span>(subject_id, tumor_Run, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>),</a>
<a class="sourceLine" id="cb63-6" data-line-number="6">      <span class="kw">startsWith</span>(tumor_Run, <span class="st">&quot;TCGA&quot;</span>) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(tumor_Run) <span class="op">~</span><span class="st"> </span>tumor_Run,</a>
<a class="sourceLine" id="cb63-7" data-line-number="7">      <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span><span class="ot">NA_character_</span></a>
<a class="sourceLine" id="cb63-8" data-line-number="8">    )</a>
<a class="sourceLine" id="cb63-9" data-line-number="9">  )</a>
<a class="sourceLine" id="cb63-10" data-line-number="10"></a>
<a class="sourceLine" id="cb63-11" data-line-number="11">PurityInfo</a>
<a class="sourceLine" id="cb63-12" data-line-number="12"><span class="co">#&gt; # A tibble: 938 x 3</span></a>
<a class="sourceLine" id="cb63-13" data-line-number="13"><span class="co">#&gt;    purity ploidy sample             </span></a>
<a class="sourceLine" id="cb63-14" data-line-number="14"><span class="co">#&gt;     &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;              </span></a>
<a class="sourceLine" id="cb63-15" data-line-number="15"><span class="co">#&gt;  1   0.59    1.9 00-000450-SRR474658</span></a>
<a class="sourceLine" id="cb63-16" data-line-number="16"><span class="co">#&gt;  2   0.55    1.9 00-1165-SRR461844  </span></a>
<a class="sourceLine" id="cb63-17" data-line-number="17"><span class="co">#&gt;  3   0.75    1.9 00-160-SRR471373   </span></a>
<a class="sourceLine" id="cb63-18" data-line-number="18"><span class="co">#&gt;  4   0.72    2   00-1823-SRR471613  </span></a>
<a class="sourceLine" id="cb63-19" data-line-number="19"><span class="co">#&gt;  5   0.73    2   01-1934-SRR471793  </span></a>
<a class="sourceLine" id="cb63-20" data-line-number="20"><span class="co">#&gt;  6   0.4     1.9 01-2382-SRR471433  </span></a>
<a class="sourceLine" id="cb63-21" data-line-number="21"><span class="co">#&gt;  7   0.45    4.1 01-2492-SRR471733  </span></a>
<a class="sourceLine" id="cb63-22" data-line-number="22"><span class="co">#&gt;  8   0.73    2   01-2554-SRR471673  </span></a>
<a class="sourceLine" id="cb63-23" data-line-number="23"><span class="co">#&gt;  9   0.22    3.4 01-28-SRR471073    </span></a>
<a class="sourceLine" id="cb63-24" data-line-number="24"><span class="co">#&gt; 10   0.35    2   02-1082-SRR472616  </span></a>
<a class="sourceLine" id="cb63-25" data-line-number="25"><span class="co">#&gt; # … with 928 more rows</span></a>
<a class="sourceLine" id="cb63-26" data-line-number="26"><span class="kw">colnames</span>(CNVGroupInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cnv_group&quot;</span>, <span class="st">&quot;cnv_weight&quot;</span>, <span class="st">&quot;cnv_enrich_sig&quot;</span>)</a>
<a class="sourceLine" id="cb63-27" data-line-number="27">CNVInfo</a>
<a class="sourceLine" id="cb63-28" data-line-number="28"><span class="co">#&gt;                  sample n_of_cnv n_of_amp n_of_del n_of_vchr cna_burden</span></a>
<a class="sourceLine" id="cb63-29" data-line-number="29"><span class="co">#&gt;   1:    TCGA-HI-7169-01        2        2        0         0      0.000</span></a>
<a class="sourceLine" id="cb63-30" data-line-number="30"><span class="co">#&gt;   2:    TCGA-CH-5743-01        3        2        1         1      0.000</span></a>
<a class="sourceLine" id="cb63-31" data-line-number="31"><span class="co">#&gt;   3:    TCGA-EJ-7791-01        3        3        0         1      0.000</span></a>
<a class="sourceLine" id="cb63-32" data-line-number="32"><span class="co">#&gt;   4:    TCGA-FC-A66V-01        3        3        0         1      0.000</span></a>
<a class="sourceLine" id="cb63-33" data-line-number="33"><span class="co">#&gt;   5:    TCGA-J4-AATV-01        3        3        0         0      0.000</span></a>
<a class="sourceLine" id="cb63-34" data-line-number="34"><span class="co">#&gt;  ---                                                                   </span></a>
<a class="sourceLine" id="cb63-35" data-line-number="35"><span class="co">#&gt; 933: 5115609-SRR8311697      303      291       12        22      0.856</span></a>
<a class="sourceLine" id="cb63-36" data-line-number="36"><span class="co">#&gt; 934: 5115412-SRR8311627      356      354        2        22      0.948</span></a>
<a class="sourceLine" id="cb63-37" data-line-number="37"><span class="co">#&gt; 935: 5115406-SRR8312068      367      337       30        22      0.297</span></a>
<a class="sourceLine" id="cb63-38" data-line-number="38"><span class="co">#&gt; 936:      36-SRR5876749      393      392        1        22      0.926</span></a>
<a class="sourceLine" id="cb63-39" data-line-number="39"><span class="co">#&gt; 937: 5115615-SRR8311749      420      406       14        22      0.360</span></a>
<a class="sourceLine" id="cb63-40" data-line-number="40"><span class="kw">colnames</span>(CNVExposureInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;CN_&quot;</span>, <span class="kw">colnames</span>(CNVExposureInfo)[<span class="op">-</span><span class="dv">1</span>]))</a>
<a class="sourceLine" id="cb63-41" data-line-number="41"><span class="kw">colnames</span>(TMBInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;INDELs&quot;</span>)</a>
<a class="sourceLine" id="cb63-42" data-line-number="42">TMBInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-43" data-line-number="43">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-44" data-line-number="44">  sample</a>
<a class="sourceLine" id="cb63-45" data-line-number="45">)]</a>
<a class="sourceLine" id="cb63-46" data-line-number="46">DriverDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-47" data-line-number="47">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-48" data-line-number="48">  sample</a>
<a class="sourceLine" id="cb63-49" data-line-number="49">)]</a>
<a class="sourceLine" id="cb63-50" data-line-number="50"><span class="kw">colnames</span>(SNVGroupInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;snv_group&quot;</span>, <span class="st">&quot;snv_weight&quot;</span>, <span class="st">&quot;snv_enrich_sig&quot;</span>)</a>
<a class="sourceLine" id="cb63-51" data-line-number="51">SNVGroupInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-52" data-line-number="52">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-53" data-line-number="53">  sample</a>
<a class="sourceLine" id="cb63-54" data-line-number="54">)]</a>
<a class="sourceLine" id="cb63-55" data-line-number="55"><span class="kw">colnames</span>(SNVExposureInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;SBS_&quot;</span>, <span class="kw">colnames</span>(SNVExposureInfo)[<span class="op">-</span><span class="dv">1</span>]))</a>
<a class="sourceLine" id="cb63-56" data-line-number="56">SNVExposureInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-57" data-line-number="57">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-58" data-line-number="58">  sample</a>
<a class="sourceLine" id="cb63-59" data-line-number="59">)]</a>
<a class="sourceLine" id="cb63-60" data-line-number="60"><span class="kw">colnames</span>(TitvInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>)</a>
<a class="sourceLine" id="cb63-61" data-line-number="61">TitvInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-62" data-line-number="62">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-63" data-line-number="63">  sample</a>
<a class="sourceLine" id="cb63-64" data-line-number="64">)]</a>
<a class="sourceLine" id="cb63-65" data-line-number="65"><span class="kw">colnames</span>(MathDF) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;MATH&quot;</span>)</a>
<a class="sourceLine" id="cb63-66" data-line-number="66">MathDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-67" data-line-number="67">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-68" data-line-number="68">  sample</a>
<a class="sourceLine" id="cb63-69" data-line-number="69">)]</a>
<a class="sourceLine" id="cb63-70" data-line-number="70"><span class="kw">colnames</span>(ClusterDF) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cluster&quot;</span>)</a>
<a class="sourceLine" id="cb63-71" data-line-number="71">ClusterDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-72" data-line-number="72">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-73" data-line-number="73">  sample</a>
<a class="sourceLine" id="cb63-74" data-line-number="74">)]</a>
<a class="sourceLine" id="cb63-75" data-line-number="75"></a>
<a class="sourceLine" id="cb63-76" data-line-number="76">data.table<span class="op">::</span><span class="kw">setDT</span>(summary_mutation)</a>
<a class="sourceLine" id="cb63-77" data-line-number="77">data.table<span class="op">::</span><span class="kw">setDT</span>(summary_pathway)</a>
<a class="sourceLine" id="cb63-78" data-line-number="78"></a>
<a class="sourceLine" id="cb63-79" data-line-number="79"><span class="kw">colnames</span>(summary_mutation)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;sample&quot;</span></a>
<a class="sourceLine" id="cb63-80" data-line-number="80"><span class="kw">colnames</span>(summary_pathway)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;sample&quot;</span></a>
<a class="sourceLine" id="cb63-81" data-line-number="81">summary_mutation[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-82" data-line-number="82">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-83" data-line-number="83">  sample</a>
<a class="sourceLine" id="cb63-84" data-line-number="84">)]</a>
<a class="sourceLine" id="cb63-85" data-line-number="85">summary_pathway[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</a>
<a class="sourceLine" id="cb63-86" data-line-number="86">  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</a>
<a class="sourceLine" id="cb63-87" data-line-number="87">  sample</a>
<a class="sourceLine" id="cb63-88" data-line-number="88">)]</a>
<a class="sourceLine" id="cb63-89" data-line-number="89"></a>
<a class="sourceLine" id="cb63-90" data-line-number="90"></a>
<a class="sourceLine" id="cb63-91" data-line-number="91">MergeInfo &lt;-<span class="st"> </span>Info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-92" data-line-number="92"><span class="st">  </span><span class="kw">left_join</span>(CNVInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-93" data-line-number="93"><span class="st">  </span><span class="kw">left_join</span>(CNVGroupInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-94" data-line-number="94"><span class="st">  </span><span class="kw">left_join</span>(CNVExposureInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-95" data-line-number="95"><span class="st">  </span><span class="kw">left_join</span>(PurityInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-96" data-line-number="96"><span class="st">  </span><span class="kw">left_join</span>(summary_mutation, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-97" data-line-number="97"><span class="st">  </span><span class="kw">left_join</span>(summary_pathway, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-98" data-line-number="98"><span class="st">  </span><span class="kw">left_join</span>(TMBInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-99" data-line-number="99"><span class="st">  </span><span class="kw">left_join</span>(DriverDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-100" data-line-number="100"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb63-101" data-line-number="101">    <span class="dt">n_driver =</span> <span class="kw">ifelse</span>(<span class="op">!</span><span class="kw">is.na</span>(n_driver), n_driver, <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb63-102" data-line-number="102">  ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-103" data-line-number="103"><span class="st">  </span><span class="kw">left_join</span>(TitvInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-104" data-line-number="104"><span class="st">  </span><span class="kw">left_join</span>(MathDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-105" data-line-number="105"><span class="st">  </span><span class="kw">left_join</span>(ClusterDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-106" data-line-number="106"><span class="st">  </span><span class="kw">left_join</span>(SNVGroupInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-107" data-line-number="107"><span class="st">  </span><span class="kw">left_join</span>(SNVExposureInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb63-108" data-line-number="108"><span class="st">  </span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb63-109" data-line-number="109">    <span class="dt">Stage =</span> <span class="kw">factor</span>(Stage, <span class="dt">ordered =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb63-110" data-line-number="110">    <span class="dt">Fusion =</span> <span class="kw">ifelse</span>(Fusion <span class="op">==</span><span class="st"> &quot;Negative&quot;</span>, <span class="st">&quot;No&quot;</span>, <span class="st">&quot;Yes&quot;</span>),</a>
<a class="sourceLine" id="cb63-111" data-line-number="111">    <span class="dt">sample_type =</span> <span class="kw">ifelse</span>(sample_type <span class="op">==</span><span class="st"> &quot;Unknown&quot;</span>, <span class="ot">NA_character_</span>, sample_type),</a>
<a class="sourceLine" id="cb63-112" data-line-number="112">    <span class="dt">HasFusion =</span> Fusion,</a>
<a class="sourceLine" id="cb63-113" data-line-number="113">    <span class="dt">HasFusion =</span> <span class="kw">ifelse</span>(HasFusion <span class="op">==</span><span class="st"> &quot;Yes&quot;</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>),</a>
<a class="sourceLine" id="cb63-114" data-line-number="114">    <span class="dt">IsMetastatic =</span> <span class="kw">ifelse</span>(sample_type <span class="op">==</span><span class="st"> &quot;Metastatic&quot;</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb63-115" data-line-number="115">  )</a>
<a class="sourceLine" id="cb63-116" data-line-number="116"></a>
<a class="sourceLine" id="cb63-117" data-line-number="117"><span class="kw">summary</span>(MergeInfo)</a>
<a class="sourceLine" id="cb63-118" data-line-number="118"><span class="co">#&gt;     Study            subject_id        sample_type        tumor_body_site   </span></a>
<a class="sourceLine" id="cb63-119" data-line-number="119"><span class="co">#&gt;   tumor_Run          normal_Run         PatientID              Age      </span></a>
<a class="sourceLine" id="cb63-120" data-line-number="120"><span class="co">#&gt;   Stage          PSA          GleasonScore       Fusion         </span></a>
<a class="sourceLine" id="cb63-121" data-line-number="121"><span class="co">#&gt;     CNV_ID             n_of_cnv        n_of_amp         n_of_del     </span></a>
<a class="sourceLine" id="cb63-122" data-line-number="122"><span class="co">#&gt;    n_of_vchr       cna_burden     cnv_group           cnv_weight    </span></a>
<a class="sourceLine" id="cb63-123" data-line-number="123"><span class="co">#&gt;  cnv_enrich_sig        CN_Sig1          CN_Sig2           CN_Sig3        </span></a>
<a class="sourceLine" id="cb63-124" data-line-number="124"><span class="co">#&gt;     CN_Sig4          CN_Sig5           purity          ploidy     </span></a>
<a class="sourceLine" id="cb63-125" data-line-number="125"><span class="co">#&gt;    ACVR2A          APBB2            APC              AR         </span></a>
<a class="sourceLine" id="cb63-126" data-line-number="126"><span class="co">#&gt;     ARNT            ATM            ATXN2            BIN1        </span></a>
<a class="sourceLine" id="cb63-127" data-line-number="127"><span class="co">#&gt;    CDKN1B          CPLX4           CPNE3          DEPDC1B       </span></a>
<a class="sourceLine" id="cb63-128" data-line-number="128"><span class="co">#&gt;     ETV3           FOXA1           IL6ST           KDM6A        </span></a>
<a class="sourceLine" id="cb63-129" data-line-number="129"><span class="co">#&gt;    KRT85           LCE2D           LCE4A           NUDT22       </span></a>
<a class="sourceLine" id="cb63-130" data-line-number="130"><span class="co">#&gt;     PGLS           PIK3R1          PRPF19           PTEN        </span></a>
<a class="sourceLine" id="cb63-131" data-line-number="131"><span class="co">#&gt;     RB1            RNF43            RTKN            SKA3        </span></a>
<a class="sourceLine" id="cb63-132" data-line-number="132"><span class="co">#&gt;     SPOP           SYPL1          TMPRSS2           TP53        </span></a>
<a class="sourceLine" id="cb63-133" data-line-number="133"><span class="co">#&gt;    USP28           ZNF17           CDK12           ZMYM3        </span></a>
<a class="sourceLine" id="cb63-134" data-line-number="134"><span class="co">#&gt;    CTNNB1          ZFHX3            ERF           HLA-DRB1      </span></a>
<a class="sourceLine" id="cb63-135" data-line-number="135"><span class="co">#&gt;    DHX30           CASZ1           NCOR1          COL11A1       </span></a>
<a class="sourceLine" id="cb63-136" data-line-number="136"><span class="co">#&gt;   SLC45A3          NKX3-1           JAK1         Cell_Cycle     </span></a>
<a class="sourceLine" id="cb63-137" data-line-number="137"><span class="co">#&gt;    Hippo            MYC            NOTCH            NRF2        </span></a>
<a class="sourceLine" id="cb63-138" data-line-number="138"><span class="co">#&gt;     PI3K          TGF-Beta        RTK-RAS        TP53_pathway   </span></a>
<a class="sourceLine" id="cb63-139" data-line-number="139"><span class="co">#&gt;     WNT          HR_pathway      total_mutation        INDELs       </span></a>
<a class="sourceLine" id="cb63-140" data-line-number="140"><span class="co">#&gt;     n_driver        Ti_fraction      Tv_fraction          MATH       </span></a>
<a class="sourceLine" id="cb63-141" data-line-number="141"><span class="co">#&gt;     cluster       snv_group           snv_weight      snv_enrich_sig    </span></a>
<a class="sourceLine" id="cb63-142" data-line-number="142"><span class="co">#&gt;     SBS_Sig1         SBS_Sig2           SBS_Sig3        HasFusion      </span></a>
<a class="sourceLine" id="cb63-143" data-line-number="143"><span class="co">#&gt;  IsMetastatic   </span></a>
<a class="sourceLine" id="cb63-144" data-line-number="144"><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 7 rows ]</span></a>
<a class="sourceLine" id="cb63-145" data-line-number="145"><span class="kw">saveRDS</span>(MergeInfo, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</a></code></pre></div>
</div>
<div id="association-analysis-1" class="section level2">
<h2>Association analysis</h2>
<p>Now we do association analysis between signatures (via exposure) and genotypes/phenotypes.</p>
<div id="prepare" class="section level3">
<h3>Prepare</h3>
<p>Load packages.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb64-2" data-line-number="2"><span class="kw">library</span>(sigminer)</a></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1">df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</a></code></pre></div>
<p>Set the coloumns used for analysis.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1">cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CN_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SBS_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</a>
<a class="sourceLine" id="cb66-2" data-line-number="2"><span class="co"># Driver genes identified by MutSig software</span></a>
<a class="sourceLine" id="cb66-3" data-line-number="3">cols_to_mutated_genes &lt;-<span class="st"> </span><span class="kw">colnames</span>(df.seqz)[<span class="dv">29</span><span class="op">:</span><span class="dv">75</span>]</a>
<a class="sourceLine" id="cb66-4" data-line-number="4"><span class="co"># 10 oncogenic pathways in TCGA 2018 papers + HR pathway</span></a>
<a class="sourceLine" id="cb66-5" data-line-number="5">cols_to_mutated_pathways &lt;-<span class="st"> </span><span class="kw">colnames</span>(df.seqz)[<span class="dv">76</span><span class="op">:</span><span class="dv">86</span>]</a>
<a class="sourceLine" id="cb66-6" data-line-number="6"></a>
<a class="sourceLine" id="cb66-7" data-line-number="7"><span class="co"># Exclude PSA</span></a>
<a class="sourceLine" id="cb66-8" data-line-number="8">cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb66-9" data-line-number="9">  <span class="st">&quot;IsMetastatic&quot;</span>, <span class="st">&quot;HasFusion&quot;</span>,</a>
<a class="sourceLine" id="cb66-10" data-line-number="10">  <span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</a>
<a class="sourceLine" id="cb66-11" data-line-number="11">  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;INDELs&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</a>
<a class="sourceLine" id="cb66-12" data-line-number="12">  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</a>
<a class="sourceLine" id="cb66-13" data-line-number="13">  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></a>
<a class="sourceLine" id="cb66-14" data-line-number="14">)</a>
<a class="sourceLine" id="cb66-15" data-line-number="15"></a>
<a class="sourceLine" id="cb66-16" data-line-number="16">feature_type &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;ca&quot;</span>, <span class="dv">2</span>), <span class="kw">rep</span>(<span class="st">&quot;co&quot;</span>, <span class="dv">17</span>))</a></code></pre></div>
</div>
<div id="signatures-features" class="section level3">
<h3>Signatures &amp; features</h3>
<p>Analyze the association between signatures and features.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1">tidy_data.seqz.feature &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</a>
<a class="sourceLine" id="cb67-2" data-line-number="2">  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb67-3" data-line-number="3">  <span class="dt">cols_to_features =</span> cols_to_features,</a>
<a class="sourceLine" id="cb67-4" data-line-number="4">  <span class="dt">method_co =</span> <span class="st">&quot;pearson&quot;</span>,</a>
<a class="sourceLine" id="cb67-5" data-line-number="5">  <span class="dt">type =</span> feature_type, <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb67-6" data-line-number="6">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb67-7" data-line-number="7"><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>) <span class="co"># adjust p value with FDR</span></a>
<a class="sourceLine" id="cb67-8" data-line-number="8"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb67-9" data-line-number="9"><span class="co">#&gt; -&gt; Calculating correlations for continuous variables using pearson...</span></a>
<a class="sourceLine" id="cb67-10" data-line-number="10"><span class="co">#&gt; -&gt; Calculating correlations for categorical variables using wilcox.test...</span></a>
<a class="sourceLine" id="cb67-11" data-line-number="11"><span class="co">#&gt; --&gt; building test models, be patient...</span></a>
<a class="sourceLine" id="cb67-12" data-line-number="12"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb67-13" data-line-number="13"><span class="co">#&gt; --&gt; obtaining model results...</span></a>
<a class="sourceLine" id="cb67-14" data-line-number="14"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb67-15" data-line-number="15"><span class="co">#&gt; --&gt; collecting data...</span></a>
<a class="sourceLine" id="cb67-16" data-line-number="16"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb67-17" data-line-number="17"><span class="co">#&gt; -&gt; Done</span></a></code></pre></div>
<blockquote>
<p>Association of signature exposures with other features was performed using one of two procedures: for a continuous association variable (including ordinal variable), pearson correaltion was performed; for a binary association variable, samples were divided into two groups and Mann-Whitney U-test was performed to test for differences in signature exposure medians between the two groups.</p>
</blockquote>
<p>Show the result.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1"><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.feature, <span class="dt">p_val =</span> <span class="fl">0.05</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-54-1.png" width="2700" /></p>
<blockquote>
<p>All shown circles are statistically significant results (FDR &lt; 0.05). A feature will be chopped off from the plot if has no correlation with any signatures. Same for the similar plots below.</p>
</blockquote>
<p>The result table:</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(tidy_data.seqz.feature)</a></code></pre></div>
<div id="htmlwidget-621273037699d23904ce" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-621273037699d23904ce">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216"],["CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","CN_Sig1","CN_Sig1","CN_Sig2","CN_Sig2","CN_Sig3","CN_Sig3","CN_Sig4","CN_Sig4","CN_Sig5","CN_Sig5","SBS_Sig1","SBS_Sig1","SBS_Sig2","SBS_Sig2","SBS_Sig3","SBS_Sig3"],["Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","Age","cluster","CN_Sig1","CN_Sig2","CN_Sig3","CN_Sig4","CN_Sig5","cna_burden","GleasonScore","INDELs","MATH","n_driver","n_of_amp","n_of_cnv","n_of_del","n_of_vchr","ploidy","purity","SBS_Sig1","SBS_Sig2","SBS_Sig3","Stage","Ti_fraction","total_mutation","Tv_fraction","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic","HasFusion","IsMetastatic"],[807,912,937,937,937,937,937,937,621,923,912,937,937,937,937,937,937,937,925,925,925,576,925,923,925,807,912,937,937,937,937,937,937,621,923,912,937,937,937,937,937,937,937,925,925,925,576,925,923,925,807,912,937,937,937,937,937,937,621,923,912,937,937,937,937,937,937,937,925,925,925,576,925,923,925,807,912,937,937,937,937,937,937,621,923,912,937,937,937,937,937,937,937,925,925,925,576,925,923,925,807,912,937,937,937,937,937,937,621,923,912,937,937,937,937,937,937,937,925,925,925,576,925,923,925,797,927,925,925,925,925,925,925,616,959,927,971,925,925,925,925,926,926,971,971,971,571,971,959,971,797,927,925,925,925,925,925,925,616,959,927,971,925,925,925,925,926,926,971,971,971,571,971,959,971,797,927,925,925,925,925,925,925,616,959,927,971,925,925,925,925,926,926,971,971,971,571,971,959,971,423,927,423,927,423,927,423,927,423,927,422,961,422,961,422,961],[0.171,0.114,1,0.395,0.47,0.024,-0.591,0.306,-0.0434,0.0231,0.012,0.123,0.622,0.638,0.139,0.459,0.217,-0.0316,0.162,-0.0155,0.00161,-0.0179,-0.0317,0.00536,0.0317,0.245,0.161,0.395,1,0.381,0.272,-0.663,0.283,0.128,0.0591,-0.0476,0.195,0.82,0.852,0.253,0.436,0.186,0.338,0.312,-0.00556,0.00793,0.0112,-0.084,0.0325,0.084,0.216,0.0535,0.47,0.381,1,-0.352,-0.732,0.874,0.218,0.0227,0.273,0.137,0.818,0.768,-0.25,0.681,0.8,0.0347,0.384,-0.0199,-0.0212,0.132,-0.139,0.0177,0.139,0.201,0.104,0.024,0.272,-0.352,1,-0.355,-0.317,0.0602,0.0757,-0.287,0.202,-0.0524,0.0884,0.864,0.196,-0.426,0.402,0.0993,-0.01,0.0227,0.114,-0.035,0.0162,0.035,-0.351,-0.143,-0.591,-0.663,-0.732,-0.355,1,-0.633,-0.247,-0.0766,-0.0483,-0.283,-0.836,-0.886,-0.361,-0.836,-0.473,-0.315,-0.442,0.0277,0.00454,-0.214,0.152,-0.0275,-0.152,0.274,0.289,0.162,0.312,0.384,0.0993,-0.442,0.335,0.27,0.453,0.176,0.316,0.404,0.412,0.0717,0.339,0.247,0.164,1,0.0332,0.0483,0.171,-0.265,0.179,0.265,0.0766,0.153,-0.0155,-0.00556,-0.0199,-0.01,0.0277,-0.0265,0.0667,0.271,-0.0248,0.43,-0.0147,-0.0171,-0.0161,-0.0462,-0.0219,0.0589,0.0332,1,0.867,0.0309,0.0641,0.965,-0.0641,0.0901,0.244,0.00161,0.00793,-0.0212,0.0227,0.00454,-0.0337,0.0912,0.518,-0.0336,0.571,-0.00532,-0.00341,0.0118,-0.0311,-0.0257,0.0876,0.0483,0.867,1,0.0491,0.158,0.941,-0.158,-4.04,13.2,-16,50.1,-14.6,39.6,-1.87,13.5,3.28,-10.6,-6.36,21.4,18.3,0.22,6.51,11],[2.03846153846154e-06,0.000998260869565217,0,1.08571428571429e-35,6.32432432432432e-52,0.554491017964072,1.8e-88,2.46176470588235e-21,0.383561643835616,0.574269005847953,0.774193548387097,0.000285964912280702,1.38666666666667e-100,3.22857142857143e-107,3.68807339449541e-05,3.54736842105263e-49,4.7906976744186e-11,0.428025477707006,1.52352941176471e-06,0.701098901098901,0.961,0.727173913043478,0.428025477707006,0.894358974358974,0.428025477707006,3.71428571428571e-12,1.96116504854369e-06,1.08571428571429e-35,0,3.48301886792453e-33,5.76e-17,1.28148148148148e-118,2.92957746478873e-18,0.00248275862068966,0.108358208955224,0.211188811188811,3.8021978021978e-09,1.52631578947368e-227,9.96249999999999e-264,9.05e-15,3.8e-44,2.14193548387097e-08,7.22033898305085e-26,7.73134328358209e-22,0.894358974358974,0.848167539267016,0.834920634920635,0.01696,0.420779220779221,0.01696,1.35280898876404e-09,0.156204379562044,6.32432432432432e-52,3.48301886792453e-33,0,3.75862068965517e-28,2.48333333333333e-157,8.64285714285714e-294,8.14736842105263e-08,0.574269005847953,1.36923076923077e-16,4.82142857142857e-05,7.72e-226,1.74545454545454e-182,1.76543209876543e-14,5.728e-128,6.52380952380952e-209,0.386577181208054,3.28235294117647e-33,0.619318181818182,0.596551724137931,0.00258119658119658,4.10810810810811e-05,0.668926553672316,4.10810810810811e-05,1.86521739130435e-08,0.00294915254237288,0.554491017964072,5.76e-17,3.75862068965517e-28,0,1.125e-28,7.28125e-23,0.190070921985816,0.0334375,2.74285714285714e-18,9.14942528735632e-10,0.157971014492754,0.0111147540983607,2.81333333333333e-279,3.17777777777778e-09,6.54545454545455e-42,4.00434782608696e-37,0.00416666666666667,0.809574468085106,0.574269005847953,0.0104462809917355,0.386577181208054,0.693333333333333,0.386577181208054,2.63870967741935e-24,2.62962962962963e-05,1.8e-88,1.28148148148148e-118,2.48333333333333e-157,1.125e-28,0,3.29655172413793e-105,1.01590909090909e-09,0.031496062992126,0.204225352112676,3.19444444444444e-18,3.31111111111111e-245,1.51272727271036e-312,1.19259259259259e-29,2.64705882352941e-245,9.94285714285714e-53,1.47692307692308e-22,7.75609756097561e-45,0.5,0.903553299492386,4.55670103092783e-07,5.96261682242991e-06,0.503105590062112,5.96261682242991e-06,9.05e-15,8.17391304347826e-19,1.52352941176471e-06,7.73134328358209e-22,3.28235294117647e-33,0.00416666666666667,7.75609756097561e-45,4.19672131147541e-25,2.03529411764706e-11,4.93846153846154e-49,1.41666666666667e-07,1.73333333333333e-23,4.85106382978723e-37,1.39555555555556e-38,0.0447692307692308,8.13333333333333e-26,5.5609756097561e-14,1.05510204081633e-06,0,0.4,0.188571428571429,6.86725663716814e-05,1.20519480519481e-16,4.85106382978723e-08,1.20519480519481e-16,0.0467175572519084,5.42857142857143e-06,0.701098901098901,0.894358974358974,0.619318181818182,0.809574468085106,0.5,0.518518518518518,0.144117647058824,4.13698630136986e-17,0.55,2.72380952380952e-44,0.715846994535519,0.67752808988764,0.693333333333333,0.222222222222222,0.588372093023256,0.108444444444444,0.4,0,4.78461538461538e-294,0.554491017964072,0.0691729323308271,0,0.0691729323308271,0.0174603174603175,1.11566265060241e-13,0.961,0.848167539267016,0.596551724137931,0.574269005847953,0.903553299492386,0.401307189542484,0.0365891472868217,2.66470588235294e-66,0.401307189542484,2.14545454545455e-84,0.894358974358974,0.927272727272727,0.774193548387097,0.436708860759494,0.533742331288344,0.0124227642276423,0.188571428571429,4.78461538461538e-294,0,0.332413793103448,1.412e-06,0,1.412e-06,2.06e-06,3.61066666666667e-66,0.0111384615384615,1.488e-93,3.11272727272727e-05,2.428e-49,0.721,1.83111111111111e-05,3.11272727272727e-05,4e-102,3.90666666666667e-05,3.872e-32,0.0556571428571429,1.288e-21,0.168533333333333,4.16e-14],["co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","co","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>signature<\/th>\n      <th>feature<\/th>\n      <th>count<\/th>\n      <th>measure<\/th>\n      <th>p<\/th>\n      <th>type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="signatures-mutated-genes" class="section level3">
<h3>Signatures &amp; mutated genes</h3>
<p>Analyze the association between signatures and mutated genes (only driver genes identified by MutSig).</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1">tidy_data.seqz.gene &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</a>
<a class="sourceLine" id="cb70-2" data-line-number="2">  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb70-3" data-line-number="3">  <span class="dt">cols_to_features =</span> cols_to_mutated_genes,</a>
<a class="sourceLine" id="cb70-4" data-line-number="4">  <span class="dt">type =</span> <span class="st">&quot;ca&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb70-5" data-line-number="5">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb70-6" data-line-number="6"><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb70-7" data-line-number="7"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb70-8" data-line-number="8"><span class="co">#&gt; -&gt; Calculating correlations for categorical variables using wilcox.test...</span></a>
<a class="sourceLine" id="cb70-9" data-line-number="9"><span class="co">#&gt; --&gt; building test models, be patient...</span></a>
<a class="sourceLine" id="cb70-10" data-line-number="10"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb70-11" data-line-number="11"><span class="co">#&gt; --&gt; obtaining model results...</span></a>
<a class="sourceLine" id="cb70-12" data-line-number="12"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb70-13" data-line-number="13"><span class="co">#&gt; --&gt; collecting data...</span></a>
<a class="sourceLine" id="cb70-14" data-line-number="14"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb70-15" data-line-number="15"><span class="co">#&gt; -&gt; Done</span></a></code></pre></div>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb71-1" data-line-number="1">sum_genes &lt;-<span class="st"> </span>tidy_data.seqz.gene <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb71-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(p <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb71-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(</a>
<a class="sourceLine" id="cb71-4" data-line-number="4">    <span class="dt">min =</span> <span class="kw">min</span>(measure, <span class="dt">na.rm =</span> T),</a>
<a class="sourceLine" id="cb71-5" data-line-number="5">    <span class="dt">max =</span> <span class="kw">max</span>(measure, <span class="dt">na.rm =</span> T),</a>
<a class="sourceLine" id="cb71-6" data-line-number="6">    <span class="dt">sum =</span> <span class="kw">list</span>(<span class="kw">summary</span>(measure))</a>
<a class="sourceLine" id="cb71-7" data-line-number="7">  )</a>
<a class="sourceLine" id="cb71-8" data-line-number="8"></a>
<a class="sourceLine" id="cb71-9" data-line-number="9">sum_genes<span class="op">$</span>sum</a>
<a class="sourceLine" id="cb71-10" data-line-number="10"><span class="co">#&gt; [[1]]</span></a>
<a class="sourceLine" id="cb71-11" data-line-number="11"><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></a>
<a class="sourceLine" id="cb71-12" data-line-number="12"><span class="co">#&gt;  -14.00    6.00   22.45   98.94   85.38  762.00</span></a></code></pre></div>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1"><span class="co"># myPalette &lt;- colorRampPalette(rev(brewer.pal(11, &quot;RdYlBu&quot;)))</span></a>
<a class="sourceLine" id="cb72-2" data-line-number="2"><span class="co"># sc &lt;- scale_colour_gradientn(colours = myPalette(100), limits=c(-20, 600))</span></a>
<a class="sourceLine" id="cb72-3" data-line-number="3"></a>
<a class="sourceLine" id="cb72-4" data-line-number="4">sc &lt;-<span class="st"> </span><span class="kw">scale_colour_gradientn</span>(</a>
<a class="sourceLine" id="cb72-5" data-line-number="5">  <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;red&quot;</span>),</a>
<a class="sourceLine" id="cb72-6" data-line-number="6">  <span class="dt">values =</span> scales<span class="op">::</span><span class="kw">rescale</span>(<span class="kw">c</span>(</a>
<a class="sourceLine" id="cb72-7" data-line-number="7">    sum_genes<span class="op">$</span>min,</a>
<a class="sourceLine" id="cb72-8" data-line-number="8">    <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb72-9" data-line-number="9">    <span class="dv">20</span>,</a>
<a class="sourceLine" id="cb72-10" data-line-number="10">    sum_genes<span class="op">$</span>max</a>
<a class="sourceLine" id="cb72-11" data-line-number="11">  ))</a>
<a class="sourceLine" id="cb72-12" data-line-number="12">)</a>
<a class="sourceLine" id="cb72-13" data-line-number="13"></a>
<a class="sourceLine" id="cb72-14" data-line-number="14"><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.gene,</a>
<a class="sourceLine" id="cb72-15" data-line-number="15">  <span class="dt">ylab =</span> <span class="st">&quot;Mutated genes&quot;</span>,</a>
<a class="sourceLine" id="cb72-16" data-line-number="16">  <span class="dt">ca_gradient_colors =</span> sc,</a>
<a class="sourceLine" id="cb72-17" data-line-number="17">  <span class="dt">p_val =</span> <span class="fl">0.05</span></a>
<a class="sourceLine" id="cb72-18" data-line-number="18">) <span class="co"># p_val needs to keep in line with the sum_genes p &lt;= 0.05</span></a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-58-1.png" width="2700" /></p>
<p>The result table:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb73-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(tidy_data.seqz.gene)</a></code></pre></div>
<div id="htmlwidget-c4b80160865b2d3db1f6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c4b80160865b2d3db1f6">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376"],["CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3"],["ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17","ACVR2A","APBB2","APC","AR","ARNT","ATM","ATXN2","BIN1","CASZ1","CDK12","CDKN1B","COL11A1","CPLX4","CPNE3","CTNNB1","DEPDC1B","DHX30","ERF","ETV3","FOXA1","HLA-DRB1","IL6ST","JAK1","KDM6A","KRT85","LCE2D","LCE4A","NCOR1","NKX3-1","NUDT22","PGLS","PIK3R1","PRPF19","PTEN","RB1","RNF43","RTKN","SKA3","SLC45A3","SPOP","SYPL1","TMPRSS2","TP53","USP28","ZFHX3","ZMYM3","ZNF17"],[923,null,923,923,null,923,null,null,923,923,923,923,null,null,923,null,923,923,923,923,923,923,923,923,null,null,null,923,null,null,null,null,null,923,923,923,null,null,923,923,null,923,923,923,923,923,null,923,null,923,923,null,923,null,null,923,923,923,923,null,null,923,null,923,923,923,923,923,923,923,923,null,null,null,923,null,null,null,null,null,923,923,923,null,null,923,923,null,923,923,923,923,923,null,923,null,923,923,null,923,null,null,923,923,923,923,null,null,923,null,923,923,923,923,923,923,923,923,null,null,null,923,null,null,null,null,null,923,923,923,null,null,923,923,null,923,923,923,923,923,null,923,null,923,923,null,923,null,null,923,923,923,923,null,null,923,null,923,923,923,923,923,923,923,923,null,null,null,923,null,null,null,null,null,923,923,923,null,null,923,923,null,923,923,923,923,923,null,923,null,923,923,null,923,null,null,923,923,923,923,null,null,923,null,923,923,923,923,923,923,923,923,null,null,null,923,null,null,null,null,null,923,923,923,null,null,923,923,null,923,923,923,923,923,null,959,null,959,959,null,959,null,null,959,959,959,959,null,null,959,null,959,959,959,959,959,959,959,959,null,null,null,959,null,null,null,null,null,959,959,959,null,null,959,959,null,959,959,959,959,959,null,959,null,959,959,null,959,null,null,959,959,959,959,null,null,959,null,959,959,959,959,959,959,959,959,null,null,null,959,null,null,null,null,null,959,959,959,null,null,959,959,null,959,959,959,959,959,null,959,null,959,959,null,959,null,null,959,959,959,959,null,null,959,null,959,959,959,959,959,959,959,959,null,null,null,959,null,null,null,null,null,959,959,959,null,null,959,959,null,959,959,959,959,959,null],[10.5,null,2.42,5.63,null,0.669,null,null,-3.55,3.22,0.732,3.43,null,null,1.04,null,3.12,-1.2,-2.08,0.786,4.35,-0.162,-0.615,-0.13,null,null,null,3.17,null,null,null,null,null,1.97,7.82,-0.346,null,null,10.7,-2.67,null,-3.19,5.31,5.66,0.114,2.25,null,4.72,null,9,29,null,5.66,null,null,-3.29,60.7,-1.08,23.4,null,null,17.9,null,-10.6,5.89,6.55,7.61,-4.32,16,6.08,-1.06,null,null,null,12.4,null,null,null,null,null,8.21,49.1,6.76,null,null,6.39,-14,null,-10.8,18.2,27.1,10.6,0.0651,null,0.881,null,4.98,9.58,null,3.88,null,null,-8.77,22.3,-13.1,5.62,null,null,5.83,null,-13.3,-12.3,4.16,4.32,-5.6,-17,-12,-2.09,null,null,null,12,null,null,null,null,null,-3.64,11.8,-14.1,null,null,17.6,-11.8,null,-12.7,26.8,16.9,12.6,-3.15,null,4.76,null,17.9,28,null,10.9,null,null,10.8,15.6,17,12.8,null,null,8.41,null,4.4,11.2,11.2,8.21,15,39.4,12.3,-0.49,null,null,null,11,null,null,null,null,null,10.4,16.7,10.9,null,null,8.32,6.46,null,12,-2.26,6.61,-1.95,6.66,null,-1.39,null,-3.76,-6.74,null,-2.44,null,null,0.166,-7.17,-0.325,-3.49,null,null,-2.89,null,1.51,0.377,-2.53,-2.1,-1.65,-3.1,0.0813,0.446,null,null,null,-4.29,null,null,null,null,null,-1.23,-6.22,0.34,null,null,-5.57,1.57,null,0.967,-4.91,-4.48,-1.99,-0.663,null,9.28,null,22,34.2,null,8.16,null,null,2.84,17.8,14.8,8.68,null,null,15.5,null,0.638,-0.375,15.9,7.11,-5.93,1.57,39.9,18.3,null,null,null,28,null,null,null,null,null,-3.95,23.5,17.4,null,null,3.61,-0.184,null,30.9,11.4,57.4,43,6.53,null,126,null,11.2,22.5,null,8.86,null,null,449,-0.326,46.4,279,null,null,257,null,483,737,38.2,2.21,-13.6,47.2,619,312,null,null,null,60,null,null,null,null,null,-4.13,43.1,762,null,null,-11.8,2.33,null,50.1,44.4,604,226,347,null,267,null,22.4,26.8,null,13.9,null,null,219,2.09,86.1,137,null,null,102,null,291,451,39.4,3.31,-8.97,46.4,413,133,null,null,null,93.6,null,null,null,null,null,20.8,111,458,null,null,-10.1,14.1,null,44,24.6,244,108,200,null],[0.957757009345794,null,0.336,0.000103629473684211,null,0.773925925925926,null,null,0.794391752577319,0.228,0.83776,0.979612903225806,null,null,0.284928,null,0.987221719457014,0.579218934911243,0.88947572815534,0.516711656441718,0.3136,0.515753086419753,0.773925925925926,0.47493670886076,null,null,null,0.100668235294118,null,null,null,null,null,0.0949975903614458,0.001533,0.809583756345178,null,null,0.0244954838709677,0.0143111111111111,null,0.381572413793103,3.68581818181818e-07,0.133644943820225,0.617581920903955,0.548460606060606,null,0.38717880794702,null,0.00917333333333333,1.2832e-07,null,0.197714285714286,null,null,0.875034482758621,0.000135466666666667,0.902339622641509,0.0228039344262295,null,null,0.340869565217391,null,0.987221719457014,0.439329032258064,0.341640287769784,0.243145299145299,0.231791304347826,0.20032,0.340869565217391,0.856079207920792,null,null,null,0.0578430379746835,null,null,null,null,null,0.386630136986301,8.90830769230769e-06,0.551903614457831,null,null,0.243145299145299,0.0106971428571429,null,0.99142600896861,2.2736e-10,0.20032,0.38717880794702,0.773925925925926,null,0.368666666666667,null,0.339294117647059,0.418909090909091,null,0.47368152866242,null,null,0.117664367816092,0.00236218181818182,0.135395555555556,0.876235294117647,null,null,0.333473684210526,null,0.20032,0.20032,0.809583756345178,0.35109219858156,0.896,0.285732283464567,0.142921739130435,0.776376963350785,null,null,null,0.198432323232323,null,null,null,null,null,0.630471910112359,0.264393442622951,0.1729,null,null,0.20032,0.117664367816092,null,0.586352941176471,1.66506666666667e-18,0.351774647887324,0.273170731707317,0.768173913043478,null,0.971703703703704,null,0.000497777777777778,0.0006176,null,0.215779816513761,null,null,0.399368421052632,0.261333333333333,0.0214666666666667,0.142921739130435,null,null,0.579218934911243,null,0.264393442622951,0.231791304347826,0.146683870967742,0.0316315151515152,0.0949975903614458,0.000989935483870968,0.198912,0.990846846846847,null,null,null,0.128036363636364,null,null,null,null,null,0.312558139534884,0.0578430379746835,0.224,null,null,0.284928,0.054656,null,0.38717880794702,0.0487890410958904,0.896,0.850308457711443,0.3424,null,0.724923076923077,null,0.0027584,2.3968e-07,null,0.0476,null,null,0.987221719457014,1.80693333333333e-07,0.794391752577319,0.0415536231884058,null,null,0.0588,null,0.579218934911243,0.878517073170732,0.215703703703704,0.0182,0.458051282051282,0.231791304347826,0.987221719457014,0.617581920903955,null,null,null,0.0196579310344828,null,null,null,null,null,0.323175572519084,0.000163927272727273,0.896,null,null,0.0316315151515152,0.252235294117647,null,0.773925925925926,1.2432e-18,0.149412765957447,0.29225,0.724923076923077,null,0.38717880794702,null,1.624e-05,1.80693333333333e-07,null,0.0068348717948718,null,null,0.809583756345178,4.96533333333333e-07,0.285732283464567,0.0416,null,null,0.0496432432432432,null,0.586853801169591,0.773925925925926,0.250576271186441,8.32533333333333e-05,0.957757009345794,0.5026,0.197714285714286,0.339294117647059,null,null,null,0.0143111111111111,null,null,null,null,null,0.645866666666667,0.00922046511627907,0.971703703703704,null,null,0.0754962962962963,0.0122769230769231,null,0.203290566037736,1.2544e-10,0.00307545945945946,0.0006496,0.617581920903955,null,0.0101773913043478,null,0.00274729411764706,1.42016e-05,null,0.0214666666666667,null,null,0.896,0.00010584,0.327515151515152,0.0193740350877193,null,null,0.0101773913043478,null,0.1729,0.0103897872340426,0.617581920903955,0.000184069565217391,0.477584905660377,0.817206030150754,2.14776470588235e-05,0.033265671641791,null,null,null,0.028525,null,null,null,null,null,0.228,0.000497777777777778,0.00028,null,null,0.635709497206704,1,null,0.100668235294118,1.2544e-10,0.0108864,0.0122769230769231,0.768173913043478,null,0.002856,null,0.0144174545454545,0.0578430379746835,null,0.617302325581395,null,null,0.415790849673203,0.00519326315789474,0.514782608695652,0.0415536231884058,null,null,0.0578430379746835,null,0.0270222222222222,0.0101773913043478,0.0473239436619718,0.617581920903955,0.529951219512195,0.364979020979021,0.00737560975609756,0.794391752577319,null,null,null,0.0106971428571429,null,null,null,null,null,0.896,1.1312e-05,0.000404992,null,null,0.816808080808081,0.38717880794702,null,0.776376963350785,1.08192e-20,0.208508411214953,0.007336,0.00071904,null],["ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>signature<\/th>\n      <th>feature<\/th>\n      <th>count<\/th>\n      <th>measure<\/th>\n      <th>p<\/th>\n      <th>type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="signatures-mutated-pathways" class="section level3">
<h3>Signatures &amp; mutated pathways</h3>
<p>Analyze the association between signatures and mutated pathways.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1">tidy_data.seqz.pathways &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</a>
<a class="sourceLine" id="cb74-2" data-line-number="2">  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb74-3" data-line-number="3">  <span class="dt">cols_to_features =</span> cols_to_mutated_pathways,</a>
<a class="sourceLine" id="cb74-4" data-line-number="4">  <span class="dt">type =</span> <span class="st">&quot;ca&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb74-5" data-line-number="5">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb74-6" data-line-number="6"><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb74-7" data-line-number="7"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb74-8" data-line-number="8"><span class="co">#&gt; -&gt; Calculating correlations for categorical variables using wilcox.test...</span></a>
<a class="sourceLine" id="cb74-9" data-line-number="9"><span class="co">#&gt; --&gt; building test models, be patient...</span></a>
<a class="sourceLine" id="cb74-10" data-line-number="10"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb74-11" data-line-number="11"><span class="co">#&gt; --&gt; obtaining model results...</span></a>
<a class="sourceLine" id="cb74-12" data-line-number="12"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb74-13" data-line-number="13"><span class="co">#&gt; --&gt; collecting data...</span></a>
<a class="sourceLine" id="cb74-14" data-line-number="14"><span class="co">#&gt; --&gt; done</span></a>
<a class="sourceLine" id="cb74-15" data-line-number="15"><span class="co">#&gt; -&gt; Done</span></a></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" data-line-number="1">sum_pathways &lt;-<span class="st"> </span>tidy_data.seqz.pathways <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb75-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(p <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb75-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(</a>
<a class="sourceLine" id="cb75-4" data-line-number="4">    <span class="dt">min =</span> <span class="kw">min</span>(measure, <span class="dt">na.rm =</span> T),</a>
<a class="sourceLine" id="cb75-5" data-line-number="5">    <span class="dt">max =</span> <span class="kw">max</span>(measure, <span class="dt">na.rm =</span> T),</a>
<a class="sourceLine" id="cb75-6" data-line-number="6">    <span class="dt">sum =</span> <span class="kw">list</span>(<span class="kw">summary</span>(measure))</a>
<a class="sourceLine" id="cb75-7" data-line-number="7">  )</a>
<a class="sourceLine" id="cb75-8" data-line-number="8"></a>
<a class="sourceLine" id="cb75-9" data-line-number="9">sum_pathways<span class="op">$</span>sum</a>
<a class="sourceLine" id="cb75-10" data-line-number="10"><span class="co">#&gt; [[1]]</span></a>
<a class="sourceLine" id="cb75-11" data-line-number="11"><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></a>
<a class="sourceLine" id="cb75-12" data-line-number="12"><span class="co">#&gt;   -4.68    7.83   18.00   52.73   56.70  531.00</span></a></code></pre></div>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb76-1" data-line-number="1">sc &lt;-<span class="st"> </span><span class="kw">scale_colour_gradientn</span>(</a>
<a class="sourceLine" id="cb76-2" data-line-number="2">  <span class="dt">colors =</span> <span class="kw">c</span>(<span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;red&quot;</span>),</a>
<a class="sourceLine" id="cb76-3" data-line-number="3">  <span class="dt">values =</span> scales<span class="op">::</span><span class="kw">rescale</span>(<span class="kw">c</span>(</a>
<a class="sourceLine" id="cb76-4" data-line-number="4">    sum_pathways<span class="op">$</span>min,</a>
<a class="sourceLine" id="cb76-5" data-line-number="5">    <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb76-6" data-line-number="6">    <span class="dv">20</span>,</a>
<a class="sourceLine" id="cb76-7" data-line-number="7">    sum_pathways<span class="op">$</span>max</a>
<a class="sourceLine" id="cb76-8" data-line-number="8">  ))</a>
<a class="sourceLine" id="cb76-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb76-10" data-line-number="10"></a>
<a class="sourceLine" id="cb76-11" data-line-number="11"></a>
<a class="sourceLine" id="cb76-12" data-line-number="12"><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.pathways,</a>
<a class="sourceLine" id="cb76-13" data-line-number="13">  <span class="dt">ylab =</span> <span class="st">&quot;Mutated pathways&quot;</span>,</a>
<a class="sourceLine" id="cb76-14" data-line-number="14">  <span class="dt">ca_gradient_colors =</span> sc</a>
<a class="sourceLine" id="cb76-15" data-line-number="15">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-62-1.png" width="2700" /></p>
<p>The result table:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb77-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(tidy_data.seqz.pathways)</a></code></pre></div>
<div id="htmlwidget-482476e14329fbf6a513" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-482476e14329fbf6a513">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88"],["CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig1","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig2","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig3","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig4","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","CN_Sig5","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig1","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig2","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3","SBS_Sig3"],["Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT","Cell_Cycle","Hippo","HR_pathway","MYC","NOTCH","NRF2","PI3K","RTK-RAS","TGF-Beta","TP53_pathway","WNT"],[923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,923,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959,959],[3.35,2.58,1.88,0.758,2.76,-1.62,1.57,1.85,2.46,4.5,1.58,25.7,17.2,12.1,8.12,12.8,24.8,4.93,15,-3.82,16.8,14.1,7.02,8.16,7.83,12.4,7.9,-7.02,2.26,8.62,-4.35,23.5,8.74,10.9,5.44,7.28,1.53,7.49,3.33,9,3.77,9.53,0.364,8.04,-3.51,-2.66,-2.8,-2.55,-2.85,0.0224,-2,-2.48,-0.631,-4.68,-3,18.9,18.6,27,39.4,23.6,20.9,13.3,18.2,35.1,11.3,18,173,60.8,114,323,67.5,531,78.5,56.7,297,39,59.8,144,55,73.4,148,59.2,262,67.1,51.2,197,25.9,53.8],[0.0228148148148148,0.0379866666666667,0.142535211267606,0.242,0.0669935483870968,0.756380952380952,0.0846666666666667,0.126089552238806,0.859770114942529,3.21391304347826e-07,0.0083330612244898,0.000378888888888889,5.78285714285714e-06,0.0270175438596491,0.130705882352941,7.65333333333333e-05,0.782790697674419,0.00438044444444444,1.8392e-05,0.608487804878049,5.56285714285714e-11,0.000377142857142857,0.601876543209877,0.081125,0.0416918032786885,0.0846666666666667,0.00926588235294118,0.39825641025641,0.756380952380952,0.000426461538461538,0.237013333333333,1.6368e-17,0.00550468085106383,0.0359457627118644,0.0286758620689655,0.142535211267606,0.931,0.00438044444444444,0.193837837837838,0.00421581395348837,0.345142857142857,0.142535211267606,0.188054794520548,0.00760833333333333,0.000426461538461538,5.85655172413793e-06,0.000426461538461538,0.081125,2.948e-06,0.782790697674419,0.00156933333333333,5.24740740740741e-06,0.601876543209877,1.0956e-19,1.1484e-07,3.20774193548387e-05,4.54892307692308e-11,2.832e-11,0.0106276923076923,4.95e-15,0.406582278481013,0.000106117647058824,4.60533333333333e-12,0.0240428571428571,1.21691428571429e-12,9.3808e-15,1.11936e-06,4.49533333333333e-11,2.68631578947368e-08,0.00127921951219512,5.027e-10,0.00550468085106383,7.7e-07,9.83529411764706e-10,0.0126520754716981,4.98666666666667e-14,4.532e-12,1.21104761904762e-07,1.1792e-11,0.02336,0.176,3.21391304347826e-07,0.0087648,0.0010648,7.97866666666667e-11,4.7025e-05,1.0956e-19,3.92088888888889e-09],["ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca","ca"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>signature<\/th>\n      <th>feature<\/th>\n      <th>count<\/th>\n      <th>measure<\/th>\n      <th>p<\/th>\n      <th>type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="correlation-network" class="section level3">
<h3>Correlation network</h3>
<p>To determine the structure of relationship between signatures and features, we use continuous variables to do correlation network analysis.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1">f_dt &lt;-<span class="st"> </span>df.seqz[, <span class="kw">c</span>(cols_to_sigs.seqz, cols_to_features[<span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">-2</span>)])] <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb78-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.ordered, as.numeric)</a></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb79-1" data-line-number="1"><span class="kw">library</span>(corrr)</a>
<a class="sourceLine" id="cb79-2" data-line-number="2"></a>
<a class="sourceLine" id="cb79-3" data-line-number="3">res.cor &lt;-<span class="st"> </span><span class="kw">correlate</span>(f_dt,</a>
<a class="sourceLine" id="cb79-4" data-line-number="4">  <span class="dt">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>,</a>
<a class="sourceLine" id="cb79-5" data-line-number="5">  <span class="dt">method =</span> <span class="st">&quot;pearson&quot;</span></a>
<a class="sourceLine" id="cb79-6" data-line-number="6">) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb79-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, dplyr<span class="op">::</span>coalesce, <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb79-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb79-9" data-line-number="9"><span class="co">#&gt; Correlation method: &#39;pearson&#39;</span></a>
<a class="sourceLine" id="cb79-10" data-line-number="10"><span class="co">#&gt; Missing treated using: &#39;pairwise.complete.obs&#39;</span></a>
<a class="sourceLine" id="cb79-11" data-line-number="11"></a>
<a class="sourceLine" id="cb79-12" data-line-number="12"><span class="co">## Avoid corrr error</span></a>
<a class="sourceLine" id="cb79-13" data-line-number="13"><span class="co">## https://github.com/tidymodels/corrr/issues/86</span></a>
<a class="sourceLine" id="cb79-14" data-line-number="14"><span class="co">## https://github.com/tidymodels/corrr/issues/34</span></a>
<a class="sourceLine" id="cb79-15" data-line-number="15">res.cor[, <span class="dv">-1</span>] &lt;-<span class="st"> </span>res.cor[, <span class="dv">-1</span>] <span class="op">*</span><span class="st"> </span><span class="fl">0.99</span></a>
<a class="sourceLine" id="cb79-16" data-line-number="16"></a>
<a class="sourceLine" id="cb79-17" data-line-number="17">p &lt;-<span class="st"> </span>res.cor <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb79-18" data-line-number="18"><span class="st">  </span><span class="kw">network_plot</span>(</a>
<a class="sourceLine" id="cb79-19" data-line-number="19">    <span class="dt">min_cor =</span> <span class="fl">0.2</span>,</a>
<a class="sourceLine" id="cb79-20" data-line-number="20">    <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;indianred2&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;skyblue1&quot;</span>))</a>
<a class="sourceLine" id="cb79-21" data-line-number="21">  )</a>
<a class="sourceLine" id="cb79-22" data-line-number="22"></a>
<a class="sourceLine" id="cb79-23" data-line-number="23"><span class="co"># png(&quot;corrr_network.png&quot;, width = 7, height = 7, res = 300)</span></a>
<a class="sourceLine" id="cb79-24" data-line-number="24">p <span class="op">+</span><span class="st"> </span><span class="kw">scale_size</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>))</a>
<a class="sourceLine" id="cb79-25" data-line-number="25"><span class="co">#&gt; Scale for &#39;size&#39; is already present. Adding another scale for &#39;size&#39;, which</span></a>
<a class="sourceLine" id="cb79-26" data-line-number="26"><span class="co">#&gt; will replace the existing scale.</span></a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-65-1.png" width="2400" /></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb80-1" data-line-number="1"><span class="co"># ggsave(&quot;corrr_network.pdf&quot;, plot = p + scale_size(range=c(0.1, 2)))</span></a>
<a class="sourceLine" id="cb80-2" data-line-number="2"><span class="co"># dev.off()</span></a></code></pre></div>
<p><del><a href="corrr_network.pdf">click here to see the plot</a></del></p>
<blockquote>
<p>variables that are more highly correlated appear closer together and are joined by stronger paths. Paths are also colored by their sign. The proximity of the points are determined using multidimensional clustering.</p>
</blockquote>
</div>
<div id="signatures-psa" class="section level3">
<h3>Signatures &amp; PSA</h3>
<p>In all datasets we collected, 3 of them have PSA info. However, the range of PSA value in 3 datasets differs, so I excluded PSA from features above and analyze it independently.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb81-1" data-line-number="1">df.seqz.psa &lt;-<span class="st"> </span>df.seqz <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(PSA)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="kw">contains</span>(<span class="st">&quot;Sig&quot;</span>, <span class="dt">ignore.case =</span> F), <span class="kw">c</span>(<span class="st">&quot;PSA&quot;</span>, <span class="st">&quot;Study&quot;</span>, <span class="st">&quot;CNV_ID&quot;</span>))</a>
<a class="sourceLine" id="cb81-4" data-line-number="4"></a>
<a class="sourceLine" id="cb81-5" data-line-number="5">df.seqz.psa &lt;-<span class="st"> </span>df.seqz.psa <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">:::</span><span class="kw">group_by</span>(Study) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-7" data-line-number="7"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">nest</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(</a>
<a class="sourceLine" id="cb81-9" data-line-number="9">    <span class="dt">data =</span> purrr<span class="op">:::</span><span class="kw">map</span>(data, <span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb81-10" data-line-number="10">      <span class="kw">get_sig_feature_association</span>(x,</a>
<a class="sourceLine" id="cb81-11" data-line-number="11">        <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb81-12" data-line-number="12">        <span class="dt">cols_to_features =</span> <span class="st">&quot;PSA&quot;</span>,</a>
<a class="sourceLine" id="cb81-13" data-line-number="13">        <span class="dt">method_co =</span> <span class="st">&quot;pearson&quot;</span>,</a>
<a class="sourceLine" id="cb81-14" data-line-number="14">        <span class="dt">type =</span> <span class="st">&quot;co&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb81-15" data-line-number="15">      ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-16" data-line-number="16"><span class="st">        </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb81-17" data-line-number="17">    })</a>
<a class="sourceLine" id="cb81-18" data-line-number="18">  ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-19" data-line-number="19"><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unnest</span>(<span class="st">&quot;data&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-20" data-line-number="20"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(feature <span class="op">==</span><span class="st"> &quot;PSA&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb81-21" data-line-number="21"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">feature =</span> Study)</a>
<a class="sourceLine" id="cb81-22" data-line-number="22"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb81-23" data-line-number="23"><span class="co">#&gt; -&gt; Calculating correlations for continuous variables using pearson...</span></a>
<a class="sourceLine" id="cb81-24" data-line-number="24"><span class="co">#&gt; -&gt; Done</span></a>
<a class="sourceLine" id="cb81-25" data-line-number="25"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb81-26" data-line-number="26"><span class="co">#&gt; -&gt; Calculating correlations for continuous variables using pearson...</span></a>
<a class="sourceLine" id="cb81-27" data-line-number="27"><span class="co">#&gt; -&gt; Done</span></a>
<a class="sourceLine" id="cb81-28" data-line-number="28"><span class="co">#&gt; -&gt; Detecting and transforming possibly ordinal variables...</span></a>
<a class="sourceLine" id="cb81-29" data-line-number="29"><span class="co">#&gt; -&gt; Calculating correlations for continuous variables using pearson...</span></a>
<a class="sourceLine" id="cb81-30" data-line-number="30"><span class="co">#&gt; -&gt; Done</span></a></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb82-1" data-line-number="1"><span class="kw">show_sig_feature_corrplot</span>(df.seqz.psa, <span class="dt">p_val =</span> <span class="dv">1</span>, <span class="dt">breaks_count =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">150</span>, <span class="dv">200</span>), <span class="dt">ylab =</span> <span class="st">&quot;PSA&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-67-1.png" width="2700" /></p>
<p>Alought we can observe the statistically significant correlation between signatures and PSA, the results are inconsistent across different studies. This may be explained by the dynamic property of PSA while not the sample type due to most of them are primary.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb83-1" data-line-number="1"><span class="kw">table</span>(df.seqz<span class="op">$</span>sample_type, df.seqz<span class="op">$</span>Study)</a>
<a class="sourceLine" id="cb83-2" data-line-number="2"><span class="co">#&gt;             </span></a>
<a class="sourceLine" id="cb83-3" data-line-number="3"><span class="co">#&gt;              phs000447 phs000554 phs000909 phs000915 phs001141 TCGA</span></a>
<a class="sourceLine" id="cb83-4" data-line-number="4"><span class="co">#&gt;   Metastatic         0        50        70       147        77    1</span></a>
<a class="sourceLine" id="cb83-5" data-line-number="5"><span class="co">#&gt;   Primary          112        11        28         0         0  500</span></a></code></pre></div>
</div>
</div>
</div>
<div id="group-analysis" class="section level1">
<h1>Group Analysis</h1>
<p>This part has been divided into four parts to show the differences across sample groups defined by signature exposure. It provides insight into the heterogeneity of prostate cancers and relative importance of two types of genomic variation signatures.</p>
<p>Load packages.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb84-1" data-line-number="1"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb84-2" data-line-number="2"><span class="kw">library</span>(sigminer)</a></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb85-1" data-line-number="1">df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</a></code></pre></div>
<div id="group-comparison" class="section level2">
<h2>Group comparison</h2>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb86-1" data-line-number="1">cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb86-2" data-line-number="2">  <span class="st">&quot;IsMetastatic&quot;</span>, <span class="st">&quot;HasFusion&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</a>
<a class="sourceLine" id="cb86-3" data-line-number="3">  <span class="st">&quot;Age&quot;</span>,</a>
<a class="sourceLine" id="cb86-4" data-line-number="4">  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;INDELs&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</a>
<a class="sourceLine" id="cb86-5" data-line-number="5">  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</a>
<a class="sourceLine" id="cb86-6" data-line-number="6">  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></a>
<a class="sourceLine" id="cb86-7" data-line-number="7">)</a>
<a class="sourceLine" id="cb86-8" data-line-number="8"></a>
<a class="sourceLine" id="cb86-9" data-line-number="9">feature_type &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;ca&quot;</span>, <span class="dv">4</span>), <span class="kw">rep</span>(<span class="st">&quot;co&quot;</span>, <span class="dv">15</span>))</a></code></pre></div>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb87-1" data-line-number="1">df.seqz<span class="op">$</span>GleasonScore &lt;-<span class="st"> </span><span class="kw">as.character</span>(df.seqz<span class="op">$</span>GleasonScore)</a>
<a class="sourceLine" id="cb87-2" data-line-number="2">df.seqz<span class="op">$</span>GleasonScore &lt;-<span class="st"> </span><span class="kw">factor</span>(df.seqz<span class="op">$</span>GleasonScore, <span class="dt">levels =</span> <span class="kw">as.character</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">10</span>))</a>
<a class="sourceLine" id="cb87-3" data-line-number="3"></a>
<a class="sourceLine" id="cb87-4" data-line-number="4">df.seqz2 &lt;-<span class="st"> </span>df.seqz</a>
<a class="sourceLine" id="cb87-5" data-line-number="5">df.seqz2<span class="op">$</span>cnv_enrich_sig &lt;-<span class="st"> </span><span class="kw">ifelse</span>(</a>
<a class="sourceLine" id="cb87-6" data-line-number="6">  <span class="op">!</span><span class="kw">is.na</span>(df.seqz2<span class="op">$</span>cnv_enrich_sig),</a>
<a class="sourceLine" id="cb87-7" data-line-number="7">  <span class="kw">paste</span>(df.seqz2<span class="op">$</span>cnv_enrich_sig, <span class="st">&quot;enriched&quot;</span>),</a>
<a class="sourceLine" id="cb87-8" data-line-number="8">  <span class="ot">NA</span></a>
<a class="sourceLine" id="cb87-9" data-line-number="9">)</a>
<a class="sourceLine" id="cb87-10" data-line-number="10"></a>
<a class="sourceLine" id="cb87-11" data-line-number="11">grp.cnv &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(</a>
<a class="sourceLine" id="cb87-12" data-line-number="12">  df.seqz2,</a>
<a class="sourceLine" id="cb87-13" data-line-number="13">  <span class="dt">col_group =</span> <span class="st">&quot;cnv_enrich_sig&quot;</span>, <span class="co"># or &quot;cnv_group&quot;</span></a>
<a class="sourceLine" id="cb87-14" data-line-number="14">  <span class="dt">cols_to_compare =</span> cols_to_features,</a>
<a class="sourceLine" id="cb87-15" data-line-number="15">  <span class="dt">type =</span> feature_type</a>
<a class="sourceLine" id="cb87-16" data-line-number="16">)</a></code></pre></div>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb88-1" data-line-number="1">plot.cnv &lt;-<span class="st"> </span><span class="kw">show_group_comparison</span>(</a>
<a class="sourceLine" id="cb88-2" data-line-number="2">  grp.cnv,</a>
<a class="sourceLine" id="cb88-3" data-line-number="3">  <span class="dt">xlab =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb88-4" data-line-number="4">  <span class="dt">method =</span> <span class="st">&quot;anova&quot;</span>,</a>
<a class="sourceLine" id="cb88-5" data-line-number="5">  <span class="dt">text_angle_x =</span> <span class="dv">60</span>,</a>
<a class="sourceLine" id="cb88-6" data-line-number="6">  <span class="dt">text_hjust_x =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb88-7" data-line-number="7">  <span class="dt">legend_position_ca =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb88-8" data-line-number="8">  <span class="dt">label.x.npc =</span> <span class="st">&quot;center&quot;</span>,</a>
<a class="sourceLine" id="cb88-9" data-line-number="9">  <span class="dt">label.y.npc =</span> <span class="st">&quot;top&quot;</span>,</a>
<a class="sourceLine" id="cb88-10" data-line-number="10">  <span class="dt">label =</span> <span class="st">&quot;p.format&quot;</span>,</a>
<a class="sourceLine" id="cb88-11" data-line-number="11">  <span class="dt">hjust =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb88-12" data-line-number="12">)</a></code></pre></div>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb89-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plot.cnv<span class="op">$</span>ca, <span class="dt">align =</span> <span class="st">&quot;hv&quot;</span>, <span class="dt">ncol =</span> <span class="dv">2</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-74-1.png" width="2700" /></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb90-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plot.cnv<span class="op">$</span>co, <span class="dt">align =</span> <span class="st">&quot;hv&quot;</span>, <span class="dt">ncol =</span> <span class="dv">3</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-75-1.png" width="3000" /></p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb91-1" data-line-number="1">df.seqz2<span class="op">$</span>snv_enrich_sig &lt;-<span class="st"> </span><span class="kw">ifelse</span>(</a>
<a class="sourceLine" id="cb91-2" data-line-number="2">  <span class="op">!</span><span class="kw">is.na</span>(df.seqz2<span class="op">$</span>snv_enrich_sig),</a>
<a class="sourceLine" id="cb91-3" data-line-number="3">  <span class="kw">paste</span>(df.seqz2<span class="op">$</span>snv_enrich_sig, <span class="st">&quot;enriched&quot;</span>),</a>
<a class="sourceLine" id="cb91-4" data-line-number="4">  <span class="ot">NA</span></a>
<a class="sourceLine" id="cb91-5" data-line-number="5">)</a>
<a class="sourceLine" id="cb91-6" data-line-number="6"></a>
<a class="sourceLine" id="cb91-7" data-line-number="7">grp.snv &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(</a>
<a class="sourceLine" id="cb91-8" data-line-number="8">  df.seqz2,</a>
<a class="sourceLine" id="cb91-9" data-line-number="9">  <span class="dt">col_group =</span> <span class="st">&quot;snv_enrich_sig&quot;</span>, <span class="co"># or &quot;snv_group&quot;</span></a>
<a class="sourceLine" id="cb91-10" data-line-number="10">  <span class="dt">cols_to_compare =</span> cols_to_features,</a>
<a class="sourceLine" id="cb91-11" data-line-number="11">  <span class="dt">type =</span> feature_type</a>
<a class="sourceLine" id="cb91-12" data-line-number="12">)</a></code></pre></div>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb92-1" data-line-number="1">plot.snv &lt;-<span class="st"> </span><span class="kw">show_group_comparison</span>(</a>
<a class="sourceLine" id="cb92-2" data-line-number="2">  grp.snv,</a>
<a class="sourceLine" id="cb92-3" data-line-number="3">  <span class="dt">xlab =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb92-4" data-line-number="4">  <span class="dt">method =</span> <span class="st">&quot;anova&quot;</span>,</a>
<a class="sourceLine" id="cb92-5" data-line-number="5">  <span class="dt">text_angle_x =</span> <span class="dv">60</span>,</a>
<a class="sourceLine" id="cb92-6" data-line-number="6">  <span class="dt">text_hjust_x =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb92-7" data-line-number="7">  <span class="dt">legend_position_ca =</span> <span class="st">&quot;right&quot;</span>,</a>
<a class="sourceLine" id="cb92-8" data-line-number="8">  <span class="dt">label.x.npc =</span> <span class="st">&quot;center&quot;</span>,</a>
<a class="sourceLine" id="cb92-9" data-line-number="9">  <span class="dt">label.y.npc =</span> <span class="st">&quot;top&quot;</span>,</a>
<a class="sourceLine" id="cb92-10" data-line-number="10">  <span class="dt">label =</span> <span class="st">&quot;p.format&quot;</span>,</a>
<a class="sourceLine" id="cb92-11" data-line-number="11">  <span class="dt">hjust =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb92-12" data-line-number="12">)</a></code></pre></div>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb93-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plot.snv<span class="op">$</span>ca, <span class="dt">align =</span> <span class="st">&quot;hv&quot;</span>, <span class="dt">ncol =</span> <span class="dv">2</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-78-1.png" width="2700" /></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb94-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> plot.snv<span class="op">$</span>co, <span class="dt">align =</span> <span class="st">&quot;hv&quot;</span>, <span class="dt">ncol =</span> <span class="dv">3</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-79-1.png" width="3000" /></p>
</div>
<div id="group-mapping" class="section level2">
<h2>Group mapping</h2>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb95-1" data-line-number="1">df.map &lt;-<span class="st"> </span>df.seqz[, <span class="kw">c</span>(</a>
<a class="sourceLine" id="cb95-2" data-line-number="2">  <span class="st">&quot;sample_type&quot;</span>,</a>
<a class="sourceLine" id="cb95-3" data-line-number="3">  <span class="st">&quot;cnv_enrich_sig&quot;</span>,</a>
<a class="sourceLine" id="cb95-4" data-line-number="4">  <span class="st">&quot;snv_enrich_sig&quot;</span></a>
<a class="sourceLine" id="cb95-5" data-line-number="5">)] <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb95-6" data-line-number="6"><span class="st">  </span><span class="kw">na.omit</span>()</a>
<a class="sourceLine" id="cb95-7" data-line-number="7"><span class="kw">names</span>(df.map) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Sample type&quot;</span>, <span class="st">&quot;Enriched CN signature&quot;</span>, <span class="st">&quot;Enriched SBS signature&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb96-1" data-line-number="1"><span class="kw">show_group_mapping</span>(df.map,</a>
<a class="sourceLine" id="cb96-2" data-line-number="2">  <span class="dt">col_to_flow =</span> <span class="st">&quot;Sample type&quot;</span>,</a>
<a class="sourceLine" id="cb96-3" data-line-number="3">  <span class="dt">cols_to_map =</span> <span class="kw">setdiff</span>(<span class="kw">colnames</span>(df.map), <span class="st">&quot;Sample type&quot;</span>),</a>
<a class="sourceLine" id="cb96-4" data-line-number="4">  <span class="dt">fill_na =</span> <span class="st">&quot;NA&quot;</span>, <span class="dt">include_sig =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb96-5" data-line-number="5">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-81-1.png" width="2400" /></p>
</div>
<div id="grouped-signature-exposure-profile" class="section level2">
<h2>Grouped signature exposure profile</h2>
<p>We have showed the exposure profiles, here we add group annotation to them.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb97-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a>
<a class="sourceLine" id="cb97-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a></code></pre></div>
<div id="copy-number-signatures" class="section level3">
<h3>Copy number signatures</h3>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb98-1" data-line-number="1">map_df &lt;-<span class="st"> </span>df.seqz <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb98-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(cnv_enrich_sig, CNV_ID) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb98-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(CNV_ID))</a>
<a class="sourceLine" id="cb98-4" data-line-number="4"></a>
<a class="sourceLine" id="cb98-5" data-line-number="5">groups &lt;-<span class="st"> </span>map_df<span class="op">$</span>cnv_enrich_sig</a>
<a class="sourceLine" id="cb98-6" data-line-number="6"><span class="kw">names</span>(groups) &lt;-<span class="st"> </span>map_df<span class="op">$</span>CNV_ID</a>
<a class="sourceLine" id="cb98-7" data-line-number="7"></a>
<a class="sourceLine" id="cb98-8" data-line-number="8"><span class="kw">show_sig_exposure</span>(Sig.CNV.seqz.W,</a>
<a class="sourceLine" id="cb98-9" data-line-number="9">  <span class="dt">groups =</span> groups,</a>
<a class="sourceLine" id="cb98-10" data-line-number="10">  <span class="co"># grp_order = ,</span></a>
<a class="sourceLine" id="cb98-11" data-line-number="11">  <span class="dt">rm_space =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb98-12" data-line-number="12">  <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb98-13" data-line-number="13">  <span class="dt">rm_grid_line =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb98-14" data-line-number="14">  <span class="dt">rm_panel_border =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb98-15" data-line-number="15">  <span class="dt">grp_size =</span> <span class="dv">7</span></a>
<a class="sourceLine" id="cb98-16" data-line-number="16">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-83-1.png" width="2400" /></p>
</div>
<div id="sbs-signatures" class="section level3">
<h3>SBS signatures</h3>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb99-1" data-line-number="1">map_df &lt;-<span class="st"> </span>df.seqz <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb99-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(tumor_Run, snv_enrich_sig) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb99-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(tumor_Run))</a>
<a class="sourceLine" id="cb99-4" data-line-number="4"></a>
<a class="sourceLine" id="cb99-5" data-line-number="5">groups &lt;-<span class="st"> </span>map_df<span class="op">$</span>snv_enrich_sig</a>
<a class="sourceLine" id="cb99-6" data-line-number="6"><span class="kw">names</span>(groups) &lt;-<span class="st"> </span>map_df<span class="op">$</span>tumor_Run</a>
<a class="sourceLine" id="cb99-7" data-line-number="7"></a>
<a class="sourceLine" id="cb99-8" data-line-number="8">Sig.SNV2 &lt;-<span class="st"> </span>Sig.SNV</a>
<a class="sourceLine" id="cb99-9" data-line-number="9"></a>
<a class="sourceLine" id="cb99-10" data-line-number="10">TCGA_INDEX &lt;-<span class="st"> </span><span class="kw">startsWith</span>(<span class="kw">colnames</span>(Sig.SNV2<span class="op">$</span>Exposure), <span class="st">&quot;TCGA&quot;</span>)</a>
<a class="sourceLine" id="cb99-11" data-line-number="11"><span class="kw">colnames</span>(Sig.SNV2<span class="op">$</span>Exposure)[TCGA_INDEX] &lt;-<span class="st"> </span><span class="kw">substr</span>(<span class="kw">colnames</span>(Sig.SNV2<span class="op">$</span>Exposure)[TCGA_INDEX], <span class="dv">1</span>, <span class="dv">15</span>)</a>
<a class="sourceLine" id="cb99-12" data-line-number="12"></a>
<a class="sourceLine" id="cb99-13" data-line-number="13"><span class="kw">show_sig_exposure</span>(Sig.SNV2,</a>
<a class="sourceLine" id="cb99-14" data-line-number="14">  <span class="dt">groups =</span> groups,</a>
<a class="sourceLine" id="cb99-15" data-line-number="15">  <span class="dt">cutoff =</span> <span class="dv">2000</span>,</a>
<a class="sourceLine" id="cb99-16" data-line-number="16">  <span class="dt">rm_space =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb99-17" data-line-number="17">  <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb99-18" data-line-number="18">  <span class="dt">rm_grid_line =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb99-19" data-line-number="19">  <span class="dt">rm_panel_border =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb99-20" data-line-number="20">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-84-1.png" width="2400" /></p>
</div>
</div>
<div id="copy-number-profile-of-different-groups" class="section level2">
<h2>Copy number profile of different groups</h2>
<p>There are many differences across copy number groups. Here we go further showing some classic cases by copy number profile.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb100-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</a>
<a class="sourceLine" id="cb100-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a>
<a class="sourceLine" id="cb100-3" data-line-number="3"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.tally.W.RData&quot;</span>)</a></code></pre></div>
<p>There are 5 copy number groups dominated by 4 copy number signatures.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb101-1" data-line-number="1"><span class="kw">table</span>(df.seqz<span class="op">$</span>cnv_group, df.seqz<span class="op">$</span>cnv_enrich_sig)</a>
<a class="sourceLine" id="cb101-2" data-line-number="2"><span class="co">#&gt;    </span></a>
<a class="sourceLine" id="cb101-3" data-line-number="3"><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5</span></a>
<a class="sourceLine" id="cb101-4" data-line-number="4"><span class="co">#&gt;   1   35    0    0    0    0</span></a>
<a class="sourceLine" id="cb101-5" data-line-number="5"><span class="co">#&gt;   2    0   63    0    0    0</span></a>
<a class="sourceLine" id="cb101-6" data-line-number="6"><span class="co">#&gt;   3    0    0  224    0    0</span></a>
<a class="sourceLine" id="cb101-7" data-line-number="7"><span class="co">#&gt;   4    0    0    0  170    0</span></a>
<a class="sourceLine" id="cb101-8" data-line-number="8"><span class="co">#&gt;   5    0    0    0    0  445</span></a></code></pre></div>
<p>Get the groups and relative exposures and merge them.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb102-1" data-line-number="1">cn_group &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.CNV.seqz.W, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb102-2" data-line-number="2"><span class="co">#&gt;    </span></a>
<a class="sourceLine" id="cb102-3" data-line-number="3"><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5</span></a>
<a class="sourceLine" id="cb102-4" data-line-number="4"><span class="co">#&gt;   1    9    2   14   10    0</span></a>
<a class="sourceLine" id="cb102-5" data-line-number="5"><span class="co">#&gt;   2    0   60    0    3    0</span></a>
<a class="sourceLine" id="cb102-6" data-line-number="6"><span class="co">#&gt;   3    0    3  219    2    0</span></a>
<a class="sourceLine" id="cb102-7" data-line-number="7"><span class="co">#&gt;   4    0    0    0  170    0</span></a>
<a class="sourceLine" id="cb102-8" data-line-number="8"><span class="co">#&gt;   5    0    3   35  200  207</span></a>
<a class="sourceLine" id="cb102-9" data-line-number="9">cn_group<span class="op">$</span>enrich_sig[cn_group<span class="op">$</span>group <span class="op">==</span><span class="st"> &quot;1&quot;</span>] &lt;-<span class="st"> &quot;Sig1&quot;</span></a>
<a class="sourceLine" id="cb102-10" data-line-number="10">cn_expo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.CNV.seqz.W, <span class="dt">type =</span> <span class="st">&quot;relative&quot;</span>)</a>
<a class="sourceLine" id="cb102-11" data-line-number="11"></a>
<a class="sourceLine" id="cb102-12" data-line-number="12">df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(cn_group, cn_expo)</a></code></pre></div>
<p>Now we can use data to find the cases with a specific signature enriched.</p>
<div id="sig1-cases" class="section level3">
<h3>Sig1 cases</h3>
<p>The group with enriched Sig1 is dominated by Sig3, we need to take case of it.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb103-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig1&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig1)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb103-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb103-6" data-line-number="6"></a>
<a class="sourceLine" id="cb103-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb103-8" data-line-number="8"><span class="co">#&gt; [1] &quot;5115056-SRR8311885&quot; &quot;TCGA-CH-5748-01&quot;    &quot;18-SRR5876729&quot;     </span></a>
<a class="sourceLine" id="cb103-9" data-line-number="9"><span class="co">#&gt; [4] &quot;5115276-SRR8312040&quot; &quot;3-SRR4044023&quot;       &quot;5115198-SRR8311993&quot;</span></a>
<a class="sourceLine" id="cb103-10" data-line-number="10"></a>
<a class="sourceLine" id="cb103-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb103-12" data-line-number="12">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb103-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb103-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-88-1.png" width="4200" /></p>
<p>Of note, only the segments with <strong>high copy number</strong> are contributed by Sig1.</p>
</div>
<div id="sig2-cases" class="section level3">
<h3>Sig2 cases</h3>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb104-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig2&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig2)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb104-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb104-6" data-line-number="6"></a>
<a class="sourceLine" id="cb104-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb104-8" data-line-number="8"><span class="co">#&gt; [1] &quot;5115615-SRR8311749&quot; &quot;5115406-SRR8312068&quot; &quot;45-SRR5876761&quot;     </span></a>
<a class="sourceLine" id="cb104-9" data-line-number="9"><span class="co">#&gt; [4] &quot;36-SRR5876749&quot;      &quot;1115153-SRR2404067&quot; &quot;5115152-SRR8311933&quot;</span></a>
<a class="sourceLine" id="cb104-10" data-line-number="10"></a>
<a class="sourceLine" id="cb104-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb104-12" data-line-number="12">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb104-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb104-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-89-1.png" width="4200" /></p>
</div>
<div id="sig3-cases" class="section level3">
<h3>Sig3 cases</h3>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb105-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb105-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig3&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb105-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig3)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb105-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb105-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb105-6" data-line-number="6"></a>
<a class="sourceLine" id="cb105-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb105-8" data-line-number="8"><span class="co">#&gt; [1] &quot;TCGA-HC-7745-01&quot;    &quot;1115082-SRR8304741&quot; &quot;TCGA-KK-A7B4-01&quot;   </span></a>
<a class="sourceLine" id="cb105-9" data-line-number="9"><span class="co">#&gt; [4] &quot;TCGA-ZG-A9L0-01&quot;    &quot;6115317-SRR8307505&quot; &quot;1115091-SRR8304759&quot;</span></a>
<a class="sourceLine" id="cb105-10" data-line-number="10"></a>
<a class="sourceLine" id="cb105-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb105-12" data-line-number="12">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb105-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb105-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-90-1.png" width="4200" /></p>
</div>
<div id="sig4-cases" class="section level3">
<h3>Sig4 cases</h3>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb106-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig4&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig4)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb106-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb106-6" data-line-number="6"></a>
<a class="sourceLine" id="cb106-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb106-8" data-line-number="8"><span class="co">#&gt; [1] &quot;TCGA-KK-A59X-01&quot;   &quot;06-1749-SRR472297&quot; &quot;TCGA-XK-AAJA-01&quot;  </span></a>
<a class="sourceLine" id="cb106-9" data-line-number="9"><span class="co">#&gt; [4] &quot;05-3852-SRR472039&quot; &quot;03-1426-SRR465855&quot; &quot;TCGA-J4-A67N-01&quot;</span></a>
<a class="sourceLine" id="cb106-10" data-line-number="10"></a>
<a class="sourceLine" id="cb106-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb106-12" data-line-number="12">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb106-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb106-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-91-1.png" width="4200" /></p>
</div>
<div id="sig5-cases" class="section level3">
<h3>Sig5 cases</h3>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb107-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig5&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig5)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb107-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb107-6" data-line-number="6"></a>
<a class="sourceLine" id="cb107-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb107-8" data-line-number="8"><span class="co">#&gt; [1] &quot;TCGA-KC-A7FE-01&quot;  &quot;TCGA-J4-AATV-01&quot;  &quot;TCGA-HI-7169-01&quot;  &quot;TCGA-FC-A66V-01&quot; </span></a>
<a class="sourceLine" id="cb107-9" data-line-number="9"><span class="co">#&gt; [5] &quot;TCGA-EJ-7791-01&quot;  &quot;09-120-SRR460104&quot;</span></a>
<a class="sourceLine" id="cb107-10" data-line-number="10"></a>
<a class="sourceLine" id="cb107-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb107-12" data-line-number="12">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb107-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb107-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-92-1.png" width="4200" /></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb108-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig5&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(<span class="kw">as.data.frame</span>(CNV.seqz.tally.W<span class="op">$</span>nmf_matrix) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-4" data-line-number="4"><span class="st">    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-5" data-line-number="5"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(sample, <span class="st">`</span><span class="dt">BPArm[2]</span><span class="st">`</span>, <span class="st">`</span><span class="dt">BPArm[1]</span><span class="st">`</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-6" data-line-number="6"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">BK =</span> <span class="st">`</span><span class="dt">BPArm[2]</span><span class="st">`</span> <span class="op">+</span><span class="st"> `</span><span class="dt">BPArm[1]</span><span class="st">`</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-7" data-line-number="7"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="kw">desc</span>(BK)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-8" data-line-number="8"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb108-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb108-10" data-line-number="10"></a>
<a class="sourceLine" id="cb108-11" data-line-number="11">samps_to_show</a>
<a class="sourceLine" id="cb108-12" data-line-number="12"><span class="co">#&gt; [1] &quot;TCGA-EJ-5515-01&quot;    &quot;32-SRR4043905&quot;      &quot;TCGA-EJ-5509-01&quot;   </span></a>
<a class="sourceLine" id="cb108-13" data-line-number="13"><span class="co">#&gt; [4] &quot;6115124-SRR8307108&quot; &quot;1115265-SRR8304838&quot; &quot;TCGA-EJ-5532-01&quot;</span></a>
<a class="sourceLine" id="cb108-14" data-line-number="14"></a>
<a class="sourceLine" id="cb108-15" data-line-number="15"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb108-16" data-line-number="16">  <span class="dt">data =</span> CNV.seqz, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb108-17" data-line-number="17">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb108-18" data-line-number="18">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-93-1.png" width="4200" /></p>
</div>
</div>
</div>
<div id="survival-analysis" class="section level1">
<h1>Survival Analysis</h1>
<p>To check the clinical relavance of signatures, this part we do survival analysis with Cox model.</p>
<div id="prepare-1" class="section level2">
<h2>Prepare</h2>
<p>Load package <strong>ezcox</strong> for batch Cox analysis.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb109-1" data-line-number="1"><span class="kw">library</span>(ezcox)</a></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb110-1" data-line-number="1">df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</a>
<a class="sourceLine" id="cb110-2" data-line-number="2">surv_df &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../data/PRAD_Survival.rds&quot;</span>)</a></code></pre></div>
<p>Clean the data to generate a data table for survival analysis.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb111-1" data-line-number="1">range01 &lt;-<span class="st"> </span><span class="cf">function</span>(x, ...) {</a>
<a class="sourceLine" id="cb111-2" data-line-number="2">  (x <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(x, ...)) <span class="op">/</span><span class="st"> </span>(<span class="kw">max</span>(x, ...) <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(x, ...))</a>
<a class="sourceLine" id="cb111-3" data-line-number="3">}</a>
<a class="sourceLine" id="cb111-4" data-line-number="4"></a>
<a class="sourceLine" id="cb111-5" data-line-number="5">cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CN_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SBS_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</a>
<a class="sourceLine" id="cb111-6" data-line-number="6"></a>
<a class="sourceLine" id="cb111-7" data-line-number="7"></a>
<a class="sourceLine" id="cb111-8" data-line-number="8">surv_dt &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(df.seqz <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-9" data-line-number="9"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb111-10" data-line-number="10">    <span class="st">&quot;Study&quot;</span>, <span class="st">&quot;subject_id&quot;</span>, <span class="st">&quot;tumor_body_site&quot;</span>,</a>
<a class="sourceLine" id="cb111-11" data-line-number="11">    <span class="st">&quot;tumor_Run&quot;</span>, <span class="st">&quot;normal_Run&quot;</span>, <span class="st">&quot;CNV_ID&quot;</span>,</a>
<a class="sourceLine" id="cb111-12" data-line-number="12">    <span class="st">&quot;Fusion&quot;</span>, <span class="st">&quot;sample_type&quot;</span></a>
<a class="sourceLine" id="cb111-13" data-line-number="13">  )),</a>
<a class="sourceLine" id="cb111-14" data-line-number="14">surv_df,</a>
<a class="sourceLine" id="cb111-15" data-line-number="15"><span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;PatientID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)</a>
<a class="sourceLine" id="cb111-16" data-line-number="16">)</a>
<a class="sourceLine" id="cb111-17" data-line-number="17"></a>
<a class="sourceLine" id="cb111-18" data-line-number="18">dup_ids &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">duplicated</span>(surv_dt<span class="op">$</span>PatientID))</a>
<a class="sourceLine" id="cb111-19" data-line-number="19"><span class="co"># Remove duplicated records</span></a>
<a class="sourceLine" id="cb111-20" data-line-number="20">surv_dt &lt;-<span class="st"> </span>surv_dt[<span class="op">-</span>dup_ids, ]</a>
<a class="sourceLine" id="cb111-21" data-line-number="21"></a>
<a class="sourceLine" id="cb111-22" data-line-number="22"><span class="co"># Scale the signature exposure to 0-20.</span></a>
<a class="sourceLine" id="cb111-23" data-line-number="23"><span class="co"># To better understand the effect of signature exposure on clinical events,</span></a>
<a class="sourceLine" id="cb111-24" data-line-number="24"><span class="co"># here we scale them into range 0-20. 1 increase means 5% increase of signature exposure.</span></a>
<a class="sourceLine" id="cb111-25" data-line-number="25"><span class="co">#</span></a>
<a class="sourceLine" id="cb111-26" data-line-number="26"><span class="co"># Scale the CNA burden to 0-20.</span></a>
<a class="sourceLine" id="cb111-27" data-line-number="27">surv_dt &lt;-<span class="st"> </span>surv_dt <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-28" data-line-number="28"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</a>
<a class="sourceLine" id="cb111-29" data-line-number="29">    <span class="dt">cna_burden =</span> <span class="dv">20</span> <span class="op">*</span><span class="st"> </span>cna_burden,</a>
<a class="sourceLine" id="cb111-30" data-line-number="30">    <span class="dt">Stage =</span> <span class="kw">as.character</span>(Stage) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">factor</span>(<span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;T2&quot;</span>, <span class="st">&quot;T3&quot;</span>, <span class="st">&quot;T4&quot;</span>))</a>
<a class="sourceLine" id="cb111-31" data-line-number="31">  ) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb111-32" data-line-number="32"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_at</span>(cols_to_sigs.seqz, <span class="op">~</span><span class="st"> </span><span class="dv">20</span> <span class="op">*</span><span class="st"> </span><span class="kw">range01</span>(., <span class="dt">na.rm =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb111-33" data-line-number="33"></a>
<a class="sourceLine" id="cb111-34" data-line-number="34"><span class="kw">saveRDS</span>(surv_dt, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_merged_survival_dt_seqz.rds&quot;</span>)</a></code></pre></div>
</div>
<div id="os-analysis-fo-signatures" class="section level2">
<h2>OS analysis fo signatures</h2>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb112-1" data-line-number="1"><span class="co">############</span></a>
<a class="sourceLine" id="cb112-2" data-line-number="2">surv_dt &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_merged_survival_dt_seqz.rds&quot;</span>)</a>
<a class="sourceLine" id="cb112-3" data-line-number="3">cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CN_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SBS_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</a>
<a class="sourceLine" id="cb112-4" data-line-number="4"></a>
<a class="sourceLine" id="cb112-5" data-line-number="5"><span class="co"># Unvariable analysis</span></a>
<a class="sourceLine" id="cb112-6" data-line-number="6"><span class="kw">show_forest</span>(surv_dt,</a>
<a class="sourceLine" id="cb112-7" data-line-number="7">  <span class="dt">covariates =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb112-8" data-line-number="8">  <span class="dt">time =</span> <span class="st">&quot;OS.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;OS&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb112-9" data-line-number="9">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-97-1.png" width="2400" /></p>
</div>
<div id="pfs-analysis-for-signatures" class="section level2">
<h2>PFS analysis for signatures</h2>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb113-1" data-line-number="1"><span class="kw">show_forest</span>(surv_dt,</a>
<a class="sourceLine" id="cb113-2" data-line-number="2">  <span class="dt">covariates =</span> cols_to_sigs.seqz,</a>
<a class="sourceLine" id="cb113-3" data-line-number="3">  <span class="dt">time =</span> <span class="st">&quot;PFI.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;PFI&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb113-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-98-1.png" width="2400" /></p>
</div>
<div id="os-analysis-for-features" class="section level2">
<h2>OS analysis for features</h2>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb114-1" data-line-number="1">cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb114-2" data-line-number="2">  <span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</a>
<a class="sourceLine" id="cb114-3" data-line-number="3">  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;INDELs&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</a>
<a class="sourceLine" id="cb114-4" data-line-number="4">  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</a>
<a class="sourceLine" id="cb114-5" data-line-number="5">  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></a>
<a class="sourceLine" id="cb114-6" data-line-number="6">)</a>
<a class="sourceLine" id="cb114-7" data-line-number="7"></a>
<a class="sourceLine" id="cb114-8" data-line-number="8"><span class="kw">show_forest</span>(surv_dt,</a>
<a class="sourceLine" id="cb114-9" data-line-number="9">  <span class="dt">covariates =</span> cols_to_features, <span class="dt">controls =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb114-10" data-line-number="10">  <span class="dt">time =</span> <span class="st">&quot;OS.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;OS&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span>, <span class="dt">limits =</span> <span class="kw">log</span>(<span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">10</span>))</a>
<a class="sourceLine" id="cb114-11" data-line-number="11">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-99-1.png" width="2400" /></p>
</div>
<div id="pfs-analysis-for-features" class="section level2">
<h2>PFS analysis for features</h2>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb115-1" data-line-number="1"><span class="kw">show_forest</span>(surv_dt,</a>
<a class="sourceLine" id="cb115-2" data-line-number="2">  <span class="dt">covariates =</span> cols_to_features, <span class="dt">controls =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb115-3" data-line-number="3">  <span class="dt">time =</span> <span class="st">&quot;PFI.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;PFI&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span>, <span class="dt">limits =</span> <span class="kw">c</span>(<span class="kw">log</span>(<span class="fl">0.5</span>), <span class="kw">log</span>(<span class="dv">5</span>))</a>
<a class="sourceLine" id="cb115-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-100-1.png" width="2400" /></p>
</div>
</div>
<div id="variation-summary" class="section level1">
<h1>Variation Summary</h1>
<p>This part discribes some big pictures of genomic variations in prostate cancers. They are not directly related to the core analyses previously reported.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb116-1" data-line-number="1"><span class="kw">library</span>(sigminer)</a>
<a class="sourceLine" id="cb116-2" data-line-number="2"></a>
<a class="sourceLine" id="cb116-3" data-line-number="3"></a>
<a class="sourceLine" id="cb116-4" data-line-number="4"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</a>
<a class="sourceLine" id="cb116-5" data-line-number="5"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.tally.W.RData&quot;</span>)</a>
<a class="sourceLine" id="cb116-6" data-line-number="6"><span class="kw">load</span>(<span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</a></code></pre></div>
<div id="oncoplot" class="section level2">
<h2>Oncoplot</h2>
<p>TODO: add GISTIC/MutSig, etc. data (plots ref to <a href="http://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/oncoplots.html" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/oncoplots.html</a>)</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb117-1" data-line-number="1"><span class="kw">oncoplot</span>(<span class="dt">maf =</span> Maf, <span class="dt">top =</span> <span class="dv">10</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-102-1.png" width="2400" /></p>
</div>
<div id="summary-plots" class="section level2">
<h2>Summary plots</h2>
<div id="sbs-2" class="section level3">
<h3>SBS</h3>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb118-1" data-line-number="1"><span class="kw">library</span>(maftools)</a></code></pre></div>
<div id="maf-summary" class="section level4">
<h4>MAF summary</h4>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb119-1" data-line-number="1"><span class="kw">plotmafSummary</span>(<span class="dt">maf =</span> Maf, <span class="dt">rmOutlier =</span> <span class="ot">TRUE</span>, <span class="dt">addStat =</span> <span class="st">&quot;median&quot;</span>, <span class="dt">dashboard =</span> <span class="ot">TRUE</span>, <span class="dt">titvRaw =</span> <span class="ot">FALSE</span>, <span class="dt">log_scale =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-104-1.png" width="2400" /></p>
</div>
<div id="transition-and-transversions" class="section level4">
<h4>Transition and Transversions</h4>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb120-1" data-line-number="1">maf.titv &lt;-<span class="st"> </span><span class="kw">titv</span>(<span class="dt">maf =</span> Maf, <span class="dt">plot =</span> <span class="ot">FALSE</span>, <span class="dt">useSyn =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb120-2" data-line-number="2"><span class="co"># plot titv summary</span></a>
<a class="sourceLine" id="cb120-3" data-line-number="3"><span class="kw">plotTiTv</span>(<span class="dt">res =</span> maf.titv)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-105-1.png" width="2400" /></p>
</div>
<div id="somatic-interactions" class="section level4">
<h4>Somatic Interactions</h4>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb121-1" data-line-number="1"><span class="kw">somaticInteractions</span>(<span class="dt">maf =</span> Maf, <span class="dt">top =</span> <span class="dv">25</span>, <span class="dt">pvalue =</span> <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.1</span>))</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-106-1.png" width="2400" /></p>
<pre><code>#&gt;       gene1  gene2       pValue  oddsRatio  00 11 01 10              Event
#&gt;   1: ABCA13  KMT2C 5.107880e-08 10.2128843 896 13 42 27       Co_Occurence
#&gt;   2:   RYR3 ABCA13 1.510688e-05  9.8852463 915  8 32 23       Co_Occurence
#&gt;   3:  OBSCN    TTN 2.087377e-05  4.7803808 841 15 94 28       Co_Occurence
#&gt;   4:  KMT2C  KMT2D 3.687480e-05  5.3043820 877 12 46 43       Co_Occurence
#&gt;   5:  FOXA1  CDK12 4.320195e-05  5.8603172 873 11 23 71       Co_Occurence
#&gt;  ---                                                                      
#&gt;               pair event_ratio
#&gt;   1: ABCA13, KMT2C       13/69
#&gt;   2:  ABCA13, RYR3        8/55
#&gt;   3:    OBSCN, TTN      15/122
#&gt;   4:  KMT2C, KMT2D       12/89
#&gt;   5:  CDK12, FOXA1       11/94
#&gt;  ---                          
#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 5 rows ]</code></pre>
</div>
</div>
<div id="copy-number-3" class="section level3">
<h3>Copy number</h3>
<div id="copy-number-segment-distribution" class="section level4">
<h4>Copy number segment distribution</h4>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb123-1" data-line-number="1"><span class="kw">show_cn_distribution</span>(CNV.seqz, <span class="dt">mode =</span> <span class="st">&quot;ld&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-107-1.png" width="2400" /></p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb124-1" data-line-number="1"><span class="kw">show_cn_distribution</span>(CNV.seqz, <span class="dt">mode =</span> <span class="st">&quot;cd&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-108-1.png" width="2400" /></p>
</div>
<div id="feature-distribution" class="section level4">
<h4>Feature distribution</h4>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb125-1" data-line-number="1"><span class="kw">show_cn_features</span>(CNV.seqz.tally.W<span class="op">$</span>features, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">log_y =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-109-1.png" width="3000" /></p>
</div>
<div id="component-distribution" class="section level4">
<h4>Component distribution</h4>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb126-1" data-line-number="1"><span class="kw">show_cn_components</span>(CNV.seqz.tally.W<span class="op">$</span>parameters, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">log_y =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-110-1.png" width="3000" /></p>
<p>Data details can be viewed as a table:</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb127-1" data-line-number="1">DT<span class="op">::</span><span class="kw">datatable</span>(CNV.seqz.tally.W<span class="op">$</span>parameters)</a></code></pre></div>
<div id="htmlwidget-604d5338210fce6527ad" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-604d5338210fce6527ad">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","CN","CN","CN","CN","CN","CN","CN","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","SS","SS","SS","SS","SS","SS","SS","SS","NC50","NC50","NC50","NC50","NC50","NC50","NC50","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr"],["BP10MB[0]","BP10MB[1]","BP10MB[2]","BP10MB[3]","BP10MB[4]","BP10MB[5]","BP10MB[&gt;5]","BPArm[0]","BPArm[1]","BPArm[2]","BPArm[3]","BPArm[4]","BPArm[5]","BPArm[6]","BPArm[7]","BPArm[8]","BPArm[9]","BPArm[10]","BPArm[&gt;10 &amp; &lt;=20]","BPArm[&gt;20 &amp; &lt;=30]","BPArm[&gt;30]","CN[0]","CN[1]","CN[2]","CN[3]","CN[4]","CN[&gt;4 &amp; &lt;=8]","CN[&gt;8]","CNCP[0]","CNCP[1]","CNCP[2]","CNCP[3]","CNCP[4]","CNCP[&gt;4 &amp; &lt;=8]","CNCP[&gt;8]","OsCN[0]","OsCN[1]","OsCN[2]","OsCN[3]","OsCN[4]","OsCN[&gt;4 &amp; &lt;=10]","OsCN[&gt;10]","SS[&lt;=2]","SS[&gt;2 &amp; &lt;=3]","SS[&gt;3 &amp; &lt;=4]","SS[&gt;4 &amp; &lt;=5]","SS[&gt;5 &amp; &lt;=6]","SS[&gt;6 &amp; &lt;=7]","SS[&gt;7 &amp; &lt;=8]","SS[&gt;8]","NC50[&lt;=2]","NC50[3]","NC50[4]","NC50[5]","NC50[6]","NC50[7]","NC50[&gt;7]","BoChr[1]","BoChr[2]","BoChr[3]","BoChr[4]","BoChr[5]","BoChr[6]","BoChr[7]","BoChr[8]","BoChr[9]","BoChr[10]","BoChr[11]","BoChr[12]","BoChr[13]","BoChr[14]","BoChr[15]","BoChr[16]","BoChr[17]","BoChr[18]","BoChr[19]","BoChr[20]","BoChr[21]","BoChr[22]","BoChr[23]"],["point","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","range","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","range","range","range","range","range","range","range","range","range","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point"],[0,1,2,3,4,5,5,0,1,2,3,4,5,6,7,8,9,10,10,20,30,0,1,2,3,4,4,8,0,1,2,3,4,4,8,0,1,2,3,4,4,10,null,2,3,4,5,6,7,8,null,3,4,5,6,7,7,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23],[0,1,2,3,4,5,null,0,1,2,3,4,5,6,7,8,9,10,20,30,null,0,1,2,3,4,8,null,0,1,2,3,4,8,null,0,1,2,3,4,10,null,2,3,4,5,6,7,8,null,2,3,4,5,6,7,null,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23],[262755,22436,12074,2349,1277,374,449,27296,4763,5737,1896,1836,901,824,432,358,226,186,466,50,5,1464,7431,31755,11911,13087,15272,5372,8350,29323,15725,5661,3477,5024,4594,31108,7283,834,601,129,174,6,202,1995,2478,4829,10672,22165,36157,7794,89,111,141,163,160,207,66,4303,2673,2991,2359,2485,3587,3805,2969,2299,2654,2531,2294,1405,1563,1175,2515,3060,1291,2210,1853,1209,1443,816]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>feature<\/th>\n      <th>component<\/th>\n      <th>label<\/th>\n      <th>min<\/th>\n      <th>max<\/th>\n      <th>n_obs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<blockquote>
<p><code>n_obs</code> column indicates the number of event of corresponding copy number component.</p>
</blockquote>
</div>
</div>
</div>
</div>
<div id="supplementary-analysis-and-visualization" class="section level1">
<h1>Supplementary Analysis and Visualization</h1>
<p>This part shows some supplementary analyses and descriptions.</p>
<div id="facets-results" class="section level2">
<h2>FACETS results</h2>
<p>We used 2 methods (Sequenza and FACETS) for copy number calling. However, only Sequenza was used for previous downstream analyses due to the fact that FACETS generated some unreasonable copy number profile for a few samples. We will talk it here.</p>
<p>Let’s load the packages and signature data from FACETS and Sequenza.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb128-1" data-line-number="1"><span class="kw">library</span>(sigminer)</a>
<a class="sourceLine" id="cb128-2" data-line-number="2"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb128-3" data-line-number="3"></a>
<a class="sourceLine" id="cb128-4" data-line-number="4"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.facets.W.RData&quot;</span>)</a>
<a class="sourceLine" id="cb128-5" data-line-number="5"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</a>
<a class="sourceLine" id="cb128-6" data-line-number="6"><span class="kw">load</span>(<span class="st">&quot;../output/EST.facets.W.all.RData&quot;</span>)</a></code></pre></div>
<p>According to the estimation of signature number, 6 signatures were extracted.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb129-1" data-line-number="1">p &lt;-<span class="st"> </span><span class="kw">show_sig_number_survey</span>(EST.facets.W.all, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a>
<a class="sourceLine" id="cb129-2" data-line-number="2"><span class="kw">add_h_arrow</span>(p, <span class="dt">x =</span> <span class="fl">6.1</span>, <span class="dt">y =</span> <span class="fl">0.986</span>, <span class="dt">seg_len =</span> <span class="fl">0.5</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-113-1.png" width="2400" /></p>
<p>The result signatures are shown as the below.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb130-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.facets.W, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">normalize =</span> <span class="st">&quot;feature&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb130-2" data-line-number="2"><span class="st">  </span><span class="kw">labs</span>(<span class="dt">caption =</span> <span class="st">&quot;Signatures from FACETS&quot;</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-114-1.png" width="3600" /></p>
<p>FACETS result has one special and extra signature (<strong>Sig1</strong>) when compared to <a href="#copy-number-signature-profile">Sequenza result</a>. The <strong>Sig1</strong> shows copy number segments with high percentage of component “copy number value = 0”.</p>
<blockquote>
<p>Other signatures are not identical but very similar in most features beteween FACETS and Sequenza results.</p>
</blockquote>
<p>We check some samples with <strong>Sig1</strong> enriched.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb131-1" data-line-number="1">cn_groups &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.CNV.facets.W, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb131-2" data-line-number="2"><span class="co">#&gt; =&gt; Obtaining clusters from the hierarchical clustering of the consensus matrix...</span></a>
<a class="sourceLine" id="cb131-3" data-line-number="3"><span class="co">#&gt; =&gt; Finding the dominant signature of each group...</span></a>
<a class="sourceLine" id="cb131-4" data-line-number="4"><span class="co">#&gt; =&gt; Generating a table of group and dominant signature:</span></a>
<a class="sourceLine" id="cb131-5" data-line-number="5"><span class="co">#&gt;    </span></a>
<a class="sourceLine" id="cb131-6" data-line-number="6"><span class="co">#&gt;     Sig1 Sig2 Sig3 Sig4 Sig5 Sig6</span></a>
<a class="sourceLine" id="cb131-7" data-line-number="7"><span class="co">#&gt;   1    0    0   27    1    0    0</span></a>
<a class="sourceLine" id="cb131-8" data-line-number="8"><span class="co">#&gt;   2   19    2    0    7    0    0</span></a>
<a class="sourceLine" id="cb131-9" data-line-number="9"><span class="co">#&gt;   3    0   92    1    2    0    0</span></a>
<a class="sourceLine" id="cb131-10" data-line-number="10"><span class="co">#&gt;   4    7   52    8   59    9    5</span></a>
<a class="sourceLine" id="cb131-11" data-line-number="11"><span class="co">#&gt;   5    0    0    0  188    0    0</span></a>
<a class="sourceLine" id="cb131-12" data-line-number="12"><span class="co">#&gt;   6    5    2    0  155    0  292</span></a>
<a class="sourceLine" id="cb131-13" data-line-number="13"><span class="co">#&gt; =&gt; Assigning a group to a signature with the maxium fraction (stored in &#39;map_table&#39; attr)...</span></a>
<a class="sourceLine" id="cb131-14" data-line-number="14"><span class="co">#&gt; =&gt; Summarizing...</span></a>
<a class="sourceLine" id="cb131-15" data-line-number="15"><span class="co">#&gt;  group #1: 28 samples with Sig3 enriched.</span></a>
<a class="sourceLine" id="cb131-16" data-line-number="16"><span class="co">#&gt;  group #2: 28 samples with Sig1 enriched.</span></a>
<a class="sourceLine" id="cb131-17" data-line-number="17"><span class="co">#&gt;  group #3: 95 samples with Sig2 enriched.</span></a>
<a class="sourceLine" id="cb131-18" data-line-number="18"><span class="co">#&gt;  group #4: 140 samples with Sig4 enriched.</span></a>
<a class="sourceLine" id="cb131-19" data-line-number="19"><span class="co">#&gt;  group #5: 188 samples with Sig4 enriched.</span></a>
<a class="sourceLine" id="cb131-20" data-line-number="20"><span class="co">#&gt;  group #6: 454 samples with Sig6 enriched.</span></a>
<a class="sourceLine" id="cb131-21" data-line-number="21">cn_expo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.CNV.facets.W, <span class="dt">type =</span> <span class="st">&quot;relative&quot;</span>)</a>
<a class="sourceLine" id="cb131-22" data-line-number="22"><span class="co"># Focus on Sig1</span></a>
<a class="sourceLine" id="cb131-23" data-line-number="23">df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(cn_groups, cn_expo)</a>
<a class="sourceLine" id="cb131-24" data-line-number="24"><span class="co">#&gt; Joining, by = &quot;sample&quot;</span></a>
<a class="sourceLine" id="cb131-25" data-line-number="25"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.facets.RData&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb132-1" data-line-number="1">samps_to_show &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb132-2" data-line-number="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(enrich_sig <span class="op">==</span><span class="st"> &quot;Sig1&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb132-3" data-line-number="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(dplyr<span class="op">::</span><span class="kw">desc</span>(Sig1)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb132-4" data-line-number="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">slice</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb132-5" data-line-number="5"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(sample)</a>
<a class="sourceLine" id="cb132-6" data-line-number="6"></a>
<a class="sourceLine" id="cb132-7" data-line-number="7">samps_to_show</a>
<a class="sourceLine" id="cb132-8" data-line-number="8"><span class="co">#&gt; [1] &quot;915-1115081&quot;        &quot;141-6&quot;              &quot;447-STID0000003036&quot;</span></a>
<a class="sourceLine" id="cb132-9" data-line-number="9"><span class="co">#&gt; [4] &quot;915-5115031&quot;        &quot;141-25&quot;             &quot;915-1115090&quot;</span></a>
<a class="sourceLine" id="cb132-10" data-line-number="10"></a>
<a class="sourceLine" id="cb132-11" data-line-number="11"><span class="kw">show_cn_profile</span>(</a>
<a class="sourceLine" id="cb132-12" data-line-number="12">  <span class="dt">data =</span> CNV.facets, <span class="dt">chrs =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">22</span>, <span class="st">&quot;X&quot;</span>)), <span class="dt">nrow =</span> <span class="dv">3</span>, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">show_title =</span> T,</a>
<a class="sourceLine" id="cb132-13" data-line-number="13">  <span class="dt">samples =</span> samps_to_show</a>
<a class="sourceLine" id="cb132-14" data-line-number="14">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-116-1.png" width="4200" /></p>
<p>We can observe that the samples with <strong>Sig1</strong> enriched have many homozygous deletions, this is unreasonable. I made a communication with the author of FACETS (<a href="https://github.com/mskcc/facets/issues/147">issue link</a>),</p>
<blockquote>
<p>So only logical explanation is homozygous deletion. However they are too many of them and so I believe this is assay artifact rather than real.</p>
</blockquote>
<p>This problem seems to come from the raw data. However, we checked the data and this signature cannot be found from the result of Sequenza.</p>
<p>To summary, we used result from Sequenza instead of FACETS for downstream analyses because we found there may be some unknown method problems in FACETS (in general, this method is still reliable).</p>
</div>
<div id="method-comparison-between-macintyre-et-al-and-our-study" class="section level2">
<h2>Method comparison between Macintyre et al and our study</h2>
<p>In section <a href="#tally-variation-records">“Tally variation records”</a>, I said that our method (defined as “Wang” or “W”) has some advantages over the method used in Macintyre et al (defined as “Macintyre” or “M”).</p>
<p>I will talk the details here.</p>
<p>Firsty let’s load the results from “M” method (the code to generate the data can be found <a href="https://github.com/ShixiangWang/prad_signature/blob/master/analysis/src/02-identify-copynumber-signatures.R">here</a>).</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb133-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.M.RData&quot;</span>)</a>
<a class="sourceLine" id="cb133-2" data-line-number="2"><span class="kw">load</span>(<span class="st">&quot;../output/EST.seqz.M.RData&quot;</span>)</a>
<a class="sourceLine" id="cb133-3" data-line-number="3"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.tally.M.RData&quot;</span>)</a></code></pre></div>
<div id="sample-by-component-matrix-generation" class="section level3">
<h3>Sample-by-component matrix generation</h3>
<p>The key difference between method “W” and method “M” is the way to generate the sample-by-component matrix.</p>
<ul>
<li>Method “W” defines the components and counts them (details see section <a href="#tally-variation-records">“Tally variation records”</a>).</li>
<li>Method “M” applies mixture modeling to separate the 6 copy number features distributions into mixtures of Poisson or Gaussian distributions.</li>
</ul>
<p>Here we found that method “M” has the following drawbacks:</p>
<ol style="list-style-type: decimal">
<li>the computation needs huge computing resources. For generating the matrix used for NMF algorithm, it takes about 5100s (~1.5h) with 20 cores for method “M” while only 86s (~1.5m) with 20 cores for method “W”.</li>
<li>the number of component is dependent on the raw data. If we change the sample size or apply the method “M” to another cohort, the number of component may change. Method “W” will not be affected by this because it uses a fixed number of component.</li>
<li>the distribution parameters for each distribution are not identical for different runs. Even for the same data, the result distributions and sample-by-component matrix may change.</li>
</ol>
<p>At the start of this study, I used the method “M” for all analyses. Finally, we created method “W” based on the use and rethinking of method “M” and SBS signature extraction method from previous reports.</p>
<p>Next section, I will talk more problems raised by method “M” from the visualization view.</p>
</div>
<div id="copy-number-signatures-from-m-method" class="section level3">
<h3>Copy number signatures from “M” method</h3>
<p>We still use the cophenetic plot for deciding on the signature number.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb134-1" data-line-number="1"><span class="kw">show_sig_number_survey</span>(EST.seqz.M, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-118-1.png" width="2400" /></p>
<p>We selected 4 signatures. Let’s see their signature profile.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb135-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.M,</a>
<a class="sourceLine" id="cb135-2" data-line-number="2">  <span class="dt">normalize =</span> <span class="st">&quot;column&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb135-3" data-line-number="3">  <span class="dt">paint_axis_text =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb135-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-119-1.png" width="2400" /></p>
<p>This is a bar plot similar to the one reported by Macintyre et al.</p>
<p>This plot raises some issues.</p>
<div id="issue-1" class="section level4">
<h4>Issue 1</h4>
<p>The matrix used to generate the plot is column normalized, so <strong>the heights of bar cannot be compared with each other</strong>.</p>
<p>For example, firstly we see the <strong>Sig1</strong>, we may conclude that <strong>component “bp10MB2” has more contributions than component “bp10MB1”</strong>. But if we see the raw matrix, we can see that this is wrong!</p>
<p>We cannot compare two or more componets in a row when we normalize the data by column! The plot above may misguide the readers. For the plot above, we can only compare the same component across all signatures. For example, we can conclude that <strong>Sig4</strong> captures the majority of component <strong>“segsize12”</strong> and <strong>“segsize11”</strong>.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb136-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.M,</a>
<a class="sourceLine" id="cb136-2" data-line-number="2">  <span class="dt">normalize =</span> <span class="st">&quot;raw&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb136-3" data-line-number="3">  <span class="dt">paint_axis_text =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb136-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-120-1.png" width="2400" /></p>
<p>To avoid misguiding readers, gradient colors may work. Also, we must tell readers in text that we should compare one component in a column insteads two or more componets in a row.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb137-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.M,</a>
<a class="sourceLine" id="cb137-2" data-line-number="2">  <span class="dt">normalize =</span> <span class="st">&quot;column&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb137-3" data-line-number="3">  <span class="dt">paint_axis_text =</span> <span class="ot">FALSE</span>, <span class="dt">set_gradient_color =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb137-4" data-line-number="4">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-121-1.png" width="2400" /></p>
</div>
<div id="issue-2" class="section level4">
<h4>Issue 2</h4>
<p>From plot above, we can only see different components, but we cannot see the meaning of each component. For method “M”, each component represents a distribution.</p>
<p>For example, what’s the meaning of the component “copynumber2”? To fix this issue, we tried adding the key parameter value in the plot.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb138-1" data-line-number="1"><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.M,</a>
<a class="sourceLine" id="cb138-2" data-line-number="2">  <span class="dt">normalize =</span> <span class="st">&quot;column&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>,</a>
<a class="sourceLine" id="cb138-3" data-line-number="3">  <span class="dt">paint_axis_text =</span> <span class="ot">FALSE</span>, <span class="dt">set_gradient_color =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb138-4" data-line-number="4">  <span class="dt">params =</span> CNV.seqz.tally.M<span class="op">$</span>parameters, <span class="dt">y_expand =</span> <span class="fl">1.5</span></a>
<a class="sourceLine" id="cb138-5" data-line-number="5">)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-122-1.png" width="2400" /></p>
<p>According to the text just added, we can see that component “copynumber2” represents a Gaussian distribution with mean is 2; component “copynumber1” represents a Gaussian distribution with mean is 0.84.</p>
</div>
<div id="issue-3" class="section level4">
<h4>Issue 3</h4>
<p>Even with the plot above, we cannot know the dispersion of each component. All distributions with weight for each copy number feature can be viewed as the below.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb139-1" data-line-number="1"><span class="kw">show_cn_components</span>(CNV.seqz.tally.M<span class="op">$</span>parameters, <span class="dt">show_weights =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-123-1.png" width="2400" /></p>
</div>
</div>
<div id="why-we-created-method-w" class="section level3">
<h3>Why we created method “W”</h3>
<p>Here, I want to talk something more.</p>
<p>If we take case of all available values for each features, we would notice that there are only limited choices. Let’s visualize them.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb140-1" data-line-number="1"><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.tally.W.RData&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb141-1" data-line-number="1"><span class="kw">show_cn_features</span>(CNV.seqz.tally.W<span class="op">$</span>features[<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>], <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">log_y =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-125-1.png" width="1800" /></p>
<p>Naturally we can divide them into categories and get a fixed number of components. It would be easy to read, understand and compare. Look back this part, why we choose a complicated method with many drawbacks instead of a simple one?</p>
</div>
</div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1>Reference</h1>
<div id="refs" class="references">
<div id="ref-alexandrov2013signatures">
<p>Alexandrov, Ludmil B, Serena Nik-Zainal, David C Wedge, Samuel AJR Aparicio, Sam Behjati, Andrew V Biankin, Graham R Bignell, et al. 2013. “Signatures of Mutational Processes in Human Cancer.” <em>Nature</em> 500 (7463). Nature Publishing Group: 415.</p>
</div>
<div id="ref-armenia2018long">
<p>Armenia, Joshua, Stephanie AM Wankowicz, David Liu, Jianjiong Gao, Ritika Kundra, Ed Reznik, Walid K Chatila, et al. 2018. “The Long Tail of Oncogenic Drivers in Prostate Cancer.” <em>Nature Genetics</em> 50 (5). Nature Publishing Group: 645.</p>
</div>
<div id="ref-favero2014sequenza">
<p>Favero, Francesco, Tejal Joshi, Andrea Marion Marquard, Nicolai Juul Birkbak, Marcin Krzystanek, Qiyuan Li, Z Szallasi, and Aron Charles Eklund. 2014. “Sequenza: Allele-Specific Copy Number and Mutation Profiles from Tumor Sequencing Data.” <em>Annals of Oncology</em> 26 (1). European Society for Medical Oncology: 64–70.</p>
</div>
<div id="ref-gaujoux2010flexible">
<p>Gaujoux, Renaud, and Cathal Seoighe. 2010. “A Flexible R Package for Nonnegative Matrix Factorization.” <em>BMC Bioinformatics</em> 11 (1). BioMed Central: 367.</p>
</div>
<div id="ref-krueger2015trim">
<p>Krueger, Felix. 2015. “Trim Galore.” <em>A Wrapper Tool Around Cutadapt and FastQC to Consistently Apply Quality and Adapter Trimming to FastQ Files</em>.</p>
</div>
<div id="ref-li2013aligning">
<p>Li, Heng. 2013. “Aligning Sequence Reads, Clone Sequences and Assembly Contigs with Bwa-Mem.” <em>arXiv Preprint arXiv:1303.3997</em>.</p>
</div>
<div id="ref-li2009sequence">
<p>Li, Heng, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, and Richard Durbin. 2009. “The Sequence Alignment/Map Format and Samtools.” <em>Bioinformatics</em> 25 (16). Oxford University Press: 2078–9.</p>
</div>
<div id="ref-macintyre2018copy">
<p>Macintyre, Geoff, Teodora E Goranova, Dilrini De Silva, Darren Ennis, Anna M Piskorz, Matthew Eldridge, Daoud Sie, et al. 2018. “Copy Number Signatures and Mutational Processes in Ovarian Carcinoma.” <em>Nature Genetics</em> 50 (9). Nature Publishing Group: 1262.</p>
</div>
<div id="ref-mayakonda2016maftools">
<p>Mayakonda, Anand, and H Phillip Koeffler. 2016. “Maftools: Efficient Analysis, Visualization and Summarization of Maf Files from Large-Scale Cohort Based Cancer Studies.” <em>BioRxiv</em>. Cold Spring Harbor Laboratory, 052662.</p>
</div>
<div id="ref-mclaren2016ensembl">
<p>McLaren, William, Laurent Gil, Sarah E Hunt, Harpreet Singh Riat, Graham RS Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. 2016. “The Ensembl Variant Effect Predictor.” <em>Genome Biology</em> 17 (1). BioMed Central: 122.</p>
</div>
<div id="ref-Picard2019toolkit">
<p>“Picard Toolkit.” 2019. <em>Broad Institute, GitHub Repository</em>. <a href="http://broadinstitute.github.io/picard/">http://broadinstitute.github.io/picard/</a>; Broad Institute.</p>
</div>
<div id="ref-shen_facets:_2016">
<p>Shen, Ronglai, and Venkatraman E. Seshan. 2016. “FACETS: Allele-Specific Copy Number and Clonal Heterogeneity Analysis Tool for High-Throughput DNA Sequencing.” <em>Nucleic Acids Research</em> 44 (16): e131–e131. <a href="https://doi.org/10.1093/nar/gkw520">https://doi.org/10.1093/nar/gkw520</a>.</p>
</div>
<div id="ref-van2013fastq">
<p>Van der Auwera, Geraldine A, Mauricio O Carneiro, Christopher Hartl, Ryan Poplin, Guillermo Del Angel, Ami Levy-Moonshine, Tadeusz Jordan, et al. 2013. “From Fastq Data to High-Confidence Variant Calls: The Genome Analysis Toolkit Best Practices Pipeline.” <em>Current Protocols in Bioinformatics</em> 43 (1). Wiley Online Library: 11–10.</p>
</div>
<div id="ref-wang2019ucscxenatools">
<p>Wang, Shixiang, and Xuesong Liu. 2019. “The Ucscxenatools R Package: A Toolkit for Accessing Genomics Data from Ucsc Xena Platform, from Cancer Multi-Omics to Single-Cell Rna-Seq.” <em>Journal of Open Source Software</em> 4 (40): 1627.</p>
</div>
<div id="ref-yi2018patterns">
<p>Yi, Kijong, and Young Seok Ju. 2018. “Patterns and Mechanisms of Structural Variations in Human Cancer.” <em>Experimental &amp; Molecular Medicine</em> 50 (8). Nature Publishing Group: 98.</p>
</div>
</div>
</div>
</div>


</div>

<div id="postamble" data-toggle="wy-nav-shift" class="status">
<p class="author"><span class="glyphicon glyphicon-user"></span> Shixiang Wang</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Huimin Li</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Xuan Wang</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Minfang Song</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Xue-Song Liu (Corresponding author)</p>
<p class="date"><span class="glyphicon glyphicon-calendar"></span> 2020-02-10</p>
</div>


<script>
$(document).ready(function () {
 	    $('#content img')
 	  .addClass("image-lb");
  $('#content').magnificPopup({
	      type:'image',
	      closeOnContentClick: false,
	      closeBtnInside: false,
	      delegate: 'img',
	      gallery: {enabled: true },
	      image: {
	        verticalFit: true,
          titleSrc: 'alt'
	      }
 	    });
 	});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
