<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Shixiang Wang" />
<meta name="author" content="Huimin Li" />
<meta name="author" content="Xuan Wang" />
<meta name="author" content="Minfang Song" />
<meta name="author" content="Xue-Song Liu (Corresponding author)" />

<meta name="date" content="2020-01-11" />

<title>Prostate Cancer Variation Signature Analysis Report</title>

<script src="index_files/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="index_files/bootstrap-3.3.6/css/bootstrap.min.css" rel="stylesheet" />
<script src="index_files/bootstrap-3.3.6/js/bootstrap.min.js"></script>
<script src="index_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<script src="index_files/navigation-1.1/tabsets.js"></script>
<script src="index_files/navigation-1.1/codefolding.js"></script>
<link href="index_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
<script src="index_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
<link href="index_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
<script src="index_files/readthedown-0.1/readthedown.js"></script>
<script src="index_files/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="index_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="index_files/datatables-binding-0.10/datatables.js"></script>
<link href="index_files/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="index_files/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="index_files/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="index_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="index_files/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
</style>


</head>

<body>


<div id="content" data-toggle="wy-nav-shift">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("text-table-of-contents");
});
</script>

<!-- code folding -->

<nav id="nav-top" role="navigation" aria-label="top navigation">
    <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
</nav>


<div id="header">
<h1 class="title">Prostate Cancer Variation Signature Analysis Report</h1>
</div>


<div id="table-of-contents">
    <h2><a href="#content">Prostate Cancer Variation Signature Analysis Report</a></h2>
    <div id="text-table-of-contents">
      <ul>
      <li><a href="#data-preprocessing">Data Preprocessing</a>
      <ul>
      <li><a href="#download-data">Download Data</a></li>
      <li><a href="#preprocess-data">Preprocess Data</a></li>
      </ul></li>
      <li><a href="#signature-identification">Signature Identification</a>
      <ul>
      <li><a href="#tally-variation-records">Tally variation records</a></li>
      <li><a href="#estimate-signature-number">Estimate signature number</a></li>
      <li><a href="#extract-signatures">Extract signatures</a></li>
      <li><a href="#show-signature-profile-and-aetiology">Show signature profile and aetiology</a></li>
      </ul></li>
      <li><a href="#association-analysis">Association Analysis</a>
      <ul>
      <li><a href="#data-integration">Data integration</a></li>
      <li><a href="#association-analysis-1">Association analysis</a></li>
      </ul></li>
      <li><a href="#group-analysis">Group Analysis</a>
      <ul>
      <li><a href="#group-comparison">Group comparison</a></li>
      <li><a href="#group-mapping">Group mapping</a></li>
      </ul></li>
      <li><a href="#survival-analysis">Survival Analysis</a>
      <ul>
      <li><a href="#prepare-1">Prepare</a></li>
      <li><a href="#os-analysis-fo-signatures">OS analysis fo signatures</a></li>
      <li><a href="#pfs-analysis-for-signatures">PFS analysis for signatures</a></li>
      <li><a href="#os-analysis-for-features">OS analysis for features</a></li>
      <li><a href="#pfi-analysis-for-features">PFI analysis for features</a></li>
      </ul></li>
      <li><a href="#variation-summary">Variation Summary</a></li>
      <li><a href="#supplementary-analysis-and-visualization">Supplementary Analysis and Visualization</a></li>
      <li><a href="#reference">Reference</a></li>
      </ul>
    </div>
</div>

<div id="main">
<p>Contact Prof. Xue-Song Liu: <a href="mailto:liuxs@shanghaitech.edu.cn" class="email">liuxs@shanghaitech.edu.cn</a></p>
<p>This report is written to help user understand what and how we did in this project. Please read the formal manuscript <strong>TBD</strong> for more details.</p>
<blockquote>
<p>This document is compiled from an Rmarkdown file which contains all code or description necessary to (auto-)reproduce the analysis for the accompanying project. Each section below describes a different component of the analysis and all numbers and figures are generated directly from the underlying data on compilation.</p>
</blockquote>
<div id="data-preprocessing" class="section level1">
<h1>Data Preprocessing</h1>
<div id="download-data" class="section level2">
<h2>Download Data</h2>
<div id="wes-data" class="section level3">
<h3>WES Data</h3>
</div>
<div id="phenotype-data" class="section level3">
<h3>Phenotype Data</h3>
</div>
<div id="survival-data" class="section level3">
<h3>Survival Data</h3>
</div>
</div>
<div id="preprocess-data" class="section level2">
<h2>Preprocess Data</h2>
<div id="pipeline" class="section level3">
<h3>Pipeline</h3>
<p><img src="D:/Code/prad_signature/analysis/fig/PRAD_Analysis_Pipeline.png" style="display: block; margin: auto;" /></p>
<center>
<strong>Genomic variation signature identification pipeline of prostate cancer</strong>
</center>
</div>
<div id="sequence-alignment" class="section level3">
<h3>Sequence Alignment</h3>
</div>
<div id="copy-number-calling" class="section level3">
<h3>Copy Number Calling</h3>
</div>
<div id="mutation-calling" class="section level3">
<h3>Mutation Calling</h3>
</div>
<div id="clean-data" class="section level3">
<h3>Clean Data</h3>
</div>
</div>
</div>
<div id="signature-identification" class="section level1">
<h1>Signature Identification</h1>
<p>This part has been divided into 4 sections as the following, it will describe how to get the genomic variation (or instability) signatures from mutation profile (including copy number profile and single base substitution (SBS) profile).</p>
<ul>
<li>Tally variation records</li>
<li>Estimate signature number</li>
<li>Extract signatures</li>
<li>Show signature profile and aetiology</li>
</ul>
<p>All the processes have been implemented in R package <a href="https://github.com/ShixiangWang/sigminer"><strong>sigminer</strong></a>.</p>
<div id="tally-variation-records" class="section level2">
<h2>Tally variation records</h2>
<div id="methods" class="section level3">
<h3>Methods</h3>
<p>For SBS profile, same as previously reported <span class="citation">(Alexandrov et al. <a href="#ref-alexandrov2013signatures" role="doc-biblioref">2013</a>)</span>, for each sample, we firstly classified mutation records into six substitution subtypes: C&gt;A, C&gt;G, C&gt;T, T&gt;A, T&gt;C, and T&gt;G (all substitutions are referred to by the pyrimidine of the mutated Watson–Crick base pair). Further, each of the substitutions was examined by incorporating information on the bases immediately 5’ and 3’ to each mutated base generating 96 possible mutation types (6 types of substitution ∗ 4 types of 5’ base ∗ 4 types of 3’ base). Each of 96 mutation types is called <strong>component</strong> here.</p>
<p>For copy number profile, we firstly computed the genome-wide distributions of 8 fundamental copy number features for each sample:</p>
<ul>
<li>the breakpoint count per 10 Mb (BP10MB)</li>
<li>the breakpoint count per chromosome arm (BPArm)</li>
<li>the copy number of the segments (CN)</li>
<li>the difference in copy number between adjacent segments (CNCP)</li>
<li>the lengths of oscillating copy number segment chains (OsCN)</li>
<li>the log10 based size of segments (SS)</li>
<li>the minimal number of chromosome with 50% copy number variation (NC50)</li>
<li>the burden of chromosome (BoChr)</li>
</ul>
<p>These features were selected as hallmarks of previously reported genomic aberations like chromothripsis or to denote the distribution pattern of copy number events. The first 6 features has been used in <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> to uncover the mutational processes in ovarian carcinoma.</p>
<p>Next, unlike <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> applied mixture modeling to separate the first 6 copy number features distributions into mixtures of Poisson or Gaussian distributions, we directly classified 8 copy number features distributions into different components according to the comprehensive consideration of value range, abundence and biological significance. Most of the result are discrete values, and the others are range values.</p>
<p>The setting of 8 features with 80 components are shown as below.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>DT<span class="op">::</span><span class="kw">datatable</span>(sigminer<span class="op">::</span>CN.features)</span></code></pre></div>
<div id="htmlwidget-e83dd83decade47f4973" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e83dd83decade47f4973">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BP10MB","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","BPArm","CN","CN","CN","CN","CN","CN","CN","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","CNCP","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","OsCN","SS","SS","SS","SS","SS","SS","SS","SS","NC50","NC50","NC50","NC50","NC50","NC50","NC50","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr","BoChr"],["BP10MB[0]","BP10MB[1]","BP10MB[2]","BP10MB[3]","BP10MB[4]","BP10MB[5]","BP10MB[&gt;5]","BPArm[0]","BPArm[1]","BPArm[2]","BPArm[3]","BPArm[4]","BPArm[5]","BPArm[6]","BPArm[7]","BPArm[8]","BPArm[9]","BPArm[10]","BPArm[&gt;10 &amp; &lt;=20]","BPArm[&gt;20 &amp; &lt;=30]","BPArm[&gt;30]","CN[0]","CN[1]","CN[2]","CN[3]","CN[4]","CN[&gt;4 &amp; &lt;=8]","CN[&gt;8]","CNCP[0]","CNCP[1]","CNCP[2]","CNCP[3]","CNCP[4]","CNCP[&gt;4 &amp; &lt;=8]","CNCP[&gt;8]","OsCN[0]","OsCN[1]","OsCN[2]","OsCN[3]","OsCN[4]","OsCN[&gt;4 &amp; &lt;=10]","OsCN[&gt;10]","SS[&lt;=2]","SS[&gt;2 &amp; &lt;=3]","SS[&gt;3 &amp; &lt;=4]","SS[&gt;4 &amp; &lt;=5]","SS[&gt;5 &amp; &lt;=6]","SS[&gt;6 &amp; &lt;=7]","SS[&gt;7 &amp; &lt;=8]","SS[&gt;8]","NC50[&lt;=2]","NC50[3]","NC50[4]","NC50[5]","NC50[6]","NC50[7]","NC50[&gt;7]","BoChr[1]","BoChr[2]","BoChr[3]","BoChr[4]","BoChr[5]","BoChr[6]","BoChr[7]","BoChr[8]","BoChr[9]","BoChr[10]","BoChr[11]","BoChr[12]","BoChr[13]","BoChr[14]","BoChr[15]","BoChr[16]","BoChr[17]","BoChr[18]","BoChr[19]","BoChr[20]","BoChr[21]","BoChr[22]","BoChr[23]"],["point","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","range","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","point","point","point","point","point","range","range","range","range","range","range","range","range","range","range","range","point","point","point","point","point","range","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point","point"],[0,1,2,3,4,5,5,0,1,2,3,4,5,6,7,8,9,10,10,20,30,0,1,2,3,4,4,8,0,1,2,3,4,4,8,0,1,2,3,4,4,10,null,2,3,4,5,6,7,8,null,3,4,5,6,7,7,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23],[0,1,2,3,4,5,null,0,1,2,3,4,5,6,7,8,9,10,20,30,null,0,1,2,3,4,8,null,0,1,2,3,4,8,null,0,1,2,3,4,10,null,2,3,4,5,6,7,8,null,2,3,4,5,6,7,null,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>feature<\/th>\n      <th>component<\/th>\n      <th>label<\/th>\n      <th>min<\/th>\n      <th>max<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<blockquote>
<p>When component is a discrete value (label is ‘point’), min=max</p>
<p>When component is a range value (label is ‘range’), the range is left open and right closed</p>
<p>Of note, the blank in <code>min</code> column is <code>-Inf</code> and the blank in <code>max</code> column is <code>Inf</code>. <code>[23]</code> in <code>BoChr</code> represents chromosome X (chromosome Y is excluded).</p>
</blockquote>
<p>Compare to the method from <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span>, our method has several advantages:</p>
<ol style="list-style-type: decimal">
<li>the computation is more efficient</li>
<li>the meaning of component is easier to read and understand</li>
<li>the new features NC50 and BoChr can be used to determine the distribution pattern of copy number events (global or local) and the contribution of each chromosome</li>
<li>most importantly, the classification is fixed, so it is much easier to compare the signatures within/across tumor types, the results across different studies and the result signatures to reference signatures. In a word, our method constructs the standard to extract, study and compare signatures.</li>
<li>our method is more extensible</li>
</ol>
<p>I will talk more about No.1-4 in an individual section (<strong>TBD</strong>), the No.5 will be described in <a href="">sigminer vignette</a>(<strong>TBD</strong>).</p>
<p>We generated sample-by-component matrix using the methods above and treated the result as the input of nonnegative matrix decomposition (NMF) algorithm for extracting signatures.</p>
</div>
<div id="tally-records" class="section level3">
<h3>Tally records</h3>
<div id="copy-number" class="section level4">
<h4>Copy number</h4>
<p>Read absolute copy number profile as a <code>CopyNumber</code> object.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co"># Load packages -----------------------------------------------------------</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(sigminer))</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(tidyverse))</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(NMF))</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># Set this per R session</span></span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">options</span>(<span class="dt">sigminer.sex =</span> <span class="st">&quot;male&quot;</span>, <span class="dt">sigminer.copynumber.max =</span> 20L)</span>
<span id="cb3-3"><a href="#cb3-3"></a></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="co"># Generate CopyNumber object ----------------------------------------------</span></span>
<span id="cb3-5"><a href="#cb3-5"></a></span>
<span id="cb3-6"><a href="#cb3-6"></a>CNV.seqz &lt;-<span class="st"> </span><span class="kw">read_copynumber</span>(<span class="st">&quot;../data/CNV_from_sequenza.tsv&quot;</span>,</span>
<span id="cb3-7"><a href="#cb3-7"></a>  <span class="dt">genome_build =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="dt">complement =</span> <span class="ot">FALSE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb3-9"><a href="#cb3-9"></a>)</span>
<span id="cb3-10"><a href="#cb3-10"></a></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co"># remove WCMC160-SRR3146971 with only one CNV</span></span>
<span id="cb3-12"><a href="#cb3-12"></a>CNV.seqz &lt;-<span class="st"> </span><span class="kw">subset</span>(CNV.seqz, <span class="dt">subset =</span> <span class="op">!</span>sample <span class="op">%in%</span><span class="st"> &quot;WCMC160-SRR3146971&quot;</span>)</span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="kw">save</span>(CNV.seqz, <span class="dt">file =</span> <span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</span></code></pre></div>
<p>Tally the alteration records.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>ncores &lt;-<span class="st"> </span><span class="dv">20</span> <span class="co"># Set a proper number of thread</span></span>
<span id="cb4-2"><a href="#cb4-2"></a></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co"># Use classfication method devised by me (&quot;W&quot;)</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>CNV.seqz.tally.W &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(CNV.seqz, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">cores =</span> ncores, <span class="dt">feature_setting =</span> CN.features)</span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="kw">save</span>(CNV.seqz.tally.W, <span class="dt">file =</span> <span class="st">&quot;../output/CNV.seqz.tally.W.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="sbs" class="section level4">
<h4>SBS</h4>
<p>Read SBS profile as a <code>MAF</code> object.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># Reading data ------------------------------------------------------------</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>Maf &lt;-<span class="st"> </span>data.table<span class="op">::</span><span class="kw">fread</span>(<span class="st">&quot;/public/data/maf/all.maf&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co"># Remove all NA columns</span></span>
<span id="cb5-4"><a href="#cb5-4"></a>Maf &lt;-<span class="st"> </span>Maf[, <span class="kw">which</span>(<span class="kw">unlist</span>(<span class="kw">lapply</span>(Maf, <span class="cf">function</span>(x) <span class="op">!</span><span class="kw">all</span>(<span class="kw">is.na</span>(x))))), with =<span class="st"> </span>F]</span>
<span id="cb5-5"><a href="#cb5-5"></a>Maf &lt;-<span class="st"> </span><span class="kw">read_maf</span>(Maf)</span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="kw">save</span>(Maf, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span></code></pre></div>
<p>The result <code>MAF</code> object can be used in any analysis provided by R package <strong>maftools</strong> <span class="citation">(Mayakonda and Koeffler <a href="#ref-mayakonda2016maftools" role="doc-biblioref">2016</a>)</span>.</p>
<p>Tally the SBS records.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>Maf.tally &lt;-<span class="st"> </span><span class="kw">sig_tally</span>(Maf,</span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="dt">cores =</span> ncores, <span class="dt">ref_genome =</span> <span class="st">&quot;BSgenome.Hsapiens.UCSC.hg38&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="dt">useSyn =</span> <span class="ot">TRUE</span></span>
<span id="cb6-4"><a href="#cb6-4"></a>)</span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="kw">save</span>(Maf.tally, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.tally.RData&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="estimate-signature-number" class="section level2">
<h2>Estimate signature number</h2>
<p>R package NMF <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible" role="doc-biblioref">2010</a>)</span> is used for running NMF algorithm Factorization rank <code>r</code> in NMF defines the number of signatures used to approximate the target sample-by-component matrix. A common way of deciding on <code>r</code> is to try different values, compute some quality measure of the results, and choose the best value according to this quality criteria.</p>
<p>As suggested, performing 30-50 runs is considered sufficient to get a robust estimate of the <code>r</code> value. Here we perform 50 runs.</p>
<div id="copy-number-1" class="section level3">
<h3>Copy number</h3>
<p>Estimate the number of copy number signature from 2 to 12.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>EST.seqz.W.all &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(CNV.seqz.tally.W<span class="op">$</span>nmf_matrix,</span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">12</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores, <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="dt">save_plots =</span> <span class="ot">FALSE</span>,</span>
<span id="cb7-4"><a href="#cb7-4"></a>  <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb7-5"><a href="#cb7-5"></a>)</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="kw">save</span>(EST.seqz.W.all, <span class="dt">file =</span> <span class="st">&quot;../output/EST.seqz.W.all.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="sbs-1" class="section level3">
<h3>SBS</h3>
<p>Estimate the number of SBS signature from 2 to 10.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># Remove the effect of hyper mutated samples (not removing hyper-mutated samples)</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="co"># The idead is adopted from SignatureAnalyzer package</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>nmf_matrix &lt;-<span class="st"> </span><span class="kw">handle_hyper_mutation</span>(Maf.tally<span class="op">$</span>nmf_matrix)</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a>EST.Maf &lt;-<span class="st"> </span><span class="kw">sig_estimate</span>(nmf_matrix,</span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="dt">range =</span> <span class="dv">2</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores, <span class="dt">use_random =</span> <span class="ot">TRUE</span>,</span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="dt">save_plots =</span> <span class="ot">FALSE</span>,</span>
<span id="cb8-8"><a href="#cb8-8"></a>  <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>)</span>
<span id="cb8-10"><a href="#cb8-10"></a></span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="kw">save</span>(EST.Maf, <span class="dt">file =</span> <span class="st">&quot;../output/EST.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="kw">save</span>(nmf_matrix, <span class="dt">file =</span> <span class="st">&quot;../output/Maf_matrix.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="result" class="section level3">
<h3>Result</h3>
<p>Let’s load the estimation results.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/EST.seqz.W.all.RData&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/EST.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span></code></pre></div>
<p>The most common approach is to use the cophenetic correlation coefficient. I show the measure vs. signature number as below.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">show_sig_number_survey</span>(EST.seqz.W.all, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" width="2400" /></p>
<p>The best possible signature number value is the one at which the cophenetic correlation value on the y-axis drops significantly <span class="citation">(Gaujoux and Seoighe <a href="#ref-gaujoux2010flexible" role="doc-biblioref">2010</a>)</span>. Here we can take that the value <code>5</code> is the optimal signature number. For <code>3</code>, it is small to produce the result with biological significance due to the hugo contribution of CNV in prostate cancer; for <code>8</code>, it is a little big, so it may be not easy to understand the result considering currently we know few about the mechanism and consequence of copy number events.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>p &lt;-<span class="st"> </span><span class="kw">show_sig_number_survey</span>(EST.seqz.W.all, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">add_h_arrow</span>(p, <span class="dt">x =</span> <span class="fl">5.1</span>, <span class="dt">y =</span> <span class="fl">0.99</span>, <span class="dt">seg_len =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-19-1.png" width="2400" /></p>
<p>Next we take a look at SBS.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>p &lt;-<span class="st"> </span><span class="kw">show_sig_number_survey</span>(EST.Maf, <span class="dt">right_y =</span> <span class="ot">NULL</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">add_h_arrow</span>(p, <span class="dt">x =</span> <span class="fl">3.1</span>, <span class="dt">y =</span> <span class="fl">0.982</span>, <span class="dt">seg_len =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="2400" /></p>
<p>For SBS, we select <code>3</code>. We have the knowledge that <code>3</code> mutational signatures from TCGA prostate cancers are recorded in COSMIC database (COSMIC 1, 5, 6), and the cophenetic has a sharp decrease after <code>3</code> indicating that substantially less stability is achieved using more than <code>3</code> clusters.</p>
</div>
</div>
<div id="extract-signatures" class="section level2">
<h2>Extract signatures</h2>
<p>Now that we have determined the signature number, we then extract the signatures with 50 runs.</p>
<p>Copy number:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># Extract signatures ------------------------------------------------------</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>Sig.CNV.seqz.W &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(CNV.seqz.tally.W<span class="op">$</span>nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">5</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores)</span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="kw">save</span>(Sig.CNV.seqz.W, <span class="dt">file =</span> <span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</span></code></pre></div>
<p>SBS:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># Extract signatures ------------------------------------------------------</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>Sig.SNV &lt;-<span class="st"> </span><span class="kw">sig_extract</span>(nmf_matrix, <span class="dt">n_sig =</span> <span class="dv">3</span>, <span class="dt">nrun =</span> <span class="dv">50</span>, <span class="dt">cores =</span> ncores)</span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="kw">save</span>(Sig.SNV, <span class="dt">file =</span> <span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="show-signature-profile-and-aetiology" class="section level2">
<h2>Show signature profile and aetiology</h2>
<p>Now that we have the extracted signatures, we can show the signature profile.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw">load</span>(<span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</span></code></pre></div>
<div id="sbs-signature-profile" class="section level3">
<h3>SBS signature profile</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="co"># I provide two style &#39;default&#39; and &#39;cosmic&#39; in sigminer</span></span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="kw">show_sig_profile</span>(Sig.SNV, <span class="dt">mode =</span> <span class="st">&quot;SBS&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">x_label_angle =</span> <span class="dv">90</span>, <span class="dt">x_label_vjust =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-24-1.png" width="3000" /></p>
<blockquote>
<p>SBS signatures are displayed based on the observed component frequency of the human genome, i.e., representing the relative proportions of mutations generated by each signature based on the actual trinucleotide frequencies of the reference human genome.</p>
</blockquote>
<p>The aetiologies of SBS signatures is more clear than those of copy number signatures. We can find the aetiologies of the 3 signatures above by computing their cosine similarity to COSMIC reference signatures.</p>
<p>We compare the 3 signature to COSMIC signature database v2 and v3.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>sim_v2 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(Sig.SNV, <span class="dt">sig_db =</span> <span class="st">&quot;legacy&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="co">#&gt; -Comparing against COSMIC signatures</span></span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="co">#&gt; --Found Sig1 most similar to COSMIC_3</span></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="co">#&gt;    Aetiology: defects in DNA-DSB repair by HR [similarity: 0.844]</span></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="co">#&gt; --Found Sig2 most similar to COSMIC_15</span></span>
<span id="cb17-7"><a href="#cb17-7"></a><span class="co">#&gt;    Aetiology: defective DNA mismatch repair [similarity: 0.943]</span></span>
<span id="cb17-8"><a href="#cb17-8"></a><span class="co">#&gt; --Found Sig3 most similar to COSMIC_1</span></span>
<span id="cb17-9"><a href="#cb17-9"></a><span class="co">#&gt;    Aetiology: spontaneous deamination of 5-methylcytosine [similarity: 0.935]</span></span>
<span id="cb17-10"><a href="#cb17-10"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb17-11"><a href="#cb17-11"></a><span class="co">#&gt; Return result invisiblely.</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>sim_v3 &lt;-<span class="st"> </span><span class="kw">get_sig_similarity</span>(Sig.SNV, <span class="dt">sig_db =</span> <span class="st">&quot;SBS&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="co">#&gt; -Comparing against COSMIC signatures</span></span>
<span id="cb18-3"><a href="#cb18-3"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb18-4"><a href="#cb18-4"></a><span class="co">#&gt; --Found Sig1 most similar to SBS40</span></span>
<span id="cb18-5"><a href="#cb18-5"></a><span class="co">#&gt;    Aetiology: Unknown [similarity: 0.883]</span></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="co">#&gt; --Found Sig2 most similar to SBS15</span></span>
<span id="cb18-7"><a href="#cb18-7"></a><span class="co">#&gt;    Aetiology: Defective DNA mismatch repair [similarity: 0.967]</span></span>
<span id="cb18-8"><a href="#cb18-8"></a><span class="co">#&gt; --Found Sig3 most similar to SBS1</span></span>
<span id="cb18-9"><a href="#cb18-9"></a><span class="co">#&gt;    Aetiology: spontaneous or enzymatic deamination of 5-methylcytosine [similarity: 0.966]</span></span>
<span id="cb18-10"><a href="#cb18-10"></a><span class="co">#&gt; ------------------------------------</span></span>
<span id="cb18-11"><a href="#cb18-11"></a><span class="co">#&gt; Return result invisiblely.</span></span></code></pre></div>
<p>From the results above, basically we can confirm that aetiology of <code>Sig2</code> is <strong>dMMR</strong> (defects in mismatch repair) and the aetiology of <code>Sig3</code> is <strong>aging</strong>. The aetiology of <code>Sig1</code> may be <strong>HRD</strong> (Defective homologous recombination-based DNA damage repair) or an unknown reason.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># I provide two style &#39;default&#39; and &#39;cosmic&#39; in sigminer</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>p &lt;-<span class="st"> </span><span class="kw">show_sig_profile</span>(Sig.SNV, <span class="dt">mode =</span> <span class="st">&quot;SBS&quot;</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">x_label_angle =</span> <span class="dv">90</span>, <span class="dt">x_label_vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb19-3"><a href="#cb19-3"></a><span class="kw">add_labels</span>(p,</span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="dt">x =</span> <span class="fl">0.92</span>, <span class="dt">y =</span> <span class="fl">0.3</span>, <span class="dt">y_end =</span> <span class="fl">0.85</span>, <span class="dt">n_label =</span> <span class="dv">3</span>,</span>
<span id="cb19-5"><a href="#cb19-5"></a>  <span class="dt">labels =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;HRD or unknown&quot;</span>, <span class="st">&quot;dMMR&quot;</span>, <span class="st">&quot;Aging&quot;</span>)), <span class="dt">hjust =</span> <span class="dv">1</span></span>
<span id="cb19-6"><a href="#cb19-6"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-27-1.png" width="3000" /></p>
<p>The exposure of signature in each sample is plotted as the below.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="kw">show_sig_exposure</span>(Sig.SNV, <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-28-1.png" width="2400" /></p>
<p>Set a cutoff 2000 to remove some outliers.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="kw">show_sig_exposure</span>(Sig.SNV,</span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="dt">cutoff =</span> <span class="dv">2000</span>,</span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span></span>
<span id="cb21-4"><a href="#cb21-4"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="2400" /></p>
<p>We can clearly see that</p>
<ul>
<li>most of samples have very few SBS signature exposures</li>
<li>Sig1 and Sig3 are the major resources</li>
</ul>
</div>
<div id="copy-number-signature-profile" class="section level3">
<h3>Copy number signature profile</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="kw">show_sig_profile</span>(Sig.CNV.seqz.W, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;W&quot;</span>, <span class="dt">normalize =</span> <span class="st">&quot;feature&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-30-1.png" width="3600" /></p>
<blockquote>
<p>Similar to SBS signatures, copy number signatures are displayed based on the observed component frequency of the human genome. Of note, considering the count of each feature is independent, the profile is row normalized by each feature, unlike <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span> did column normalized (this method is easy to mislead readers), so the bar height can be compared within/between feature</p>
</blockquote>
<p>There is no reference database for copy number signatures, we cannot get aetiologies of the 5 signatures by similarity computation. By reading the description of previous reports <span class="citation">(Macintyre et al. <a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>; Yi and Ju <a href="#ref-yi2018patterns" role="doc-biblioref">2018</a>)</span> and studying the signature profile carefully, we determine or limite the aetiologies of the 5 signatures with also the downstream analyses supported.</p>
<ul>
<li>Sig1 - Focal amplification due to failure of cell cyle control
<ul>
<li>large copy number changes resulting high copy number states</li>
</ul></li>
<li>Sig2 - HRD;MMBIR;Tandem duplication through CDK12 inactivation
<ul>
<li>match the pattern of MMBIR (microhomology mediated break-induced replication): many long OsCN with copy gains</li>
<li>related to CDK12 mutation</li>
<li>high number of breakpoints</li>
<li>small segment size</li>
<li>related to COSMIC 5 (v2) (from <span class="citation">Macintyre et al. (<a href="#ref-macintyre2018copy" role="doc-biblioref">2018</a>)</span>)</li>
</ul></li>
<li>Sig3 - Whole-genome duplication (due to failure of cell cycle control)
<ul>
<li>high segment copy number</li>
<li>few OsCN</li>
<li>global CNV distribution</li>
</ul></li>
<li>Sig4 - Chromothripsis
<ul>
<li>1 copy number deletions resulting considerable small OsCN</li>
<li>large segment size</li>
</ul></li>
<li>Sig5 - BFB, gene fusion and copy number neutral like events including Chromoplexy, translocation, and etc. (these may be early driver events)
<ul>
<li>few CNV with focal distribution</li>
<li>associated with gene fusion</li>
</ul></li>
</ul>
<p>The exposure of signature in each sample is plotted as the below.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="kw">show_sig_exposure</span>(Sig.CNV.seqz.W, <span class="dt">rm_space =</span> <span class="ot">TRUE</span>, <span class="dt">style =</span> <span class="st">&quot;cosmic&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="2400" /></p>
</div>
<div id="show-similarity-matrix" class="section level3">
<h3>Show similarity matrix</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim_v2<span class="op">$</span>similarity,</span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="dt">cluster_rows =</span> F</span>
<span id="cb24-3"><a href="#cb24-3"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-32-1.png" width="3300" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(sim_v3<span class="op">$</span>similarity,</span>
<span id="cb25-2"><a href="#cb25-2"></a>  <span class="dt">cluster_rows =</span> F</span>
<span id="cb25-3"><a href="#cb25-3"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-33-1.png" width="4500" /></p>
<hr />
</div>
</div>
</div>
<div id="association-analysis" class="section level1">
<h1>Association Analysis</h1>
<p>This part has been divided into 2 sections: data integration and association analysis.</p>
<div id="data-integration" class="section level2">
<h2>Data integration</h2>
<p>In this section, I integrate all genotype/phenotype data as a tidy data table used for downstream analyses.</p>
<p>Load packages and prepared data.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="kw">library</span>(sigminer)</span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="kw">library</span>(maftools)</span>
<span id="cb26-4"><a href="#cb26-4"></a></span>
<span id="cb26-5"><a href="#cb26-5"></a><span class="co"># Loading clinical related data -------------------------------------------</span></span>
<span id="cb26-6"><a href="#cb26-6"></a></span>
<span id="cb26-7"><a href="#cb26-7"></a>Info &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;../data/PRAD_CLINICAL.rds&quot;</span>)</span>
<span id="cb26-8"><a href="#cb26-8"></a><span class="co"># Purity and ploidy info from facets</span></span>
<span id="cb26-9"><a href="#cb26-9"></a>PurityInfo &lt;-<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/PRAD_Purity_and_Ploidy_Sequenza.tsv&quot;</span>)</span>
<span id="cb26-10"><a href="#cb26-10"></a></span>
<span id="cb26-11"><a href="#cb26-11"></a><span class="co"># Processing CNV data -----------------------------------------------------</span></span>
<span id="cb26-12"><a href="#cb26-12"></a></span>
<span id="cb26-13"><a href="#cb26-13"></a><span class="kw">load</span>(<span class="st">&quot;../output/CNV.seqz.RData&quot;</span>)</span>
<span id="cb26-14"><a href="#cb26-14"></a><span class="kw">load</span>(<span class="st">&quot;../output/Sig.CNV.seqz.W.RData&quot;</span>)</span>
<span id="cb26-15"><a href="#cb26-15"></a>CNV &lt;-<span class="st"> </span>CNV.seqz</span>
<span id="cb26-16"><a href="#cb26-16"></a>Sig.CNV &lt;-<span class="st"> </span>Sig.CNV.seqz.W</span>
<span id="cb26-17"><a href="#cb26-17"></a><span class="kw">rm</span>(Sig.CNV.seqz.W, CNV.seqz)</span></code></pre></div>
<p>After running NMF, we can get robust clusters from consensus matrix, more see <code>?NMF::predict</code>.</p>
<p>I also create a function <code>get_sig_exposure()</code> to get the absolute/relative exposure of signatures.</p>
<blockquote>
<p>Relative exposure is more useful for clustering and absolute exposure is more useful for association analysis.</p>
</blockquote>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="co"># CNV</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>CNVGroupInfo &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.CNV, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</span>
<span id="cb27-3"><a href="#cb27-3"></a>CNVInfo &lt;-<span class="st"> </span>CNV<span class="op">@</span>summary.per.sample</span>
<span id="cb27-4"><a href="#cb27-4"></a><span class="co"># Default is absolute exposure</span></span>
<span id="cb27-5"><a href="#cb27-5"></a>CNVExposureInfo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.CNV)</span></code></pre></div>
<p>Next, we process mutation data to get TMB, driver info, clusters, and etc (the detail has been described in R script under <code>analysis/src/</code> of the repo).</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="co"># Processing mutation data ------------------------------------------------</span></span>
<span id="cb28-2"><a href="#cb28-2"></a></span>
<span id="cb28-3"><a href="#cb28-3"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span>
<span id="cb28-4"><a href="#cb28-4"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/Sig.PRAD_TCGA_plus_dbGap_Maf.RData&quot;</span>)</span>
<span id="cb28-5"><a href="#cb28-5"></a></span>
<span id="cb28-6"><a href="#cb28-6"></a>TMBInfo &lt;-<span class="st"> </span><span class="kw">getSampleSummary</span>(Maf)[, .(Tumor_Sample_Barcode, total)]</span>
<span id="cb28-7"><a href="#cb28-7"></a></span>
<span id="cb28-8"><a href="#cb28-8"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_driver_info.RData&quot;</span>)</span>
<span id="cb28-9"><a href="#cb28-9"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_heter_info.RData&quot;</span>)</span>
<span id="cb28-10"><a href="#cb28-10"></a></span>
<span id="cb28-11"><a href="#cb28-11"></a>SNVGroupInfo &lt;-<span class="st"> </span><span class="kw">get_groups</span>(Sig.SNV, <span class="dt">method =</span> <span class="st">&quot;consensus&quot;</span>, <span class="dt">match_consensus =</span> <span class="ot">TRUE</span>)</span>
<span id="cb28-12"><a href="#cb28-12"></a>SNVExposureInfo &lt;-<span class="st"> </span><span class="kw">get_sig_exposure</span>(Sig.SNV)</span>
<span id="cb28-13"><a href="#cb28-13"></a></span>
<span id="cb28-14"><a href="#cb28-14"></a><span class="co"># Processing gene and pathway mutation ------------------------------------</span></span>
<span id="cb28-15"><a href="#cb28-15"></a></span>
<span id="cb28-16"><a href="#cb28-16"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_gene_and_pathway_mutation.RData&quot;</span>)</span></code></pre></div>
<p>Next, we keep keys of all <code>data.frame</code>s are same and merge them into one.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="co"># Merge data --------------------------------------------------------------</span></span>
<span id="cb29-2"><a href="#cb29-2"></a>Info &lt;-<span class="st"> </span>Info <span class="op">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</span>
<span id="cb29-4"><a href="#cb29-4"></a>    <span class="dt">CNV_ID =</span> dplyr<span class="op">::</span><span class="kw">case_when</span>(</span>
<span id="cb29-5"><a href="#cb29-5"></a>      <span class="op">!</span><span class="kw">startsWith</span>(tumor_Run, <span class="st">&quot;TCGA&quot;</span>) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(tumor_Run) <span class="op">~</span><span class="st"> </span><span class="kw">paste</span>(subject_id, tumor_Run, <span class="dt">sep =</span> <span class="st">&quot;-&quot;</span>),</span>
<span id="cb29-6"><a href="#cb29-6"></a>      <span class="kw">startsWith</span>(tumor_Run, <span class="st">&quot;TCGA&quot;</span>) <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw">is.na</span>(tumor_Run) <span class="op">~</span><span class="st"> </span>tumor_Run,</span>
<span id="cb29-7"><a href="#cb29-7"></a>      <span class="ot">TRUE</span> <span class="op">~</span><span class="st"> </span><span class="ot">NA_character_</span></span>
<span id="cb29-8"><a href="#cb29-8"></a>    )</span>
<span id="cb29-9"><a href="#cb29-9"></a>  )</span>
<span id="cb29-10"><a href="#cb29-10"></a></span>
<span id="cb29-11"><a href="#cb29-11"></a>PurityInfo</span>
<span id="cb29-12"><a href="#cb29-12"></a><span class="kw">colnames</span>(CNVGroupInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cnv_group&quot;</span>, <span class="st">&quot;cnv_weight&quot;</span>, <span class="st">&quot;cnv_enrich_sig&quot;</span>)</span>
<span id="cb29-13"><a href="#cb29-13"></a>CNVInfo</span>
<span id="cb29-14"><a href="#cb29-14"></a><span class="kw">colnames</span>(CNVExposureInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;CNV_&quot;</span>, <span class="kw">colnames</span>(CNVExposureInfo)[<span class="op">-</span><span class="dv">1</span>]))</span>
<span id="cb29-15"><a href="#cb29-15"></a><span class="kw">colnames</span>(TMBInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;total_mutation&quot;</span>)</span>
<span id="cb29-16"><a href="#cb29-16"></a>TMBInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-17"><a href="#cb29-17"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-18"><a href="#cb29-18"></a>  sample</span>
<span id="cb29-19"><a href="#cb29-19"></a>)]</span>
<span id="cb29-20"><a href="#cb29-20"></a>DriverDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-21"><a href="#cb29-21"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-22"><a href="#cb29-22"></a>  sample</span>
<span id="cb29-23"><a href="#cb29-23"></a>)]</span>
<span id="cb29-24"><a href="#cb29-24"></a><span class="kw">colnames</span>(SNVGroupInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;snv_group&quot;</span>, <span class="st">&quot;snv_weight&quot;</span>, <span class="st">&quot;snv_enrich_sig&quot;</span>)</span>
<span id="cb29-25"><a href="#cb29-25"></a>SNVGroupInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-26"><a href="#cb29-26"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-27"><a href="#cb29-27"></a>  sample</span>
<span id="cb29-28"><a href="#cb29-28"></a>)]</span>
<span id="cb29-29"><a href="#cb29-29"></a><span class="kw">colnames</span>(SNVExposureInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="kw">paste0</span>(<span class="st">&quot;SNV_&quot;</span>, <span class="kw">colnames</span>(SNVExposureInfo)[<span class="op">-</span><span class="dv">1</span>]))</span>
<span id="cb29-30"><a href="#cb29-30"></a>SNVExposureInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-31"><a href="#cb29-31"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-32"><a href="#cb29-32"></a>  sample</span>
<span id="cb29-33"><a href="#cb29-33"></a>)]</span>
<span id="cb29-34"><a href="#cb29-34"></a><span class="kw">colnames</span>(TitvInfo) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>)</span>
<span id="cb29-35"><a href="#cb29-35"></a>TitvInfo[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-36"><a href="#cb29-36"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-37"><a href="#cb29-37"></a>  sample</span>
<span id="cb29-38"><a href="#cb29-38"></a>)]</span>
<span id="cb29-39"><a href="#cb29-39"></a><span class="kw">colnames</span>(MathDF) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;MATH&quot;</span>)</span>
<span id="cb29-40"><a href="#cb29-40"></a>MathDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-41"><a href="#cb29-41"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-42"><a href="#cb29-42"></a>  sample</span>
<span id="cb29-43"><a href="#cb29-43"></a>)]</span>
<span id="cb29-44"><a href="#cb29-44"></a><span class="kw">colnames</span>(ClusterDF) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cluster&quot;</span>)</span>
<span id="cb29-45"><a href="#cb29-45"></a>ClusterDF[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-46"><a href="#cb29-46"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-47"><a href="#cb29-47"></a>  sample</span>
<span id="cb29-48"><a href="#cb29-48"></a>)]</span>
<span id="cb29-49"><a href="#cb29-49"></a></span>
<span id="cb29-50"><a href="#cb29-50"></a>data.table<span class="op">::</span><span class="kw">setDT</span>(summary_mutation)</span>
<span id="cb29-51"><a href="#cb29-51"></a>data.table<span class="op">::</span><span class="kw">setDT</span>(summary_pathway)</span>
<span id="cb29-52"><a href="#cb29-52"></a></span>
<span id="cb29-53"><a href="#cb29-53"></a><span class="kw">colnames</span>(summary_mutation)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;sample&quot;</span></span>
<span id="cb29-54"><a href="#cb29-54"></a><span class="kw">colnames</span>(summary_pathway)[<span class="dv">1</span>] &lt;-<span class="st"> &quot;sample&quot;</span></span>
<span id="cb29-55"><a href="#cb29-55"></a>summary_mutation[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-56"><a href="#cb29-56"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-57"><a href="#cb29-57"></a>  sample</span>
<span id="cb29-58"><a href="#cb29-58"></a>)]</span>
<span id="cb29-59"><a href="#cb29-59"></a>summary_pathway[, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">as.character</span>(sample)][, sample <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">startsWith</span>(sample, <span class="st">&quot;TCGA&quot;</span>),</span>
<span id="cb29-60"><a href="#cb29-60"></a>  <span class="kw">substr</span>(sample, <span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb29-61"><a href="#cb29-61"></a>  sample</span>
<span id="cb29-62"><a href="#cb29-62"></a>)]</span>
<span id="cb29-63"><a href="#cb29-63"></a></span>
<span id="cb29-64"><a href="#cb29-64"></a></span>
<span id="cb29-65"><a href="#cb29-65"></a>MergeInfo &lt;-<span class="st"> </span>Info <span class="op">%&gt;%</span></span>
<span id="cb29-66"><a href="#cb29-66"></a><span class="st">  </span><span class="kw">left_join</span>(CNVInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-67"><a href="#cb29-67"></a><span class="st">  </span><span class="kw">left_join</span>(CNVGroupInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-68"><a href="#cb29-68"></a><span class="st">  </span><span class="kw">left_join</span>(CNVExposureInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-69"><a href="#cb29-69"></a><span class="st">  </span><span class="kw">left_join</span>(PurityInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;CNV_ID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-70"><a href="#cb29-70"></a><span class="st">  </span><span class="kw">left_join</span>(summary_mutation, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-71"><a href="#cb29-71"></a><span class="st">  </span><span class="kw">left_join</span>(summary_pathway, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-72"><a href="#cb29-72"></a><span class="st">  </span><span class="kw">left_join</span>(TMBInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-73"><a href="#cb29-73"></a><span class="st">  </span><span class="kw">left_join</span>(DriverDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-74"><a href="#cb29-74"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</span>
<span id="cb29-75"><a href="#cb29-75"></a>    <span class="dt">n_driver =</span> <span class="kw">ifelse</span>(<span class="op">!</span><span class="kw">is.na</span>(n_driver), n_driver, <span class="dv">0</span>)</span>
<span id="cb29-76"><a href="#cb29-76"></a>  ) <span class="op">%&gt;%</span></span>
<span id="cb29-77"><a href="#cb29-77"></a><span class="st">  </span><span class="kw">left_join</span>(TitvInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-78"><a href="#cb29-78"></a><span class="st">  </span><span class="kw">left_join</span>(MathDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-79"><a href="#cb29-79"></a><span class="st">  </span><span class="kw">left_join</span>(ClusterDF, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-80"><a href="#cb29-80"></a><span class="st">  </span><span class="kw">left_join</span>(SNVGroupInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-81"><a href="#cb29-81"></a><span class="st">  </span><span class="kw">left_join</span>(SNVExposureInfo, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;tumor_Run&quot;</span> =<span class="st"> &quot;sample&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb29-82"><a href="#cb29-82"></a><span class="st">  </span><span class="kw">mutate</span>(</span>
<span id="cb29-83"><a href="#cb29-83"></a>    <span class="dt">Stage =</span> <span class="kw">factor</span>(Stage, <span class="dt">ordered =</span> <span class="ot">TRUE</span>),</span>
<span id="cb29-84"><a href="#cb29-84"></a>    <span class="dt">Fusion =</span> <span class="kw">ifelse</span>(Fusion <span class="op">==</span><span class="st"> &quot;Negative&quot;</span>, <span class="st">&quot;No&quot;</span>, <span class="st">&quot;Yes&quot;</span>),</span>
<span id="cb29-85"><a href="#cb29-85"></a>    <span class="dt">sample_type =</span> <span class="kw">ifelse</span>(sample_type <span class="op">==</span><span class="st"> &quot;Unknown&quot;</span>, <span class="ot">NA_character_</span>, sample_type),</span>
<span id="cb29-86"><a href="#cb29-86"></a>    <span class="dt">HasFusion =</span> Fusion,</span>
<span id="cb29-87"><a href="#cb29-87"></a>    <span class="dt">HasFusion =</span> <span class="kw">ifelse</span>(HasFusion <span class="op">==</span><span class="st"> &quot;Yes&quot;</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>),</span>
<span id="cb29-88"><a href="#cb29-88"></a>    <span class="dt">IsMetastatic =</span> <span class="kw">ifelse</span>(sample_type <span class="op">==</span><span class="st"> &quot;Metastatic&quot;</span>, <span class="ot">TRUE</span>, <span class="ot">FALSE</span>)</span>
<span id="cb29-89"><a href="#cb29-89"></a>  )</span>
<span id="cb29-90"><a href="#cb29-90"></a></span>
<span id="cb29-91"><a href="#cb29-91"></a><span class="kw">summary</span>(MergeInfo)</span>
<span id="cb29-92"><a href="#cb29-92"></a><span class="kw">saveRDS</span>(MergeInfo, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="association-analysis-1" class="section level2">
<h2>Association analysis</h2>
<p>Now we do association analysis between signatures (via exposure) and genotypes/phenotypes.</p>
<div id="prepare" class="section level3">
<h3>Prepare</h3>
<p>Load packages.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb30-2"><a href="#cb30-2"></a><span class="kw">library</span>(sigminer)</span></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</span></code></pre></div>
<p>Set the coloumns used for analysis.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="co"># Driver genes identified by MutSig software</span></span>
<span id="cb32-3"><a href="#cb32-3"></a>cols_to_mutated_genes &lt;-<span class="st"> </span><span class="kw">colnames</span>(df.seqz)[<span class="dv">29</span><span class="op">:</span><span class="dv">75</span>]</span>
<span id="cb32-4"><a href="#cb32-4"></a><span class="co"># 10 oncogenic pathways in TCGA 2018 papers + HR pathway</span></span>
<span id="cb32-5"><a href="#cb32-5"></a>cols_to_mutated_pathways &lt;-<span class="st"> </span><span class="kw">colnames</span>(df.seqz)[<span class="dv">76</span><span class="op">:</span><span class="dv">86</span>]</span>
<span id="cb32-6"><a href="#cb32-6"></a></span>
<span id="cb32-7"><a href="#cb32-7"></a><span class="co"># Exclude PSA</span></span>
<span id="cb32-8"><a href="#cb32-8"></a>cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb32-9"><a href="#cb32-9"></a>  <span class="st">&quot;IsMetastatic&quot;</span>, <span class="st">&quot;HasFusion&quot;</span>,</span>
<span id="cb32-10"><a href="#cb32-10"></a>  <span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</span>
<span id="cb32-11"><a href="#cb32-11"></a>  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</span>
<span id="cb32-12"><a href="#cb32-12"></a>  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</span>
<span id="cb32-13"><a href="#cb32-13"></a>  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></span>
<span id="cb32-14"><a href="#cb32-14"></a>)</span>
<span id="cb32-15"><a href="#cb32-15"></a></span>
<span id="cb32-16"><a href="#cb32-16"></a>feature_type &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;ca&quot;</span>, <span class="dv">2</span>), <span class="kw">rep</span>(<span class="st">&quot;co&quot;</span>, <span class="dv">16</span>))</span></code></pre></div>
</div>
<div id="signatures-features" class="section level3">
<h3>Signatures &amp; features</h3>
<p>Analyze the association between signatures and features.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>tidy_data.seqz.feature &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</span>
<span id="cb33-2"><a href="#cb33-2"></a>  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</span>
<span id="cb33-3"><a href="#cb33-3"></a>  <span class="dt">cols_to_features =</span> cols_to_features,</span>
<span id="cb33-4"><a href="#cb33-4"></a>  <span class="dt">method_co =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb33-5"><a href="#cb33-5"></a>  <span class="dt">type =</span> feature_type, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb33-6"><a href="#cb33-6"></a>) <span class="op">%&gt;%</span></span>
<span id="cb33-7"><a href="#cb33-7"></a><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>) <span class="co"># adjust p value with FDR</span></span></code></pre></div>
<blockquote>
<p>Association of signature exposures with other features was performed using one of two procedures: for a continuous association variable (including ordinal variable), pearson correaltion was performed; for a binary association variable, samples were divided into two groups and Mann-Whitney U-test was performed to test for differences in signature exposure medians between the two groups.</p>
</blockquote>
<p>Show the result.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.feature)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-42-1.png" width="2700" /></p>
<blockquote>
<p>All shown circles are statistically significant results (FDR &lt; 0.05). A feature will be chopped off from the plot if has no correlation with any signatures. Same for the similar plots below.</p>
</blockquote>
</div>
<div id="signatures-mutated-genes" class="section level3">
<h3>Signatures &amp; mutated genes</h3>
<p>Analyze the association between signatures and mutated genes (only driver genes identified by MutSig).</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>tidy_data.seqz.gene &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</span>
<span id="cb35-2"><a href="#cb35-2"></a>  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</span>
<span id="cb35-3"><a href="#cb35-3"></a>  <span class="dt">cols_to_features =</span> cols_to_mutated_genes,</span>
<span id="cb35-4"><a href="#cb35-4"></a>  <span class="dt">type =</span> <span class="st">&quot;ca&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb35-5"><a href="#cb35-5"></a>) <span class="op">%&gt;%</span></span>
<span id="cb35-6"><a href="#cb35-6"></a><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.gene, <span class="dt">ylab =</span> <span class="st">&quot;Mutated genes&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-44-1.png" width="2700" /></p>
</div>
<div id="signatures-mutated-pathways" class="section level3">
<h3>Signatures &amp; mutated pathways</h3>
<p>Analyze the association between signatures and mutated pathways.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>tidy_data.seqz.pathways &lt;-<span class="st"> </span><span class="kw">get_sig_feature_association</span>(df.seqz,</span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</span>
<span id="cb37-3"><a href="#cb37-3"></a>  <span class="dt">cols_to_features =</span> cols_to_mutated_pathways,</span>
<span id="cb37-4"><a href="#cb37-4"></a>  <span class="dt">type =</span> <span class="st">&quot;ca&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb37-5"><a href="#cb37-5"></a>) <span class="op">%&gt;%</span></span>
<span id="cb37-6"><a href="#cb37-6"></a><span class="st">  </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a><span class="kw">show_sig_feature_corrplot</span>(tidy_data.seqz.pathways, <span class="dt">ylab =</span> <span class="st">&quot;Mutated pathways&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-46-1.png" width="2700" /></p>
</div>
<div id="correlation-network" class="section level3">
<h3>Correlation network</h3>
<p>To determine the structure of relationship between signatures and features, we use continuous variables to do correlation network analysis.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>f_dt &lt;-<span class="st"> </span>df.seqz[, <span class="kw">c</span>(cols_to_sigs.seqz, cols_to_features[<span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">-2</span>)])] <span class="op">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.ordered, as.numeric)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="kw">library</span>(corrr)</span>
<span id="cb40-2"><a href="#cb40-2"></a></span>
<span id="cb40-3"><a href="#cb40-3"></a>res.cor &lt;-<span class="st"> </span><span class="kw">correlate</span>(f_dt,</span>
<span id="cb40-4"><a href="#cb40-4"></a>  <span class="dt">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>,</span>
<span id="cb40-5"><a href="#cb40-5"></a>  <span class="dt">method =</span> <span class="st">&quot;pearson&quot;</span></span>
<span id="cb40-6"><a href="#cb40-6"></a>) <span class="op">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_if</span>(is.numeric, dplyr<span class="op">::</span>coalesce, <span class="dv">0</span>)</span>
<span id="cb40-8"><a href="#cb40-8"></a></span>
<span id="cb40-9"><a href="#cb40-9"></a><span class="co">## Avoid corrr error</span></span>
<span id="cb40-10"><a href="#cb40-10"></a><span class="co">## https://github.com/tidymodels/corrr/issues/86</span></span>
<span id="cb40-11"><a href="#cb40-11"></a><span class="co">## https://github.com/tidymodels/corrr/issues/34</span></span>
<span id="cb40-12"><a href="#cb40-12"></a>res.cor[, <span class="dv">-1</span>] &lt;-<span class="st"> </span>res.cor[, <span class="dv">-1</span>] <span class="op">*</span><span class="st"> </span><span class="fl">0.99</span></span>
<span id="cb40-13"><a href="#cb40-13"></a></span>
<span id="cb40-14"><a href="#cb40-14"></a>p &lt;-<span class="st"> </span>res.cor <span class="op">%&gt;%</span></span>
<span id="cb40-15"><a href="#cb40-15"></a><span class="st">  </span><span class="kw">network_plot</span>(</span>
<span id="cb40-16"><a href="#cb40-16"></a>    <span class="dt">min_cor =</span> <span class="fl">0.2</span>,</span>
<span id="cb40-17"><a href="#cb40-17"></a>    <span class="dt">colours =</span> <span class="kw">rev</span>(<span class="kw">c</span>(<span class="st">&quot;indianred2&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;skyblue1&quot;</span>))</span>
<span id="cb40-18"><a href="#cb40-18"></a>  )</span>
<span id="cb40-19"><a href="#cb40-19"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">scale_size</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="dv">2</span>))</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-48-1.png" width="2400" /></p>
<blockquote>
<p>variables that are more highly correlated appear closer together and are joined by stronger paths. Paths are also colored by their sign. The proximity of the points are determined using multidimensional clustering.</p>
</blockquote>
</div>
<div id="signatures-psa" class="section level3">
<h3>Signatures &amp; PSA</h3>
<p>In all datasets we collected, 3 of them have PSA info. However, the range of PSA value in 3 datasets differs, so I excluded PSA from features above and analyze it independently.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>df.seqz.psa &lt;-<span class="st"> </span>df.seqz <span class="op">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(PSA)) <span class="op">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="kw">contains</span>(<span class="st">&quot;Sig&quot;</span>, <span class="dt">ignore.case =</span> F), <span class="kw">c</span>(<span class="st">&quot;PSA&quot;</span>, <span class="st">&quot;Study&quot;</span>, <span class="st">&quot;CNV_ID&quot;</span>))</span>
<span id="cb41-4"><a href="#cb41-4"></a></span>
<span id="cb41-5"><a href="#cb41-5"></a>df.seqz.psa &lt;-<span class="st"> </span>df.seqz.psa <span class="op">%&gt;%</span></span>
<span id="cb41-6"><a href="#cb41-6"></a><span class="st">  </span>dplyr<span class="op">:::</span><span class="kw">group_by</span>(Study) <span class="op">%&gt;%</span></span>
<span id="cb41-7"><a href="#cb41-7"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">nest</span>() <span class="op">%&gt;%</span></span>
<span id="cb41-8"><a href="#cb41-8"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarise</span>(</span>
<span id="cb41-9"><a href="#cb41-9"></a>    <span class="dt">data =</span> purrr<span class="op">:::</span><span class="kw">map</span>(data, <span class="cf">function</span>(x) {</span>
<span id="cb41-10"><a href="#cb41-10"></a>      <span class="kw">get_sig_feature_association</span>(x,</span>
<span id="cb41-11"><a href="#cb41-11"></a>        <span class="dt">cols_to_sigs =</span> cols_to_sigs.seqz,</span>
<span id="cb41-12"><a href="#cb41-12"></a>        <span class="dt">cols_to_features =</span> <span class="st">&quot;PSA&quot;</span>,</span>
<span id="cb41-13"><a href="#cb41-13"></a>        <span class="dt">method_co =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb41-14"><a href="#cb41-14"></a>        <span class="dt">type =</span> <span class="st">&quot;co&quot;</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span></span>
<span id="cb41-15"><a href="#cb41-15"></a>      ) <span class="op">%&gt;%</span></span>
<span id="cb41-16"><a href="#cb41-16"></a><span class="st">        </span><span class="kw">get_tidy_association</span>(<span class="dt">p_adjust =</span> <span class="ot">TRUE</span>)</span>
<span id="cb41-17"><a href="#cb41-17"></a>    })</span>
<span id="cb41-18"><a href="#cb41-18"></a>  ) <span class="op">%&gt;%</span></span>
<span id="cb41-19"><a href="#cb41-19"></a><span class="st">  </span>tidyr<span class="op">::</span><span class="kw">unnest</span>(<span class="st">&quot;data&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb41-20"><a href="#cb41-20"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(feature <span class="op">==</span><span class="st"> &quot;PSA&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb41-21"><a href="#cb41-21"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">feature =</span> Study)</span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="kw">show_sig_feature_corrplot</span>(df.seqz.psa, <span class="dt">p_val =</span> <span class="dv">1</span>, <span class="dt">breaks_count =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">50</span>, <span class="dv">100</span>, <span class="dv">150</span>, <span class="dv">200</span>), <span class="dt">ylab =</span> <span class="st">&quot;PSA&quot;</span>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-50-1.png" width="2700" /></p>
<p>Alought we can observe the statistically significant correlation between signatures and PSA, the results are inconsistent across different studies. This may be explained by the dynamic property of PSA while not the sample type due to most of them are primary.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a><span class="kw">table</span>(df.seqz<span class="op">$</span>sample_type, df.seqz<span class="op">$</span>Study)</span>
<span id="cb43-2"><a href="#cb43-2"></a><span class="co">#&gt;             </span></span>
<span id="cb43-3"><a href="#cb43-3"></a><span class="co">#&gt;              phs000447 phs000554 phs000909 phs000915 phs001141 TCGA</span></span>
<span id="cb43-4"><a href="#cb43-4"></a><span class="co">#&gt;   Metastatic         0        50        70       147        77    1</span></span>
<span id="cb43-5"><a href="#cb43-5"></a><span class="co">#&gt;   Primary          112        11        28         0         0  500</span></span></code></pre></div>
</div>
</div>
</div>
<div id="group-analysis" class="section level1">
<h1>Group Analysis</h1>
<p>Load packages.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb44-2"><a href="#cb44-2"></a><span class="kw">library</span>(sigminer)</span></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</span></code></pre></div>
<div id="group-comparison" class="section level2">
<h2>Group comparison</h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb46-2"><a href="#cb46-2"></a>  <span class="st">&quot;IsMetastatic&quot;</span>, <span class="st">&quot;HasFusion&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</span>
<span id="cb46-3"><a href="#cb46-3"></a>  <span class="st">&quot;Age&quot;</span>,</span>
<span id="cb46-4"><a href="#cb46-4"></a>  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</span>
<span id="cb46-5"><a href="#cb46-5"></a>  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</span>
<span id="cb46-6"><a href="#cb46-6"></a>  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></span>
<span id="cb46-7"><a href="#cb46-7"></a>)</span>
<span id="cb46-8"><a href="#cb46-8"></a></span>
<span id="cb46-9"><a href="#cb46-9"></a>feature_type &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;ca&quot;</span>, <span class="dv">4</span>), <span class="kw">rep</span>(<span class="st">&quot;co&quot;</span>, <span class="dv">14</span>))</span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a>df.seqz<span class="op">$</span>GleasonScore &lt;-<span class="st"> </span><span class="kw">as.character</span>(df.seqz<span class="op">$</span>GleasonScore)</span>
<span id="cb47-2"><a href="#cb47-2"></a>df.seqz<span class="op">$</span>GleasonScore &lt;-<span class="st"> </span><span class="kw">factor</span>(df.seqz<span class="op">$</span>GleasonScore, <span class="dt">levels =</span> <span class="kw">as.character</span>(<span class="dv">6</span><span class="op">:</span><span class="dv">10</span>))</span>
<span id="cb47-3"><a href="#cb47-3"></a></span>
<span id="cb47-4"><a href="#cb47-4"></a>df.seqz2 &lt;-<span class="st"> </span>df.seqz</span>
<span id="cb47-5"><a href="#cb47-5"></a>df.seqz2<span class="op">$</span>cnv_enrich_sig &lt;-<span class="st"> </span><span class="kw">ifelse</span>(</span>
<span id="cb47-6"><a href="#cb47-6"></a>  <span class="op">!</span><span class="kw">is.na</span>(df.seqz2<span class="op">$</span>cnv_enrich_sig),</span>
<span id="cb47-7"><a href="#cb47-7"></a>  <span class="kw">paste</span>(df.seqz2<span class="op">$</span>cnv_enrich_sig, <span class="st">&quot;dominant&quot;</span>),</span>
<span id="cb47-8"><a href="#cb47-8"></a>  <span class="ot">NA</span></span>
<span id="cb47-9"><a href="#cb47-9"></a>)</span>
<span id="cb47-10"><a href="#cb47-10"></a></span>
<span id="cb47-11"><a href="#cb47-11"></a>grp.cnv &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(</span>
<span id="cb47-12"><a href="#cb47-12"></a>  df.seqz2,</span>
<span id="cb47-13"><a href="#cb47-13"></a>  <span class="dt">col_group =</span> <span class="st">&quot;cnv_enrich_sig&quot;</span>, <span class="co"># or &quot;cnv_group&quot;</span></span>
<span id="cb47-14"><a href="#cb47-14"></a>  <span class="dt">cols_to_compare =</span> cols_to_features,</span>
<span id="cb47-15"><a href="#cb47-15"></a>  <span class="dt">type =</span> feature_type</span>
<span id="cb47-16"><a href="#cb47-16"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>plot.cnv &lt;-<span class="st"> </span><span class="kw">show_group_comparison</span>(</span>
<span id="cb48-2"><a href="#cb48-2"></a>  grp.cnv,</span>
<span id="cb48-3"><a href="#cb48-3"></a>  <span class="dt">xlab =</span> <span class="ot">NA</span>,</span>
<span id="cb48-4"><a href="#cb48-4"></a>  <span class="dt">method =</span> <span class="st">&quot;anova&quot;</span>,</span>
<span id="cb48-5"><a href="#cb48-5"></a>  <span class="dt">text_angle_x =</span> <span class="dv">60</span>,</span>
<span id="cb48-6"><a href="#cb48-6"></a>  <span class="dt">text_hjust_x =</span> <span class="dv">1</span>,</span>
<span id="cb48-7"><a href="#cb48-7"></a>  <span class="dt">legend_position_ca =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb48-8"><a href="#cb48-8"></a>  <span class="dt">label.x.npc =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb48-9"><a href="#cb48-9"></a>  <span class="dt">label.y.npc =</span> <span class="st">&quot;top&quot;</span>,</span>
<span id="cb48-10"><a href="#cb48-10"></a>  <span class="dt">label =</span> <span class="st">&quot;p.format&quot;</span>,</span>
<span id="cb48-11"><a href="#cb48-11"></a>  <span class="dt">hjust =</span> <span class="fl">0.5</span></span>
<span id="cb48-12"><a href="#cb48-12"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>plot.cnv<span class="op">$</span>ca_comb</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-57-1.png" width="2400" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>plot.cnv<span class="op">$</span>co_comb</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-58-1.png" width="3000" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>df.seqz2<span class="op">$</span>snv_enrich_sig &lt;-<span class="st"> </span><span class="kw">ifelse</span>(</span>
<span id="cb51-2"><a href="#cb51-2"></a>  <span class="op">!</span><span class="kw">is.na</span>(df.seqz2<span class="op">$</span>snv_enrich_sig),</span>
<span id="cb51-3"><a href="#cb51-3"></a>  <span class="kw">paste</span>(df.seqz2<span class="op">$</span>snv_enrich_sig, <span class="st">&quot;dominant&quot;</span>),</span>
<span id="cb51-4"><a href="#cb51-4"></a>  <span class="ot">NA</span></span>
<span id="cb51-5"><a href="#cb51-5"></a>)</span>
<span id="cb51-6"><a href="#cb51-6"></a></span>
<span id="cb51-7"><a href="#cb51-7"></a>grp.snv &lt;-<span class="st"> </span><span class="kw">get_group_comparison</span>(</span>
<span id="cb51-8"><a href="#cb51-8"></a>  df.seqz2,</span>
<span id="cb51-9"><a href="#cb51-9"></a>  <span class="dt">col_group =</span> <span class="st">&quot;snv_enrich_sig&quot;</span>, <span class="co"># or &quot;snv_group&quot;</span></span>
<span id="cb51-10"><a href="#cb51-10"></a>  <span class="dt">cols_to_compare =</span> cols_to_features,</span>
<span id="cb51-11"><a href="#cb51-11"></a>  <span class="dt">type =</span> feature_type</span>
<span id="cb51-12"><a href="#cb51-12"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>plot.snv &lt;-<span class="st"> </span><span class="kw">show_group_comparison</span>(</span>
<span id="cb52-2"><a href="#cb52-2"></a>  grp.snv,</span>
<span id="cb52-3"><a href="#cb52-3"></a>  <span class="dt">xlab =</span> <span class="ot">NA</span>,</span>
<span id="cb52-4"><a href="#cb52-4"></a>  <span class="dt">method =</span> <span class="st">&quot;anova&quot;</span>,</span>
<span id="cb52-5"><a href="#cb52-5"></a>  <span class="dt">text_angle_x =</span> <span class="dv">60</span>,</span>
<span id="cb52-6"><a href="#cb52-6"></a>  <span class="dt">text_hjust_x =</span> <span class="dv">1</span>,</span>
<span id="cb52-7"><a href="#cb52-7"></a>  <span class="dt">legend_position_ca =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb52-8"><a href="#cb52-8"></a>  <span class="dt">label.x.npc =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb52-9"><a href="#cb52-9"></a>  <span class="dt">label.y.npc =</span> <span class="st">&quot;top&quot;</span>,</span>
<span id="cb52-10"><a href="#cb52-10"></a>  <span class="dt">label =</span> <span class="st">&quot;p.format&quot;</span>,</span>
<span id="cb52-11"><a href="#cb52-11"></a>  <span class="dt">hjust =</span> <span class="fl">0.5</span></span>
<span id="cb52-12"><a href="#cb52-12"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>plot.snv<span class="op">$</span>ca_comb</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-61-1.png" width="2400" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1"></a>plot.snv<span class="op">$</span>co_comb</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-62-1.png" width="2700" /></p>
</div>
<div id="group-mapping" class="section level2">
<h2>Group mapping</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>df.map &lt;-<span class="st"> </span>df.seqz[, <span class="kw">c</span>(</span>
<span id="cb55-2"><a href="#cb55-2"></a>  <span class="st">&quot;sample_type&quot;</span>,</span>
<span id="cb55-3"><a href="#cb55-3"></a>  <span class="st">&quot;cnv_enrich_sig&quot;</span>,</span>
<span id="cb55-4"><a href="#cb55-4"></a>  <span class="st">&quot;snv_enrich_sig&quot;</span></span>
<span id="cb55-5"><a href="#cb55-5"></a>)] <span class="op">%&gt;%</span></span>
<span id="cb55-6"><a href="#cb55-6"></a><span class="st">  </span><span class="kw">na.omit</span>()</span>
<span id="cb55-7"><a href="#cb55-7"></a><span class="kw">names</span>(df.map) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Sample type&quot;</span>, <span class="st">&quot;Dominant CNV signature&quot;</span>, <span class="st">&quot;Dominant SNV signature&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a><span class="kw">show_group_mapping</span>(df.map,</span>
<span id="cb56-2"><a href="#cb56-2"></a>  <span class="dt">col_to_flow =</span> <span class="st">&quot;Sample type&quot;</span>,</span>
<span id="cb56-3"><a href="#cb56-3"></a>  <span class="dt">cols_to_map =</span> <span class="kw">setdiff</span>(<span class="kw">colnames</span>(df.map), <span class="st">&quot;Sample type&quot;</span>),</span>
<span id="cb56-4"><a href="#cb56-4"></a>  <span class="dt">fill_na =</span> <span class="st">&quot;NA&quot;</span>, <span class="dt">include_sig =</span> <span class="ot">TRUE</span></span>
<span id="cb56-5"><a href="#cb56-5"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-64-1.png" width="2400" /></p>
</div>
</div>
<div id="survival-analysis" class="section level1">
<h1>Survival Analysis</h1>
<div id="prepare-1" class="section level2">
<h2>Prepare</h2>
<p>Load package <strong>ezcox</strong> for batch Cox analysis.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a><span class="kw">library</span>(ezcox)</span></code></pre></div>
<p>Load the data.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>df.seqz &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_Merge_Info_CNV_from_sequenza.RData&quot;</span>)</span>
<span id="cb58-2"><a href="#cb58-2"></a>surv_df &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../data/PRAD_Survival.rds&quot;</span>)</span></code></pre></div>
<p>Clean the data to generate a data table for survival analysis.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>range01 &lt;-<span class="st"> </span><span class="cf">function</span>(x, ...) {</span>
<span id="cb59-2"><a href="#cb59-2"></a>  (x <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(x, ...)) <span class="op">/</span><span class="st"> </span>(<span class="kw">max</span>(x, ...) <span class="op">-</span><span class="st"> </span><span class="kw">min</span>(x, ...))</span>
<span id="cb59-3"><a href="#cb59-3"></a>}</span>
<span id="cb59-4"><a href="#cb59-4"></a></span>
<span id="cb59-5"><a href="#cb59-5"></a>cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</span>
<span id="cb59-6"><a href="#cb59-6"></a></span>
<span id="cb59-7"><a href="#cb59-7"></a></span>
<span id="cb59-8"><a href="#cb59-8"></a>surv_dt &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(df.seqz <span class="op">%&gt;%</span></span>
<span id="cb59-9"><a href="#cb59-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="kw">c</span>(</span>
<span id="cb59-10"><a href="#cb59-10"></a>    <span class="st">&quot;Study&quot;</span>, <span class="st">&quot;subject_id&quot;</span>, <span class="st">&quot;tumor_body_site&quot;</span>,</span>
<span id="cb59-11"><a href="#cb59-11"></a>    <span class="st">&quot;tumor_Run&quot;</span>, <span class="st">&quot;normal_Run&quot;</span>, <span class="st">&quot;CNV_ID&quot;</span>,</span>
<span id="cb59-12"><a href="#cb59-12"></a>    <span class="st">&quot;Fusion&quot;</span>, <span class="st">&quot;sample_type&quot;</span></span>
<span id="cb59-13"><a href="#cb59-13"></a>  )),</span>
<span id="cb59-14"><a href="#cb59-14"></a>surv_df,</span>
<span id="cb59-15"><a href="#cb59-15"></a><span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;PatientID&quot;</span> =<span class="st"> &quot;sample&quot;</span>)</span>
<span id="cb59-16"><a href="#cb59-16"></a>)</span>
<span id="cb59-17"><a href="#cb59-17"></a></span>
<span id="cb59-18"><a href="#cb59-18"></a>dup_ids &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">duplicated</span>(surv_dt<span class="op">$</span>PatientID))</span>
<span id="cb59-19"><a href="#cb59-19"></a><span class="co"># Remove duplicated records</span></span>
<span id="cb59-20"><a href="#cb59-20"></a>surv_dt &lt;-<span class="st"> </span>surv_dt[<span class="op">-</span>dup_ids, ]</span>
<span id="cb59-21"><a href="#cb59-21"></a></span>
<span id="cb59-22"><a href="#cb59-22"></a><span class="co"># Scale the signature activity to 0-20.</span></span>
<span id="cb59-23"><a href="#cb59-23"></a><span class="co"># To better understand the effect of signature activity on clinical events,</span></span>
<span id="cb59-24"><a href="#cb59-24"></a><span class="co"># here we scale them into range 0-20. 1 increase means 5% increase of signature activity.</span></span>
<span id="cb59-25"><a href="#cb59-25"></a><span class="co">#</span></span>
<span id="cb59-26"><a href="#cb59-26"></a><span class="co"># Scale the CNA burden to 0-20.</span></span>
<span id="cb59-27"><a href="#cb59-27"></a>surv_dt &lt;-<span class="st"> </span>surv_dt <span class="op">%&gt;%</span></span>
<span id="cb59-28"><a href="#cb59-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(</span>
<span id="cb59-29"><a href="#cb59-29"></a>    <span class="dt">cna_burden =</span> <span class="dv">20</span> <span class="op">*</span><span class="st"> </span>cna_burden,</span>
<span id="cb59-30"><a href="#cb59-30"></a>    <span class="dt">Stage =</span> <span class="kw">as.character</span>(Stage) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">factor</span>(<span class="dt">levels =</span> <span class="kw">c</span>(<span class="st">&quot;T2&quot;</span>, <span class="st">&quot;T3&quot;</span>, <span class="st">&quot;T4&quot;</span>))</span>
<span id="cb59-31"><a href="#cb59-31"></a>  ) <span class="op">%&gt;%</span></span>
<span id="cb59-32"><a href="#cb59-32"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate_at</span>(cols_to_sigs.seqz, <span class="op">~</span><span class="st"> </span><span class="dv">20</span> <span class="op">*</span><span class="st"> </span><span class="kw">range01</span>(., <span class="dt">na.rm =</span> <span class="ot">TRUE</span>))</span>
<span id="cb59-33"><a href="#cb59-33"></a></span>
<span id="cb59-34"><a href="#cb59-34"></a><span class="kw">saveRDS</span>(surv_dt, <span class="dt">file =</span> <span class="st">&quot;../output/PRAD_merged_survival_dt_seqz.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="os-analysis-fo-signatures" class="section level2">
<h2>OS analysis fo signatures</h2>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1"></a><span class="co">############</span></span>
<span id="cb60-2"><a href="#cb60-2"></a>surv_dt &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;../output/PRAD_merged_survival_dt_seqz.rds&quot;</span>)</span>
<span id="cb60-3"><a href="#cb60-3"></a>cols_to_sigs.seqz &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">paste0</span>(<span class="st">&quot;CNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>), <span class="kw">paste0</span>(<span class="st">&quot;SNV_Sig&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>))</span>
<span id="cb60-4"><a href="#cb60-4"></a></span>
<span id="cb60-5"><a href="#cb60-5"></a><span class="co"># Unvariable analysis</span></span>
<span id="cb60-6"><a href="#cb60-6"></a><span class="kw">show_forest</span>(surv_dt,</span>
<span id="cb60-7"><a href="#cb60-7"></a>  <span class="dt">covariates =</span> cols_to_sigs.seqz,</span>
<span id="cb60-8"><a href="#cb60-8"></a>  <span class="dt">time =</span> <span class="st">&quot;OS.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;OS&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span></span>
<span id="cb60-9"><a href="#cb60-9"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-68-1.png" width="2400" /></p>
</div>
<div id="pfs-analysis-for-signatures" class="section level2">
<h2>PFS analysis for signatures</h2>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a><span class="kw">show_forest</span>(surv_dt,</span>
<span id="cb61-2"><a href="#cb61-2"></a>  <span class="dt">covariates =</span> cols_to_sigs.seqz,</span>
<span id="cb61-3"><a href="#cb61-3"></a>  <span class="dt">time =</span> <span class="st">&quot;PFI.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;PFI&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span></span>
<span id="cb61-4"><a href="#cb61-4"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-69-1.png" width="2400" /></p>
</div>
<div id="os-analysis-for-features" class="section level2">
<h2>OS analysis for features</h2>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1"></a>cols_to_features &lt;-<span class="st"> </span><span class="kw">c</span>(</span>
<span id="cb62-2"><a href="#cb62-2"></a>  <span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Stage&quot;</span>, <span class="st">&quot;GleasonScore&quot;</span>,</span>
<span id="cb62-3"><a href="#cb62-3"></a>  <span class="st">&quot;total_mutation&quot;</span>, <span class="st">&quot;n_driver&quot;</span>, <span class="st">&quot;Ti_fraction&quot;</span>, <span class="st">&quot;Tv_fraction&quot;</span>,</span>
<span id="cb62-4"><a href="#cb62-4"></a>  <span class="st">&quot;n_of_cnv&quot;</span>, <span class="st">&quot;n_of_amp&quot;</span>, <span class="st">&quot;n_of_del&quot;</span>, <span class="st">&quot;n_of_vchr&quot;</span>, <span class="st">&quot;cna_burden&quot;</span>,</span>
<span id="cb62-5"><a href="#cb62-5"></a>  <span class="st">&quot;MATH&quot;</span>, <span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;purity&quot;</span>, <span class="st">&quot;ploidy&quot;</span></span>
<span id="cb62-6"><a href="#cb62-6"></a>)</span>
<span id="cb62-7"><a href="#cb62-7"></a></span>
<span id="cb62-8"><a href="#cb62-8"></a><span class="kw">show_forest</span>(surv_dt,</span>
<span id="cb62-9"><a href="#cb62-9"></a>  <span class="dt">covariates =</span> cols_to_features, <span class="dt">controls =</span> <span class="ot">NULL</span>,</span>
<span id="cb62-10"><a href="#cb62-10"></a>  <span class="dt">time =</span> <span class="st">&quot;OS.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;OS&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span>, <span class="dt">limits =</span> <span class="kw">log</span>(<span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">10</span>))</span>
<span id="cb62-11"><a href="#cb62-11"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-70-1.png" width="2400" /></p>
</div>
<div id="pfi-analysis-for-features" class="section level2">
<h2>PFI analysis for features</h2>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="kw">show_forest</span>(surv_dt,</span>
<span id="cb63-2"><a href="#cb63-2"></a>  <span class="dt">covariates =</span> cols_to_features, <span class="dt">controls =</span> <span class="ot">NULL</span>,</span>
<span id="cb63-3"><a href="#cb63-3"></a>  <span class="dt">time =</span> <span class="st">&quot;PFI.time&quot;</span>, <span class="dt">status =</span> <span class="st">&quot;PFI&quot;</span>, <span class="dt">merge_models =</span> <span class="ot">TRUE</span>, <span class="dt">limits =</span> <span class="kw">c</span>(<span class="kw">log</span>(<span class="fl">0.5</span>), <span class="kw">log</span>(<span class="dv">5</span>))</span>
<span id="cb63-4"><a href="#cb63-4"></a>)</span></code></pre></div>
<p><img src="index_files/figure-html/unnamed-chunk-71-1.png" width="2400" /></p>
</div>
</div>
<div id="variation-summary" class="section level1">
<h1>Variation Summary</h1>
</div>
<div id="supplementary-analysis-and-visualization" class="section level1">
<h1>Supplementary Analysis and Visualization</h1>
</div>
<div id="reference" class="section level1">
<h1 class="unnumbered">Reference</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-alexandrov2013signatures">
<p>Alexandrov, Ludmil B, Serena Nik-Zainal, David C Wedge, Samuel AJR Aparicio, Sam Behjati, Andrew V Biankin, Graham R Bignell, et al. 2013. “Signatures of Mutational Processes in Human Cancer.” <em>Nature</em> 500 (7463): 415.</p>
</div>
<div id="ref-gaujoux2010flexible">
<p>Gaujoux, Renaud, and Cathal Seoighe. 2010. “A Flexible R Package for Nonnegative Matrix Factorization.” <em>BMC Bioinformatics</em> 11 (1): 367.</p>
</div>
<div id="ref-macintyre2018copy">
<p>Macintyre, Geoff, Teodora E Goranova, Dilrini De Silva, Darren Ennis, Anna M Piskorz, Matthew Eldridge, Daoud Sie, et al. 2018. “Copy Number Signatures and Mutational Processes in Ovarian Carcinoma.” <em>Nature Genetics</em> 50 (9): 1262.</p>
</div>
<div id="ref-mayakonda2016maftools">
<p>Mayakonda, Anand, and H Phillip Koeffler. 2016. “Maftools: Efficient Analysis, Visualization and Summarization of Maf Files from Large-Scale Cohort Based Cancer Studies.” <em>BioRxiv</em>, 052662.</p>
</div>
<div id="ref-yi2018patterns">
<p>Yi, Kijong, and Young Seok Ju. 2018. “Patterns and Mechanisms of Structural Variations in Human Cancer.” <em>Experimental &amp; Molecular Medicine</em> 50 (8): 98.</p>
</div>
</div>
</div>
</div>


</div>

<div id="postamble" data-toggle="wy-nav-shift" class="status">
<p class="author"><span class="glyphicon glyphicon-user"></span> Shixiang Wang</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Huimin Li</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Xuan Wang</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Minfang Song</p>
<p class="author"><span class="glyphicon glyphicon-user"></span> Xue-Song Liu (Corresponding author)</p>
<p class="date"><span class="glyphicon glyphicon-calendar"></span> 2020-01-11</p>
</div>


<script>
$(document).ready(function () {
 	    $('#content img')
 	  .addClass("image-lb");
  $('#content').magnificPopup({
	      type:'image',
	      closeOnContentClick: false,
	      closeBtnInside: false,
	      delegate: 'img',
	      gallery: {enabled: true },
	      image: {
	        verticalFit: true,
          titleSrc: 'alt'
	      }
 	    });
 	});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
